{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import glob\n",
    "import json\n",
    "import re"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def return_useful_data(r_out):\n",
    "    git_hits = 0\n",
    "    osf_hits = 0\n",
    "    nda_hits = 0\n",
    "    open_neuro = 0\n",
    "    ndar_hits = 0\n",
    "    ndar_long_hits = 0\n",
    "    #fcon1000_hits = 0 this is the same as indi?\n",
    "    allen_institute_hits = 0\n",
    "    hcp_hits = 0\n",
    "    balsa_hits = 0\n",
    "    loni_hits = 0\n",
    "    loni_narrow_hits = 0\n",
    "    fmridc_hits = 0\n",
    "    ccrns_hits = 0\n",
    "    datalad_hits = 0\n",
    "    dataverse_hits = 0\n",
    "    dbgap_hits = 0\n",
    "    dbgap_url_hits = 0\n",
    "    dryad_hits = 0\n",
    "    figshare_hits = 0\n",
    "    indi_hits = 0\n",
    "    nitrc_hits = 0\n",
    "    omega_hits = 0\n",
    "    xnat_hits = 0\n",
    "    zenodo_hits = 0\n",
    "    aws_data_hits = 0\n",
    "\n",
    "    try:\n",
    "        doi = r_out['documents'][0]['passages'][0]['infons']['article-id_doi']\n",
    "    except:\n",
    "        doi = None\n",
    "    try:\n",
    "        pmcid = r_out['documents'][0]['id']\n",
    "    except:\n",
    "        pmcid = None\n",
    "    for passage in r_out['documents'][0]['passages']:\n",
    "        if re.search(re.compile('github'), passage['text'].lower()):\n",
    "            git_hits = git_hits + 1\n",
    "        if re.search(re.compile('osf\\.io'), passage['text'].lower()):\n",
    "            osf_hits = osf_hits + 1\n",
    "        if re.search(re.compile('nda\\.nih\\.gov'), passage['text'].lower()):\n",
    "            nda_hits = nda_hits + 1\n",
    "        if re.search(re.compile('openneuro'), passage['text'].lower()):\n",
    "            open_neuro = open_neuro + 1\n",
    "        if re.search(re.compile('ndar'), passage['text'].lower()):\n",
    "            ndar_hits = ndar_hits + 1    \n",
    "        if re.search(re.compile('national database for autism research'), passage['text'].lower()):\n",
    "            ndar_long_hits = ndar_long_hits + 1 \n",
    "        if re.search(re.compile('brain-map\\.org'), passage['text'].lower()):\n",
    "            allen_institute_hits = allen_institute_hits + 1 \n",
    "        if re.search(re.compile('humanconnectome\\.org'), passage['text'].lower()):\n",
    "            hcp_hits = hcp_hits + 1 \n",
    "        if re.search(re.compile('balsa\\.wustl\\.edu'), passage['text'].lower()):\n",
    "            balsa_hits = balsa_hits + 1 \n",
    "        if re.search(re.compile('loni\\.usc\\.edu'), passage['text'].lower()):\n",
    "            loni_hits = loni_hits + 1\n",
    "        if re.search(re.compile('ida\\.loni\\.usc\\.edu'), passage['text'].lower()):\n",
    "            loni_narrow_hits = loni_narrow_hits + 1\n",
    "        if re.search(re.compile('fmridc'), passage['text'].lower()):\n",
    "            fmridc_hits = fmridc_hits + 1\n",
    "        if re.search(re.compile('ccrns'), passage['text'].lower()):\n",
    "            ccrns_hits = ccrns_hits + 1\n",
    "        if re.search(re.compile('datalad'), passage['text'].lower()):\n",
    "            datalad_hits = datalad_hits + 1\n",
    "        if re.search(re.compile('dataverse'), passage['text'].lower()):\n",
    "            dataverse_hits = dataverse_hits + 1\n",
    "        if re.search(re.compile('dbgap'), passage['text'].lower()):\n",
    "            dbgap_hits = dbgap_hits + 1\n",
    "        if re.search(re.compile('nih\\.gov\\/gap'), passage['text'].lower()):\n",
    "            dbgap_url_hits = dbgap_url_hits + 1\n",
    "        if re.search(re.compile('dryad'), passage['text'].lower()):\n",
    "            dryad_hits = dryad_hits + 1\n",
    "        if re.search(re.compile('figshare'), passage['text'].lower()):\n",
    "            figshare_hits = figshare_hits + 1\n",
    "        if re.search(re.compile('fcon_1000\\.projects'), passage['text'].lower()):\n",
    "            indi_hits = indi_hits + 1\n",
    "        if re.search(re.compile('nitrc'), passage['text'].lower()):\n",
    "            nitrc_hits = nitrc_hits + 1\n",
    "        if re.search(re.compile('mcgill\\.ca\\/bic\\/resources\\/omega'), passage['text'].lower()):\n",
    "            omega_hits = omega_hits + 1\n",
    "        if re.search(re.compile('xnat\\.org'), passage['text'].lower()):\n",
    "            xnat_hits = xnat_hits + 1\n",
    "        if re.search(re.compile('zenodo'), passage['text'].lower()):\n",
    "            zenodo_hits = zenodo_hits + 1\n",
    "        if re.search(re.compile('opendata\\.aws'), passage['text'].lower()):\n",
    "            aws_data_hits = aws_data_hits + 1\n",
    "        \n",
    "    return(doi, pmcid, git_hits, osf_hits, nda_hits, open_neuro, ndar_hits, \n",
    "           ndar_long_hits, allen_institute_hits, hcp_hits, balsa_hits, loni_hits,\n",
    "          loni_narrow_hits, fmridc_hits, ccrns_hits, datalad_hits, dataverse_hits,\n",
    "          dbgap_hits, dryad_hits, figshare_hits, indi_hits, nitrc_hits, omega_hits,\n",
    "          xnat_hits, zenodo_hits, aws_data_hits)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "already_retrieved = glob.glob('/home/riddleta/ac_knowl/output/full_texts/papes*')\n",
    "already_retrieved = sorted(already_retrieved)\n",
    "status_prints = already_retrieved[::2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/riddleta/ac_knowl/output/full_texts/papes_0.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_100000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1002500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1007500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1012500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1017500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1022500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1025000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1030000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1035000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1040000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1045000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_105000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1052500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1057500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1062500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1067500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1072500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1075000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1080000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1085000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1090000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1095000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_110000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1102500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1107500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1112500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1117500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1122500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1125000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1130000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1135000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1140000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1145000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_115000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1152500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1157500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1162500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1167500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1172500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1175000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1180000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1185000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1190000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1195000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_120000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1202500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1207500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1212500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1217500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1222500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1225000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1230000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1235000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1240000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1245000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_12500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1250000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1255000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1260000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1265000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1270000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_127500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1277500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1282500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1287500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1292500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1297500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1300000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1305000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1310000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1315000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1320000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_132500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1327500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1332500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1337500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1342500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1347500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1350000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1355000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1360000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1365000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1370000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_137500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1377500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1382500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1387500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1392500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1397500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1400000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1405000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1410000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1415000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1420000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_142500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1427500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1432500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1437500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1442500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1447500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1450000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1455000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1460000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1465000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1470000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_147500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1477500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1482500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1487500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1492500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1497500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_150000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1502500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1507500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1512500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1517500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1522500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1525000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1530000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1535000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1540000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1545000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_155000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1552500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1557500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1562500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1567500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1572500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1575000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1580000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1585000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1590000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1595000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_160000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1602500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1607500.txt\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/riddleta/ac_knowl/output/full_texts/papes_1612500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1617500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1622500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1625000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1630000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1635000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1640000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1645000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_165000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1652500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1657500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1662500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1667500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1672500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1675000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1680000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1685000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1690000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1695000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_170000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1702500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1707500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1712500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1717500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1722500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1725000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1730000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1735000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1740000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1745000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_17500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1750000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1755000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1760000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1765000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1770000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_177500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1777500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1782500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1787500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1792500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1797500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1800000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1805000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1810000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1815000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1820000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_182500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1827500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1832500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1837500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1842500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1847500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1850000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1855000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1860000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1865000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1870000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_187500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1877500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1882500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1887500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1892500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1897500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1900000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1905000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1910000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1915000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1920000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_192500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1927500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1932500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1937500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1942500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1947500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1950000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1955000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1960000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1965000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1970000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_197500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1977500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1982500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1987500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1992500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_1997500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_200000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2002500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2007500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2012500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2017500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2022500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2025000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2030000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2035000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2040000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2045000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_205000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2052500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2057500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2062500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2067500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2072500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2075000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2080000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2085000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2090000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2095000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_210000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2102500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2107500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2112500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2117500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2122500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2125000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2130000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2135000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2140000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2145000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_215000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2152500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2157500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2162500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2167500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2172500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2175000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2180000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2185000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2190000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2195000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_220000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2202500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2207500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2212500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2217500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2222500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2225000.txt\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/riddleta/ac_knowl/output/full_texts/papes_2230000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2235000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2240000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2245000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_22500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2250000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2255000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2260000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2265000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2270000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_227500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2277500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2282500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2287500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2292500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2297500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2300000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2305000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2310000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2315000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2320000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_232500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2327500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2332500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2337500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2342500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2347500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2350000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2355000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2360000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2365000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2370000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_237500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2377500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2382500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2387500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2392500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2397500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2400000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2405000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2410000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2415000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2420000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_242500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2427500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2432500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2437500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2442500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2447500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2450000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2455000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2460000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2465000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2470000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_247500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2477500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2482500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2487500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2492500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2497500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_25000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2500000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2505000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2510000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2515000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2520000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_252500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2527500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2532500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2537500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2542500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2547500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2550000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2555000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2560000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2565000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2570000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_257500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2577500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2582500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2587500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2592500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2597500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2600000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2605000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2610000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_2615000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_262500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_267500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_272500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_275000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_280000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_285000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_290000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_295000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_30000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_302500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_307500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_312500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_317500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_322500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_325000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_330000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_335000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_340000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_345000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_35000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_352500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_357500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_362500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_367500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_372500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_375000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_380000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_385000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_390000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_395000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_40000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_402500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_407500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_412500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_417500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_422500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_425000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_430000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_435000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_440000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_445000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_45000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_452500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_457500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_462500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_467500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_472500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_475000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_480000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_485000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_490000.txt\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/riddleta/ac_knowl/output/full_texts/papes_495000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_5000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_500000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_505000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_510000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_515000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_520000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_52500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_527500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_532500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_537500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_542500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_547500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_550000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_555000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_560000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_565000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_570000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_57500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_577500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_582500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_587500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_592500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_597500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_600000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_605000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_610000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_615000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_620000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_62500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_627500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_632500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_637500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_642500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_647500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_650000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_655000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_660000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_665000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_670000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_67500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_677500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_682500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_687500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_692500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_697500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_700000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_705000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_710000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_715000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_720000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_72500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_727500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_732500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_737500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_742500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_747500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_75000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_752500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_757500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_762500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_767500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_772500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_775000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_780000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_785000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_790000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_795000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_80000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_802500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_807500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_812500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_817500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_822500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_825000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_830000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_835000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_840000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_845000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_85000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_852500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_857500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_862500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_867500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_872500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_875000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_880000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_885000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_890000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_895000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_90000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_902500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_907500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_912500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_917500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_922500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_925000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_930000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_935000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_940000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_945000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_95000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_952500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_957500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_962500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_967500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_972500.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_975000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_980000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_985000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_990000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_995000.txt\n",
      "/home/riddleta/ac_knowl/output/full_texts/papes_stragglers.txt\n"
     ]
    }
   ],
   "source": [
    "data_collect = []\n",
    "\n",
    "for i in already_retrieved:\n",
    "    if i in status_prints:\n",
    "        print(i)\n",
    "    with open(i) as infile:\n",
    "        dat = json.load(infile)\n",
    "        for j, paper in enumerate(dat):\n",
    "            data_collect.append(return_useful_data(paper))    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.DataFrame(data_collect, columns = ['doi', 'pmcid', 'git_hits', 'osf_hits', 'nda_hits',\n",
    "                                          'open_neuro', 'ndar_hits', 'ndar_long_hits', 'allen_institute_hits',\n",
    "                                          'hcp_hits', 'balsa_hits', 'loni_hits', 'loni_narrow_hits', 'fmridc_hits',\n",
    "                                          'ccrns_hits', 'datalad_hits', 'dataverse_hits', 'dbgap_hits',\n",
    "                                          'dryad_hits', 'figshare_hits', 'indi_hits', 'nitrc_hits', 'omega_hits',\n",
    "                                          'xnat_hits', 'zenodo_hits', 'aws_data_hits'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(2616185, 26)"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "git_hits                  25378\n",
       "osf_hits                   1852\n",
       "nda_hits                      0\n",
       "open_neuro                   39\n",
       "ndar_hits               2047012\n",
       "ndar_long_hits              155\n",
       "allen_institute_hits       1356\n",
       "hcp_hits                    274\n",
       "balsa_hits                   12\n",
       "loni_hits                   923\n",
       "loni_narrow_hits             66\n",
       "fmridc_hits                  70\n",
       "ccrns_hits                    9\n",
       "datalad_hits                 15\n",
       "dataverse_hits              469\n",
       "dbgap_hits                 3150\n",
       "dryad_hits                 7118\n",
       "figshare_hits              5296\n",
       "indi_hits                   392\n",
       "nitrc_hits                 2290\n",
       "omega_hits                    1\n",
       "xnat_hits                    46\n",
       "zenodo_hits                2301\n",
       "aws_data_hits                 1\n",
       "dtype: int64"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.agg({'git_hits':lambda x: sum(x>0),\n",
    "        'osf_hits':lambda x: sum(x>0),\n",
    "        'nda_hits':lambda x: sum(x>0),\n",
    "        'open_neuro':lambda x: sum(x>0),\n",
    "        'ndar_hits':lambda x: sum(x>0),\n",
    "        'ndar_long_hits':lambda x: sum(x>0),\n",
    "        'allen_institute_hits':lambda x: sum(x>0),\n",
    "        'hcp_hits':lambda x: sum(x>0),\n",
    "        'balsa_hits':lambda x: sum(x>0),\n",
    "        'loni_hits':lambda x: sum(x>0),\n",
    "        'loni_narrow_hits':lambda x: sum(x>0),\n",
    "        'fmridc_hits':lambda x: sum(x>0),\n",
    "        'ccrns_hits':lambda x: sum(x>0),\n",
    "        'datalad_hits':lambda x: sum(x>0),\n",
    "        'dataverse_hits':lambda x: sum(x>0),\n",
    "        'dbgap_hits':lambda x: sum(x>0),\n",
    "        'dryad_hits':lambda x: sum(x>0),\n",
    "        'figshare_hits':lambda x: sum(x>0),\n",
    "        'indi_hits':lambda x: sum(x>0),\n",
    "        'nitrc_hits':lambda x: sum(x>0),\n",
    "        'omega_hits':lambda x: sum(x>0),\n",
    "        'xnat_hits':lambda x: sum(x>0),\n",
    "        'zenodo_hits':lambda x: sum(x>0),\n",
    "        'aws_data_hits':lambda x: sum(x>0)})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>doi</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>git_hits</th>\n",
       "      <th>osf_hits</th>\n",
       "      <th>nda_hits</th>\n",
       "      <th>open_neuro</th>\n",
       "      <th>ndar_hits</th>\n",
       "      <th>ndar_long_hits</th>\n",
       "      <th>allen_institute_hits</th>\n",
       "      <th>hcp_hits</th>\n",
       "      <th>...</th>\n",
       "      <th>dataverse_hits</th>\n",
       "      <th>dbgap_hits</th>\n",
       "      <th>dryad_hits</th>\n",
       "      <th>figshare_hits</th>\n",
       "      <th>indi_hits</th>\n",
       "      <th>nitrc_hits</th>\n",
       "      <th>omega_hits</th>\n",
       "      <th>xnat_hits</th>\n",
       "      <th>zenodo_hits</th>\n",
       "      <th>aws_data_hits</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10.1186/s12969-016-0125-1</td>\n",
       "      <td>5139083</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10.1371/journal.pone.0121406</td>\n",
       "      <td>4388569</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10.1371/journal.pone.0121575</td>\n",
       "      <td>4388570</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10.1371/journal.pgen.1004969</td>\n",
       "      <td>4388571</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10.1371/journal.pone.0122216</td>\n",
       "      <td>4388572</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                            doi    pmcid  git_hits  osf_hits  nda_hits  \\\n",
       "0     10.1186/s12969-016-0125-1  5139083         0         0         0   \n",
       "1  10.1371/journal.pone.0121406  4388569         0         0         0   \n",
       "2  10.1371/journal.pone.0121575  4388570         0         0         0   \n",
       "3  10.1371/journal.pgen.1004969  4388571         0         0         0   \n",
       "4  10.1371/journal.pone.0122216  4388572         0         0         0   \n",
       "\n",
       "   open_neuro  ndar_hits  ndar_long_hits  allen_institute_hits  hcp_hits  ...  \\\n",
       "0           0          2               0                     0         0  ...   \n",
       "1           0          8               0                     0         0  ...   \n",
       "2           0          3               0                     0         0  ...   \n",
       "3           0         11               0                     0         0  ...   \n",
       "4           0          4               0                     0         0  ...   \n",
       "\n",
       "   dataverse_hits  dbgap_hits  dryad_hits  figshare_hits  indi_hits  \\\n",
       "0               0           0           0              0          0   \n",
       "1               0           0           0              0          0   \n",
       "2               0           0           0              0          0   \n",
       "3               0           0           0              0          0   \n",
       "4               0           0           0              0          0   \n",
       "\n",
       "   nitrc_hits  omega_hits  xnat_hits  zenodo_hits  aws_data_hits  \n",
       "0           0           0          0            0              0  \n",
       "1           0           0          0            0              0  \n",
       "2           0           0          0            0              0  \n",
       "3           0           0          0            0              0  \n",
       "4           0           0          0            0              0  \n",
       "\n",
       "[5 rows x 26 columns]"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>doi</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>git_hits</th>\n",
       "      <th>osf_hits</th>\n",
       "      <th>nda_hits</th>\n",
       "      <th>open_neuro</th>\n",
       "      <th>ndar_hits</th>\n",
       "      <th>ndar_long_hits</th>\n",
       "      <th>allen_institute_hits</th>\n",
       "      <th>hcp_hits</th>\n",
       "      <th>...</th>\n",
       "      <th>dataverse_hits</th>\n",
       "      <th>dbgap_hits</th>\n",
       "      <th>dryad_hits</th>\n",
       "      <th>figshare_hits</th>\n",
       "      <th>indi_hits</th>\n",
       "      <th>nitrc_hits</th>\n",
       "      <th>omega_hits</th>\n",
       "      <th>xnat_hits</th>\n",
       "      <th>zenodo_hits</th>\n",
       "      <th>aws_data_hits</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1651659</th>\n",
       "      <td>10.1093/molbev/msw151</td>\n",
       "      <td>5026260</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                           doi    pmcid  git_hits  osf_hits  nda_hits  \\\n",
       "1651659  10.1093/molbev/msw151  5026260         0         0         0   \n",
       "\n",
       "         open_neuro  ndar_hits  ndar_long_hits  allen_institute_hits  \\\n",
       "1651659           0          0               0                     0   \n",
       "\n",
       "         hcp_hits  ...  dataverse_hits  dbgap_hits  dryad_hits  figshare_hits  \\\n",
       "1651659         0  ...               0           0           0              0   \n",
       "\n",
       "         indi_hits  nitrc_hits  omega_hits  xnat_hits  zenodo_hits  \\\n",
       "1651659          0           0           0          0            0   \n",
       "\n",
       "         aws_data_hits  \n",
       "1651659              0  \n",
       "\n",
       "[1 rows x 26 columns]"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df.pmcid=='5026260']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "id_str = 'PMC5026260'\n",
    "pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + id_str + '/unicode'\n",
    "r = requests.get(pth)\n",
    "#git_links = []\n",
    "#missed = []\n",
    "\n",
    "#for i, j in enumerate(unique_ids):\n",
    "#    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + j + '/unicode'\n",
    "#    try:\n",
    "#        r = requests.get(pth)\n",
    "#        git_links.append(return_useful_data(r) + (j, i))\n",
    "#    except:\n",
    "#        missed.append([i,j])\n",
    "\n",
    "#df = pd.DataFrame(git_links, columns=['date', 'src', 'doi', 'git_hits', \n",
    "#                                      'osf_hits', 'nda_hits', 'open_neuro', \n",
    "#                                      'fmri', 'res', 'yr', 'pmcid', 'idx'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Intron Length Coevolution across Mammalian Genomes\n",
      "Although they do not contribute directly to the proteome, introns frequently contain regulatory elements and can extend the protein coding potential of the genome through alternative splicing. For some genes, the contribution of introns to the time required for transcription can also be functionally significant. We have previously shown that intron length in genes associated with developmental patterning is often highly conserved. In general, sets of genes that require precise coordination in the timing of their expression may be sensitive to changes in transcript length. A prediction of this hypothesis is that evolutionary changes in intron length, when they occur, may be correlated between sets of coordinately expressed genes. To test this hypothesis, we analyzed intron length coevolution in alignments from nine eutherian mammals. Overall, genes that belong to the same protein complex or that are coexpressed were significantly more likely to show evidence of intron length coevolution than matched, randomly sampled genes. Individually, protein complexes involved in the cell cycle showed the strongest evidence of coevolution of intron lengths and clusters of coexpressed genes enriched for cell cycle genes also showed significant evidence of intron length coevolution. Our results reveal a novel aspect of gene coevolution and provide a means to identify genes, protein complexes and biological processes that may be particularly sensitive to changes in transcriptional dynamics.\n",
      "Introduction\n",
      "The origin, prevalence, and function of introns in eukaryotic genomes have long been a subject of great interest in molecular evolution. Although typically they do not contribute directly to the protein coding sequences of genes, introns have been shown to play several important functional roles. Introns frequently contain regulatory elements and allow for the process of alternative splicing, expanding the protein coding capacity of the genome, and giving rise to additional modes of gene regulation. In mammals, introns comprise up to 95% of transcribed primary protein-coding sequences, imposing a significant cost in terms of the energy required for transcription and in terms of the potential for erroneous transcripts resulting from mis-splicing, particularly for genes with longer introns. The time taken to transcribe introns can be significant. Estimates of the rate of transcription in eukaryotes range from 1 to 4 Kb per minute with a median rate of transcription estimated from over 2,000 human genes at 1.5 Kb per minute. As an extreme example of the time taken to transcribe long introns, the human dystrophin gene, which is 2.3 Mb in length, 99% of which is intronic, can take up to 16 h to transcribe, corresponding to a rate of 2.4 Kb per minute. In contrast, genes that are highly expressed tend to have short introns, and genes that require rapid induction in response to stress are intron poor.\n",
      "The evolution and functional significance of large introns in mammals has been much debated.) proposed that intron expansion is a largely a semi-neutral process, resulting from insufficient purifying selection to remove insertions in organisms with small effective population sizes. The genome design model asserts that intron length reflects gene function. This model proposes that housekeeping genes tend to have short introns because they require very little regulation. Conversely, genes that require more complex modes of gene regulation, such as tissue-specific genes are longer due, in part, to the presence of intronic regulatory elements. In opposition to this view, a number of authors have argued that the short introns in highly and ubiquitously expressed genes, such as housekeeping genes, are the result of selection for efficiency, resulting in reduced energetic costs of transcription.\n",
      "For some genes, the transcriptional delay resulting from the time taken to transcribe introns is functionally significant. This intron delay is particularly important in embryonic development, where precise temporal control of gene expression is required to ensure proper development of the embryo. During embryonic development, cells undergo rapid divisions creating a temporal constraint on transcription. In Drosophila, introns in genes expressed by the embryo during this period are short, allowing for rapid transcription between cell cycles. Intron length can also contribute to more complex forms of gene regulation. Delayed expression, resulting in part from the transcription of introns, combined with negative feedback loops, can create oscillating patterns of gene expression. The Hes7 gene, which is involved in somite segmentation during development, represents an example of this. Removal of introns from Hes7 in mouse resulted in Hes7 being expressed 19 min early and the abolition of expression oscillations, resulting in severe developmental defects. The intron lengths, though in general not the intron sequences, of Hes7 as well as several other genes involved in developmental patterning are highly conserved, suggesting that these genes are under purifying selection to maintain intron length. Intron length also appears to be relevant in protein complex formation as genes that occur within the same complex usually have very similar length.\n",
      "Here, we investigated the extent of intron length coevolution across several mammalian genomes using whole genome alignments together with reconstructed ancestral sequences. We hypothesized that sets of genes that require precise coordination in the timing of their expression may be sensitive to changes in intron length and that natural selection can act to maintain the relative lengths of such genes, resulting in correlated changes in intron content. We found that, overall, coexpressed genes and genes belonging to the same protein complex were significantly more likely to have coevolving intron lengths than randomly sampled genes, matched for total intron content. We also found a number of gene modules that showed particularly strongly correlated intron length evolution that may reflect selection acting on relative transcript lengths. We propose that transcript length coevolution can reveal functional constraints on intron lengths and that intron length coevolution may suggest biological processes that require precise temporal regulation of gene expression.\n",
      "Results\n",
      "Phylogenetic tree of the nine mammals included in this study.\n",
      "We hypothesized that sets of genes that require precise coordination in the timing of their expression may be particularly sensitive to evolutionary changes in intron content, and that changes in intron content due to insertions and deletions would be correlated among such sets of genes. To identify sets of genes where this may be the case, we compared changes in intron content across the species phylogeny for 9,396 genes from nine mammalian genomes. The phylogenetic tree representing the nine species used in this analysis is shown in figure 1.  \n",
      "Protein Complexes Show Significant Evidence of Intron Length Coevolution\n",
      "Multi-protein complexes are often precisely coregulated in order to ensure proper assembly and function and thus we predicted that the intron content of members of some protein complexes may display evidence of coevolution. Subunits of protein complexes tend to be coexpressed, a pattern which is conserved across a wide range of species. Studies in yeast have shown that gene expression levels among sets of interacting proteins coevolve, suggesting that the mechanisms that control this coregulation may also coevolve. It has also been suggested that gene length contributes to the coregulation of protein complexes, as genes encoding subunits of multi-protein complexes tend to have similar lengths. Furthermore, protein–protein interactions are known to be largely conserved across mammals.\n",
      "We sought to determine whether intron length coevolution could be detected among subunits of known protein complexes, potentially representing a novel aspect of the coevolutionary dynamics of members of multi-protein complexes. To do this, we downloaded all human protein complexes from the core set of the CORUM database of mammalian protein complexes. We then carried out a randomization test to determine if members of the same protein complex show significantly greater intron length correlation compared to sets of randomly sampled genes with similar intron content (see “Materials and Methods” section). For this test, we only considered complexes for which at least three genes were present in our original 9,396 gene dataset, resulting in 440 complexes that could be tested.\n",
      "Distribution of the mean partial Pearson correlation values from 10,000 randomizations of (A) CORUM protein complexes and (B) GTEx coexpression clusters. The mean value of the real data is shown as a red dot.\n",
      "Distribution of p-values obtained from 10,000 randomizations of (A) CORUM protein complexes and (B) GTEx coexpression clusters. π0, the expected proportion of tests consistent with the null hypothesis is shown as a red dotted line. The uniform distribution, expected under the null hypothesis is shown in blue.\n",
      "Top 10 Significantly Coevolving CORUM Protein Complexes.\n",
      "CORUM complex ID\tComplex description\tNo. of genes analyzed\tMean Pearson coefficient\tP-value\tq-value\n",
      "1413\tNCOR1 complex\t8\t0.43\t<0.0001\t0.04\n",
      "2446\tITGA9-ITGB1-FIGF complex\t3\t0.77\t0.001\t0.12\n",
      "5589\tLINC complex, S-phase\t6\t0.42\t0.001\t0.16\n",
      "799\tDMAP1-associated complex\t8\t0.32\t0.002\t0.16\n",
      "1179\tCENP-A NAC-CAD complex\t6\t0.42\t0.002\t0.16\n",
      "657\tRetromer complex (SNX1, SNX2, VPS35, VPS29, VPS26A)\t3\t0.66\t0.004\t0.23\n",
      "570\tp300-CBP-p270-SWI/SNF complex\t6\t0.36\t0.005\t0.26\n",
      "781\tURI complex (Unconventional prefoldin RPB5 Interactor)\t6\t0.35\t0.005\t0.26\n",
      "351\tSpliceosome\t69\t0.04\t0.007\t0.28\n",
      "1332\tLarge Drosha complex\t10\t0.19\t0.007\t0.28\n",
      "\n",
      "First, we asked if intron length coevolution could be detected by considering all interacting proteins from across all protein complexes at once. This was done using a randomization test described in the “Materials and Methods” section. We found evidence of intron length coevolution of genes belonging to the same protein complex using this approach (P < 0.0001; fig. 2A). We next asked which protein complexes showed the greatest evidence of coevolution. We repeated the randomization test, this time considering each protein complex individually. Of the 440 complexes tested, the members of 44 clusters showed evidence of intron length coevolution (uncorrected P < 0.05), the top 10 of which are shown in table 1. Given that there was overall evidence of coevolution of members of the same protein complex and an excess of nominally significant complexes over the null expectation we applied the q-value method to investigate properties of the collection of statistical tests applied to protein complexes. The proportion, π0, of tests estimated to conform to the null hypothesis was 0.9, indicating that 10% of tests were not consistent with the null hypothesis (fig. 3A). Just one complex, annotated in CORUM as the NCOR1 complex, had a q-value below 0.05 (q-value = 0.04). The full list of nominally significant complexes and their q-values is provided in supplementary table S1, Supplementary Material online.   \n",
      "Intron Length Coevolution in Coexpressed Genes\n",
      "Significantly Coevolving GTEx Coexpression Clusters.\n",
      "ID\tMost significantly enriched GO Biological Process\tNo. of genes\tMean correlation\tP-value\tq-value\n",
      "2\tSynaptic transmission\t429\t0.06\t<0.0001\t<0.0001\n",
      "9\tNucleic acid metabolic process\t83\t0.09\t<0.0001\t<0.0001\n",
      "5\tCell cycle\t171\t0.04\t0.0003\t0.002\n",
      "4\tSynaptic transmission\t213\t0.05\t0.004\t0.004\n",
      "8\tncRNA metabolic process\t122\t0.03\t0.01\t0.031\n",
      "15\tRibosome biogenesis\t85\t0.04\t0.015\t0.037\n",
      "30\tCardiac muscle tissue development\t30\t0.1\t0.02\t0.037\n",
      "14\tExtracellular muscle organization\t102\t0.05\t0.021\t0.037\n",
      "21\tExtracellular muscle organization\t49\t0.08\t0.021\t0.037\n",
      "19\tCellular respiration\t64\t0.04\t0.024\t0.038\n",
      "10\tMuscle system process\t110\t0.03\t0.03\t0.038\n",
      "13\tCarboxylic acid metabolic process\t63\t0.04\t0.033\t0.038\n",
      "11\tTranslation elongation\t34\t0.05\t0.033\t0.038\n",
      "34\tPeptide hormone processing\t19\t0.09\t0.038\t0.038\n",
      "46\tCarboxylic acid transport\t7\t0.2\t0.038\t0.038\n",
      "1\tSpermatogenesis\t511\t0.02\t0.039\t0.038\n",
      "44\tSurfactant homeostasis\t7\t0.19\t0.049\t0.045\n",
      "\n",
      "We next asked if intron length coevolution is a general feature of genes that are coregulated. Sets of genes that are coregulated include coexpressed genes. Indeed, interacting proteins are often coexpressed. We constructed a gene coexpression network containing 60 clusters using publicly available gene expression data from the Genotype-Tissue Expression (GTEx) consortium (see supplementary table S2, Supplementary Material online, for a list of genes included in each coexpression cluster). Although we allowed for overlapping clusters in our method (see “Materials and Methods” section), we found that the clusters identified were largely nonoverlapping and independent (supplementary fig. S1, Supplementary Material online). For this analysis, we only considered the 50 clusters for which at least three genes were present in our original 9,396 gene dataset. To first test if intron length coevolution could be detected across the gene coexpression network as a whole, we applied the randomization test used for protein complexes to the 50 clusters. Similarly to protein complexes, we found that coexpressed genes were significantly more likely to be coevolving compared to sets of random genes (P < 0.0001; fig. 2B). We next applied this test to each cluster individually to determine which clusters were most enriched for intron length coevolution. We found 17 clusters with significant evidence of coevolution (P < 0.05; uncorrected P-values; table 2). π0 was estimated to be 0.3, suggesting that 70% of tests were not consistent with the null hypothesis (fig. 3B). All 17 tests remained significant after correction for multiple testing using the q-value method (q-value < 0.05; table 2). These clusters are highlighted in the gene coexpression network in supplementary figure S1, Supplementary Material online. \n",
      "It is possible that the intron content coevolution we observed for coexpressed genes or members of protein complexes reflects similar patterns of intron length change in different expression classes (e.g., highly or broadly expressed genes). To explore this possibility, we repeated the randomization tests, matching on median, or maximum gene expression across GTEx tissues in addition to intron length (see “Materials and Methods” section). The randomization test remained significant for both the protein complexes (P = 0.0016 and P = 0.0001, matching on median or maximum expression, respectively) and gene coexpression clusters (P = 0.0002 and P < 0.0001). Thus, we did not find evidence that expression class was the driver of intron content coevolution of members of protein complexes or coexpression clusters.\n",
      "We also considered the possibility that correlated change in intron length of coexpressed genes and members of protein complexes results from genomic colocation. It has previously been shown that genes in close proximity to each other tend to have similar patterns of gene expression and similarly sized introns. To investigate the possible impact of this on our analysis, we repeated our randomization test but this time excluding all gene pairs within a genomic distance of 1 Mb. We found that this did not significantly alter our results, as both protein complexes and coexpressed genes were still found to be significant after the exclusion of these gene pairs (P < 0.0001 for both the protein complex and coexpressed genes). Therefore, we conclude that chromosomal location or linkage does not significantly contribute to the patterns of intron length coevolution we observed.\n",
      "Functional Enrichment in Coevolving Gene Modules\n",
      "Heatmap showing the results of hierarchical clustering of the Simpson coefficient. Gene modules identified using the dynamicTreeCut method are indicated by colored bars along the x- and y-axes. Four modules that were significantly enriched for specific biological processes are colored red, blue, green, and purple. Modules that were tested, but not significantly enriched are shown in grey.\n",
      "Considering all genes, we constructed a network of genes with edges corresponding to gene pairs that showed highly correlated change in intron content across the phylogeny (partial Pearson correlation > 0.7; see “Materials and Methods” section for details), and then calculated the Simpson coefficient for each pair of genes in the resulting network. Hierarchical clustering of this matrix identified 14 distinct modules (fig. 4). To assess the sensitivity of this method to the choice of correlation threshold, we repeated this analysis at various thresholds ranging from 0.1 to 0.9. We found that this did not significantly alter the structure or content of these gene modules (supplementary fig. S2, Supplementary Material online), indicating that this analysis is robust to choice of threshold.  \n",
      "GO Biological Processes Most Significantly Enriched in Gene Coevolution Modules.\n",
      "Module number\tGO term\tDescription\tSize of module\tExpected count\tCount\tSize of term\tLog odds ratio\tFDR\n",
      "1\tGO:0048513\tOrgan development\t2355\t324.8\t417\t2601\t0.39\t3.3 × 10−5\n",
      "2\tGO:0048856\tAnatomical structure development\t661\t147.4\t205\t4374\t0.48\t0.001\n",
      "3\tGO:0006396\tRNA processing\t264\t9.9\t27\t762\t1.09\t0.035\n",
      "4\tGO:0032879\tRegulation of localization\t251\t20.7\t46\t1721\t0.95\t0.003\n",
      "\n",
      "Four of these modules were functionally enriched for biological processes (FDR < 0.05; see supplementary table S3, Supplementary Material online, for a list of genes in each of these coevolution modules). The most significantly enriched biological process for each of these modules is shown in table 3 (for a complete list of enriched processes, see supplementary table S4, Supplementary Material online). Module 1 was most significantly enriched for organ development. Interestingly, both modules 1 and 2 were enriched for biological processes related to development, in particular brain development (supplementary table S4, Supplementary Material online), suggesting that coevolution of intron length may be particularly important in this process. \n",
      "Next, we examined the patterns of change in intron content across the phylogeny for each of the four coevolving modules that were enriched for biological processes (supplementary fig. S3, Supplementary Material online; see “Materials and Methods” section for details). We were interested to determine whether the coordinated change in intron length observed in these enriched modules reflected a trend towards longer or shorter intron lengths in these gene sets along specific lineages. In general, increasing intron length was observed more frequently than decreases in intron length, consistent with what has been reported previously. Modules 1 and 3 showed some similarity in the rate of intron change across branches in the primate clade (supplementary fig. S3A and C, Supplementary Material online). In modules 2 and 4, much of the signal appeared to arise from rapid increase in intron length in the enriched gene sets on individual branches (supplementary fig. S3B and D, Supplementary Material online).\n",
      "In the case of Pearson correlation, coordinated change in intron content along a single branch among members of a gene set may be sufficient to generate a signal of coevolution. In contrast, the rank-based Spearman correlation would require coordinated changes on multiple branches. To identify signals of coevolution likely to result from coordinated changes in multiple branches, we repeated the analysis using Spearman correlation (see “Materials and Methods” section for details). With a lower threshold in this case (ρ > 0.5), we were able to construct 16 modules from this data, four of which were functionally enriched (FDR < 0.05; supplementary fig. S4, Supplementary Material online; see supplementary table S5, Supplementary Material online, for a list of genes in each enriched module). These modules were enriched for developmental processes, including central nervous system development (modules S2 and S3; supplementary table S6, Supplementary Material online), consistent with our results from analysis of the partial Pearson correlations. Interestingly, we also found that these modules were enriched for several metabolic processes, as well as several processes related to the cell cycle (module S1; supplementary table S6, Supplementary Material online). There was also a significant overlap between these modules and modules identified using Pearson correlation (supplementary table S7, Supplementary Material online). Although some of the Pearson modules overlapped significantly with more than one Spearman module, the same themes emerged. In both cases, the enriched coevolving modules were enriched for developmental and cell cycle processes. Clear modules were only visible at lower thresholds than for the Pearson correlation, although traces of these modules persisted at higher thresholds (supplementary fig. S5, Supplementary Material online).\n",
      "Effects of Species Choice\n",
      "The nine species in figure 1 were selected from the 15 species for which alignments were available as a tradeoff between maximizing the number of species and maximizing the number of genes for which an ortholog was found in all species. We also considered a second tree (supplementary fig. S6, Supplementary Material online), consisting of ten species and including a slightly reduced number (8719) of genes. Results obtained using this set of species was similar to those obtained on the original data set. Both protein complexes and coexpressed genes were significantly more likely to show evidence of intron length coevolution compared to sets of random genes with similarly sized introns (P = 0.0001 and P < 0.0001 for the CORUM protein complexes and GTEx coexpressed genes, respectively), consistent with our previous findings from the nine species tree. We were able to identify 13 coevolving modules (supplementary fig. S7, Supplementary Material online), one of which was significantly enriched for developmental processes, though not, specifically, for brain development (supplementary tables S8 and S9, Supplementary Material online). We also tested the effects of the human-centric approach we adopted to defining gene models (see “Materials and Methods” section) and instead defined introns according to gene models annotated in cow. Evidence for intron length coevolution remained significant for both the protein complexes and GTEx coexpression clusters (P = 0.005 and P < 0.0001, respectively).\n",
      "Coevolution of Intergenic Lengths\n",
      "Although we hypothesized that genes that require precise coordination in the timing of their expression may show evidence of coevolution in intron content, this does not imply that selection acting on transcription time is the only possible cause of coevolution in intron content. For example, similarities in the genomic context of functionally related genes may result in similar patterns of insertion and deletion events across lineages that could result in correlated evolution. Furthermore, introns, particularly first introns, contain regulatory elements, and functionally related genes may have similar regulatory complexity. Changes in the selection pressures acting on functionally related sets of genes along distinct lineages could allow them to become more or less permissive to loss or gains of regulatory sites through deletions or insertions, potentially giving rise to correlated change in intron content across the phylogeny. As a test of these alternative and other similar potential sources of intron content coevolution, we redid the analysis but this time treating the length of the intergenic region as the property of interest. Genes belonging to the same protein complex did not show evidence of coevolution of the length of intergenic regions (P = 0.72). Coexpressed genes showed marginally significant evidence (P = 0.07) of coevolution of the lengths of intergenic regions, but the extent of coevolution for intergenic lengths was significantly less than for intron lengths (e.g., partial Pearson correlation coefficients of genes belonging to the same GTEx coexpression clusters were significantly higher in the case of introns than intergenic regions; P = 7.4 × 10−8). However, coevolution modules similar to the modules shown in figure 4 were also observed for intergenic lengths and these were also significantly enriched for genes associated with developmental processes, but not for cell cycle processes (supplementary fig. S8 and tables S10 and S11, Supplementary Material online). Therefore, we conclude that there may be alternative explanations for coevolution of intron lengths of members of some of these modules; however, the coevolution of intron content only, in the case of members of protein complexes and the far stronger coevolution of intron content than of intergenic content for coexpressed gene clusters is consistent with a role for transcription time in intron length coevolution.\n",
      "Discussion\n",
      "In mammals, introns comprise up to 95% of the length of primary protein coding transcripts, raising questions as to the evolution and functional consequences of such large introns. While some theories have argued that the expansion of introns in eukaryotes is largely a semi neutral process resulting from the failure to prevent insertions that have only a slightly deleterious effect on fitness in organisms with small effective population sizes, others have focused more on the functional roles of introns, particularly in gene regulation. The genome design theory suggests that intron length reflects gene function. Others have argued that the shorter introns found in highly expressed genes are due to selection for transcriptional efficiency \n",
      "In this study, we carried out an investigation into intron length coevolution in mammals. We have shown that evolutionary changes in intron content are positively correlated among several sets of coregulated genes, including multi-protein complexes and coexpressed genes. Transcriptional coregulation of protein complex subunits is conserved over evolutionary time. Our results show that members of the same protein complex are significantly more likely to show coevolution of intron length compared to sets of randomly sampled genes with similar intron content. Coordinated expression of the components of protein complexes is important for the proper formation of the complex. The time that it takes to transcribe genes that encode members of protein complexes places a constraint on the evolution of the intron content of coregulated genes, particularly for genes such as those involved in development or formation of certain protein complexes that are sensitive to small changes in expression timing.\n",
      "Coevolution in intron content was more apparent for some protein complexes than for others. For example, the NCOR1 complex (table 2), a complex that is involved in chromatin modification. Interestingly, four of the top five complexes, including NCOR1, with the strongest evidence of intron length coevolution are annotated in CORUM as being involved in the cell cycle, a time sensitive process. These are NCOR1, CENP-A NCOR/CAD complex, DMAP1 associated complex, and the LINC complex. Strikingly, our analysis of gene coexpression clusters is in agreement with this result, as coexpression clusters that were enriched for cell cycle genes were also found to be significantly coevolving (clusters 1 and 5; supplementary table S12, Supplementary Material online). The cell cycle is a dynamic process that requires precise temporal regulation of protein complex components at various time points. The presence of cell cycle processes in the set of genes associated with high levels of intron length coevolution is consistent with our initial hypothesis that genes that require precise coordination in the timing of their expression may show signs of intron length coevolution. We propose that for a substantial proportion of such genes large changes in intron length may have deleterious consequences.\n",
      "Delays in gene expression that result from the transcription of introns are thought to be important in embryonic development. We have previously found that genes with the greatest conservation in intron length are highly enriched for processes relating to developmental patterning. Although, we found little evidence for conserved sequences in the introns of these genes, we could not exclude that the conservation of intron lengths was the result of conserved functional elements within these introns that limited the scope for changes in length through insertion and deletion. The present study takes our original observation further by showing that, not only is the intron content of genes involved in developmental processes conserved, but when changes are observed in the intron content they tend to be shared among coexpressed sets of genes involved in these processes.\n",
      "Our results suggest that constraints acting on transcription time are not likely to be the only source of intron content coevolution. Our initial hypothesis was that genes that are particularly sensitive to changes in expression timing (e.g., genes involved in developmental patterning) may be particularly sensitive to changes in intron content and that sets of genes with coordinated expression timing (e.g., members of tightly regulated protein complexes) may show evidence of coevolution of intron content. However, this does not preclude that other factors may also cause coevolution of intron content. For example, functionally related genes may sometimes occur in similar genomic contexts that may experience similar mutational patterns across different lineages, setting up correlation in intron length. As a means of exploring these alternatives, we compared the coevolution of intron content to the coevolution of the lengths of intergenic sequences. In the case of members of protein complexes and coexpressed gene pairs, the correlation in intron content was either not found or found to a much lesser extent when we considered intergenic length instead of intronic content, reducing the plausibility of shared genomic context or selective pressures acting to preserve gene regulation as the cause of correlated intron content of these genes. Some of the functionally enriched clusters with coevolving intron content could be reproduced with intergenic lengths rather than intron content, suggesting that these alternative explanations may apply in these cases. This included clusters that were enriched for brain development and consequently we cannot conclude that the coevolution of the intron content of genes involved in brain development reflects the effects of constraints acting on transcription timing.\n",
      "In summary, we hypothesized that the primary transcript length of sets of genes that require precise coordination in expression timing coevolves. In support of this hypothesis, we found that, in general, genes belonging to the same protein complex and pairs of coexpressed genes are significantly more likely to have coevolving intron length than randomly sampled gene pairs, matched for intron content. We found that some biological processes that are particularly time-sensitive, such as the cell cycle, are particularly enriched for genes with coevolving intron content. Our results suggest that for some sets of genes, intron length is functionally relevant and evolves under natural selection, representing a novel aspect of intron evolution and of the coevolutionary dynamics of coregulated genes.\n",
      "Materials and Methods\n",
      "Data\n",
      "Gene models for human protein coding genes were downloaded from Ensembl release 74 via BioMart. To identify insertion and deletion events (indels) in the introns of these genes, we downloaded whole genome alignments of 15 eutherian mammals from Ensembl Compara release 74. This data included ancestral sequences, reconstructed using the Enredo-Pecan-Ortheus (EPO) pipeline, at each branch of the phylogenetic tree, allowing indels to be placed on the branch on which they are inferred to have occurred. We selected nine species for which complete ortholog sets were available for a large number of genes. This selection was made on the basis of a tradeoff between maximizing the number of species included in the tree and maximizing the number of genes for which orthologs were available for each species in the tree. The latter consideration was important for our analyses, particularly in order to have good coverage of protein complexes. These species were Felis catus, Canis familiaris, Equus caballus, Ovis aries, Bos taurus, Pan troglodytes, Homo sapiens, Gorilla gorilla, and Pongo abelii. Only alignments containing all nine species were used for analysis, resulting in a dataset consisting of 9396 genes. The phylogenetic tree of these species is shown in figure 1.\n",
      "Partial Correlation of Intron Change\n",
      "A prediction of our hypothesis is that changes in intron length due to insertions and deletions should be correlated among sets of coevolving genes. To test this, we first calculated the intron content of each gene, defined as the sum of the lengths of all introns in the canonical transcript, using the positions of the intron/exon boundaries of the human gene as a reference. Canonical transcripts were obtained directly from Ensembl. For protein coding genes, these are defined as the transcript with the longest coding sequence without stop codons. Using the indels inferred on the branches of the phylogeny, we also calculated the intron content at each ancestral node. This was done by first calculating the intron content of the last common ancestor of all nine species by subtracting the length of all indels along the human lineage from the intron content of the human gene, and then applying the set of all indels leading to each node. We then calculated the proportional change in the intron content along each branch of the phylogeny. Partial Pearson correlation of this proportional change was calculated for each pair of genes, controlling for the median (over genes) change in intron content along each branch. This allowed us to control for differences in the overall rate of evolution along different branches.\n",
      "Identification of Coevolving Gene Modules\n",
      "From the partial correlations calculated above, we constructed a network with edges corresponding to pairs of highly correlated genes (partial Pearson coefficient > 0.7). This network comprised 9,286 genes with 1.2 × 106 edges. To extract modules of coevolving genes, we calculated the Simpson coefficient for each pair of genes a and b using the formula\n",
      "where A and B are the sets of genes that share an edge with genes a and b respectively. We then converted this similarity measure to a distance (1 – Simpson coefficient), and carried out average linkage hierarchical clustering of the resulting matrix in R (version 3.1.1). Modules were extracted from the dendrogram using the dynamicTreeCut method (version 1.62). To investigate the functions associated with these modules, we carried out Gene Ontology (GO) enrichment analysis using GOseq (version 1.16.2). We included intron content as a covariate, to exclude the possibility that the observed functional enrichment was a consequence of shared intron content of genes in certain functional classes (with genes with larger or smaller intron content having had similar proportional changes in intron content over the phylogeny). Each module was tested for enriched biological processes. Correction for multiple testing of the P-values produced by GOseq was done using the Benjamini–Hochberg method in R. To rank these processes in order of enrichment, we calculated the log odds ratio, and the 95% confidence interval of the log odds ratio for each enriched GO term using logistic regression with intron length included as a covariate.\n",
      "Visualization of Intron Length Change\n",
      "To show the changes in intron length across the phylogeny of the gene coevolution modules, we carried out a linear regression of the proportional change in intron content against the median change along each branch, for each gene in the module. We then calculated the residuals for these regressions and plotted the mean value obtained on the phylogenetic tree of the nine species. This was carried out for all functionally enriched gene modules and is represented in supplementary figure S3, Supplementary Material online, using colors to denote the value of the residuals on each branch.\n",
      "Test for Significantly Coevolving Gene Sets\n",
      "We developed a randomization-based test that can be used to determine if a given set of genes is significantly more likely to be correlated than expected, compared to random sets of genes with similarly sized introns. First, to account for the possibility that genes with similar intron content may evolve in a similar way, we divided the genes into five intron size classes (<5 Kb, 5–20 Kb, 20–50 Kb, 50–100 Kb and >100 Kb). For each gene within a set of interest, we randomly selected another gene at random from among all genes of the same intron size class. We then calculated the mean partial Pearson correlation of the random pairings. This was repeated 10,000 times and produced a P-value that represents the proportion of times the mean correlation of the random gene set was greater than or equal to that of the real data. A gene set was deemed to be significant if this P-value was less than 0.05. To control for the false discovery rate from these tests, we calculated π0, the expected proportion of tests that are consistent with the null hypothesis that coevolution in the gene set is no different to randomly selected genes. This was done using the q-value package in R (version 2.0.0).\n",
      "We applied this test to a set of human protein complexes obtained from the core set of the CORUM database of mammalian protein complexes, and to gene coexpression networks constructed using publicly available gene expression data obtained from the GTEx consortium.\n",
      "To investigate a possible influence of gene expression level on intron length coevolution, we modified the above test to also control for gene expression level in addition to intron size class. To do this, we first calculated the median and maximum expression value for each gene across all tissues of the GTEx gene expression dataset. Genes were then placed into one of five categories based on their median or maximum expression, and during the randomization test, genes were randomly resampled from both their relevant intron size class and gene expression class.\n",
      "Construction of Gene Coexpression Networks\n",
      "We downloaded gene expression data from the GTEx consortium (data release V4) via the GTEx web portal (www.gtexportal.org). This data consists of Reads Per Kilobase per Million mapped reads (RPKM) expression values calculated from RNA sequencing applied to 42 human tissues/cell-types, sampled from 190 individuals. To reduce noise, we only considered genes with RPKM > 0.5 in at least 10% of all samples (22,718 genes). Pearson correlations were then calculated for each gene pair. We constructed a gene coexpression network with edges corresponding to correlated gene pairs (|r| > 0.7). Gene clusters were then identified using the ClusterONE algorithm (version 1.0), which can identify densely connected and overlapping gene clusters within large biological networks. ClusterONE was run with a minimum cluster size of 10 and maximum matching coefficient between clusters of 0.5. Clusters above this threshold were merged to form single clusters. All other parameters were set at default values. For unweighted networks, ClusterONE determines the statistical significance of the clusters using a one-sided Mann–Whitney U test conducted on the number of inward versus outward edges. A low P-value is obtained when the number of inward edges is significantly greater than the number of outward edges. A cluster was deemed significant if this P-value was less than 0.05. To relate these clusters to gene function, we annotated each cluster using the most significantly enriched biological process identified using the GOstats package (version 2.30.0) in R. GO terms with an FDR < 0.05 were deemed to be significantly enriched in the coexpression clusters. To visualize the coexpression network, we carried out hierarchical clustering of the adjacency matrix for the network in R. The results of this are shown as a heatmap in supplementary figure S1, Supplementary Material online.\n",
      "Availability of Software\n",
      "The analysis presented here was conducted primarily using custom Perl scripts. In order to ensure reproducibility of our results, we have made these scripts available on GitHub at https://github.com/petebio/Intron_coevolution together with detailed usage instructions.\n",
      "Supplementary Material\n",
      "Supplementary figures S1–S8, tables S1–S12, and references are available at Molecular Biology and Evolution online (http://www.mbe.oxfordjournals.org/).\n",
      "References\n",
      "Tracking rates of transcription and splicing in vivo\n",
      "Transcript length mediates developmental timing of gene expression across Drosophila\n",
      "Chromatin remodelling is a major source of coexpression of linked genes in yeast\n",
      "Selection for short introns in highly expressed genes\n",
      "Evidence for gene length as a determinant of gene coexpression in protein complexes\n",
      "The function of introns\n",
      "Ensembl 2015\n",
      "Human housekeeping genes are compact\n",
      "Using GOstats to test gene lists for GO term association\n",
      "The architecture of pre-mRNAs affects mechanisms of splice-site pairing\n",
      "Coevolution of gene expression among interacting proteins\n",
      "Changes in exon–intron structure during vertebrate evolution affect the splicing pattern of exons\n",
      "Ensembl 2009\n",
      "Relating whole-genome expression data with protein-protein interactions\n",
      "Rapidly regulated genes are intron poor\n",
      "\n",
      "Dynamic complex formation during the yeast cell cycle\n",
      "The genotype-tissue expression (GTEx) project\n",
      "Intron evolution as a population-genetic process\n",
      "The origins of eukaryotic gene structure\n",
      "The evolution of controlled multitasked gene networks: the role of introns and other noncoding RNAs in the development of complex organisms\n",
      "Detecting overlapping protein complexes in protein-protein interaction networks\n",
      "Dosage sensitivity and the evolution of gene families in yeast\n",
      "A conserved mammalian protein interaction network\n",
      "CORUM: the comprehensive resource of mammalian protein complexes\n",
      "Gametophytic selection in Arabidopsis thaliana supports the selective model of intron length reduction\n",
      "Evidence for intron length conservation in a set of mammalian genes associated with embryonic development\n",
      "Progression of the cell cycle through mitosis leads to abortion of nascent transcripts\n",
      "Statistical significance for genomewide studies\n",
      "\n",
      "Intron delays and transcriptional timing during development\n",
      "Intronic delay is essential for oscillatory expression in the segmentation clock\n",
      "The human dystrophin gene requires 16 hours to be transcribed and is cotranscriptionally spliced\n",
      "A novel nuclear receptor corepressor complex, N-CoR, contains components of the mammalian SWI/SNF complex and the corepressor KAP-1\n",
      "The signature of selection mediated by expression on human genes\n",
      "Rate of elongation by RNA polymerase II is associated with specific gene features and epigenetic modifications\n",
      "“Genome design” model: Evidence from conserved intronic sequence in human–mouse comparison\n",
      "The transcriptional regulation of protein complexes; a cross-species perspective\n",
      "Gene ontology analysis for RNA-seq: accounting for selection bias\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "('10.1093/molbev/msw151',\n",
       " '5026260',\n",
       " 1,\n",
       " 0,\n",
       " 0,\n",
       " 0,\n",
       " 1,\n",
       " 0,\n",
       " 0,\n",
       " 0,\n",
       " 0,\n",
       " 0,\n",
       " 0,\n",
       " 0,\n",
       " 0,\n",
       " 0,\n",
       " 0,\n",
       " 0,\n",
       " 0,\n",
       " 0,\n",
       " 0,\n",
       " 0,\n",
       " 0,\n",
       " 0,\n",
       " 0,\n",
       " 0)"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "return_useful_data(r.json())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "intron length coevolution across mammalian genomes\n",
      "although they do not contribute directly to the proteome, introns frequently contain regulatory elements and can extend the protein coding potential of the genome through alternative splicing. for some genes, the contribution of introns to the time required for transcription can also be functionally significant. we have previously shown that intron length in genes associated with developmental patterning is often highly conserved. in general, sets of genes that require precise coordination in the timing of their expression may be sensitive to changes in transcript length. a prediction of this hypothesis is that evolutionary changes in intron length, when they occur, may be correlated between sets of coordinately expressed genes. to test this hypothesis, we analyzed intron length coevolution in alignments from nine eutherian mammals. overall, genes that belong to the same protein complex or that are coexpressed were significantly more likely to show evidence of intron length coevolution than matched, randomly sampled genes. individually, protein complexes involved in the cell cycle showed the strongest evidence of coevolution of intron lengths and clusters of coexpressed genes enriched for cell cycle genes also showed significant evidence of intron length coevolution. our results reveal a novel aspect of gene coevolution and provide a means to identify genes, protein complexes and biological processes that may be particularly sensitive to changes in transcriptional dynamics.\n",
      "introduction\n",
      "the origin, prevalence, and function of introns in eukaryotic genomes have long been a subject of great interest in molecular evolution. although typically they do not contribute directly to the protein coding sequences of genes, introns have been shown to play several important functional roles. introns frequently contain regulatory elements and allow for the process of alternative splicing, expanding the protein coding capacity of the genome, and giving rise to additional modes of gene regulation. in mammals, introns comprise up to 95% of transcribed primary protein-coding sequences, imposing a significant cost in terms of the energy required for transcription and in terms of the potential for erroneous transcripts resulting from mis-splicing, particularly for genes with longer introns. the time taken to transcribe introns can be significant. estimates of the rate of transcription in eukaryotes range from 1 to 4 kb per minute with a median rate of transcription estimated from over 2,000 human genes at 1.5 kb per minute. as an extreme example of the time taken to transcribe long introns, the human dystrophin gene, which is 2.3 mb in length, 99% of which is intronic, can take up to 16 h to transcribe, corresponding to a rate of 2.4 kb per minute. in contrast, genes that are highly expressed tend to have short introns, and genes that require rapid induction in response to stress are intron poor.\n",
      "the evolution and functional significance of large introns in mammals has been much debated.) proposed that intron expansion is a largely a semi-neutral process, resulting from insufficient purifying selection to remove insertions in organisms with small effective population sizes. the genome design model asserts that intron length reflects gene function. this model proposes that housekeeping genes tend to have short introns because they require very little regulation. conversely, genes that require more complex modes of gene regulation, such as tissue-specific genes are longer due, in part, to the presence of intronic regulatory elements. in opposition to this view, a number of authors have argued that the short introns in highly and ubiquitously expressed genes, such as housekeeping genes, are the result of selection for efficiency, resulting in reduced energetic costs of transcription.\n",
      "for some genes, the transcriptional delay resulting from the time taken to transcribe introns is functionally significant. this intron delay is particularly important in embryonic development, where precise temporal control of gene expression is required to ensure proper development of the embryo. during embryonic development, cells undergo rapid divisions creating a temporal constraint on transcription. in drosophila, introns in genes expressed by the embryo during this period are short, allowing for rapid transcription between cell cycles. intron length can also contribute to more complex forms of gene regulation. delayed expression, resulting in part from the transcription of introns, combined with negative feedback loops, can create oscillating patterns of gene expression. the hes7 gene, which is involved in somite segmentation during development, represents an example of this. removal of introns from hes7 in mouse resulted in hes7 being expressed 19 min early and the abolition of expression oscillations, resulting in severe developmental defects. the intron lengths, though in general not the intron sequences, of hes7 as well as several other genes involved in developmental patterning are highly conserved, suggesting that these genes are under purifying selection to maintain intron length. intron length also appears to be relevant in protein complex formation as genes that occur within the same complex usually have very similar length.\n",
      "here, we investigated the extent of intron length coevolution across several mammalian genomes using whole genome alignments together with reconstructed ancestral sequences. we hypothesized that sets of genes that require precise coordination in the timing of their expression may be sensitive to changes in intron length and that natural selection can act to maintain the relative lengths of such genes, resulting in correlated changes in intron content. we found that, overall, coexpressed genes and genes belonging to the same protein complex were significantly more likely to have coevolving intron lengths than randomly sampled genes, matched for total intron content. we also found a number of gene modules that showed particularly strongly correlated intron length evolution that may reflect selection acting on relative transcript lengths. we propose that transcript length coevolution can reveal functional constraints on intron lengths and that intron length coevolution may suggest biological processes that require precise temporal regulation of gene expression.\n",
      "results\n",
      "phylogenetic tree of the nine mammals included in this study.\n",
      "we hypothesized that sets of genes that require precise coordination in the timing of their expression may be particularly sensitive to evolutionary changes in intron content, and that changes in intron content due to insertions and deletions would be correlated among such sets of genes. to identify sets of genes where this may be the case, we compared changes in intron content across the species phylogeny for 9,396 genes from nine mammalian genomes. the phylogenetic tree representing the nine species used in this analysis is shown in figure 1.  \n",
      "protein complexes show significant evidence of intron length coevolution\n",
      "multi-protein complexes are often precisely coregulated in order to ensure proper assembly and function and thus we predicted that the intron content of members of some protein complexes may display evidence of coevolution. subunits of protein complexes tend to be coexpressed, a pattern which is conserved across a wide range of species. studies in yeast have shown that gene expression levels among sets of interacting proteins coevolve, suggesting that the mechanisms that control this coregulation may also coevolve. it has also been suggested that gene length contributes to the coregulation of protein complexes, as genes encoding subunits of multi-protein complexes tend to have similar lengths. furthermore, protein–protein interactions are known to be largely conserved across mammals.\n",
      "we sought to determine whether intron length coevolution could be detected among subunits of known protein complexes, potentially representing a novel aspect of the coevolutionary dynamics of members of multi-protein complexes. to do this, we downloaded all human protein complexes from the core set of the corum database of mammalian protein complexes. we then carried out a randomization test to determine if members of the same protein complex show significantly greater intron length correlation compared to sets of randomly sampled genes with similar intron content (see “materials and methods” section). for this test, we only considered complexes for which at least three genes were present in our original 9,396 gene dataset, resulting in 440 complexes that could be tested.\n",
      "distribution of the mean partial pearson correlation values from 10,000 randomizations of (a) corum protein complexes and (b) gtex coexpression clusters. the mean value of the real data is shown as a red dot.\n",
      "distribution of p-values obtained from 10,000 randomizations of (a) corum protein complexes and (b) gtex coexpression clusters. π0, the expected proportion of tests consistent with the null hypothesis is shown as a red dotted line. the uniform distribution, expected under the null hypothesis is shown in blue.\n",
      "top 10 significantly coevolving corum protein complexes.\n",
      "corum complex id\tcomplex description\tno. of genes analyzed\tmean pearson coefficient\tp-value\tq-value\n",
      "1413\tncor1 complex\t8\t0.43\t<0.0001\t0.04\n",
      "2446\titga9-itgb1-figf complex\t3\t0.77\t0.001\t0.12\n",
      "5589\tlinc complex, s-phase\t6\t0.42\t0.001\t0.16\n",
      "799\tdmap1-associated complex\t8\t0.32\t0.002\t0.16\n",
      "1179\tcenp-a nac-cad complex\t6\t0.42\t0.002\t0.16\n",
      "657\tretromer complex (snx1, snx2, vps35, vps29, vps26a)\t3\t0.66\t0.004\t0.23\n",
      "570\tp300-cbp-p270-swi/snf complex\t6\t0.36\t0.005\t0.26\n",
      "781\turi complex (unconventional prefoldin rpb5 interactor)\t6\t0.35\t0.005\t0.26\n",
      "351\tspliceosome\t69\t0.04\t0.007\t0.28\n",
      "1332\tlarge drosha complex\t10\t0.19\t0.007\t0.28\n",
      "\n",
      "first, we asked if intron length coevolution could be detected by considering all interacting proteins from across all protein complexes at once. this was done using a randomization test described in the “materials and methods” section. we found evidence of intron length coevolution of genes belonging to the same protein complex using this approach (p < 0.0001; fig. 2a). we next asked which protein complexes showed the greatest evidence of coevolution. we repeated the randomization test, this time considering each protein complex individually. of the 440 complexes tested, the members of 44 clusters showed evidence of intron length coevolution (uncorrected p < 0.05), the top 10 of which are shown in table 1. given that there was overall evidence of coevolution of members of the same protein complex and an excess of nominally significant complexes over the null expectation we applied the q-value method to investigate properties of the collection of statistical tests applied to protein complexes. the proportion, π0, of tests estimated to conform to the null hypothesis was 0.9, indicating that 10% of tests were not consistent with the null hypothesis (fig. 3a). just one complex, annotated in corum as the ncor1 complex, had a q-value below 0.05 (q-value = 0.04). the full list of nominally significant complexes and their q-values is provided in supplementary table s1, supplementary material online.   \n",
      "intron length coevolution in coexpressed genes\n",
      "significantly coevolving gtex coexpression clusters.\n",
      "id\tmost significantly enriched go biological process\tno. of genes\tmean correlation\tp-value\tq-value\n",
      "2\tsynaptic transmission\t429\t0.06\t<0.0001\t<0.0001\n",
      "9\tnucleic acid metabolic process\t83\t0.09\t<0.0001\t<0.0001\n",
      "5\tcell cycle\t171\t0.04\t0.0003\t0.002\n",
      "4\tsynaptic transmission\t213\t0.05\t0.004\t0.004\n",
      "8\tncrna metabolic process\t122\t0.03\t0.01\t0.031\n",
      "15\tribosome biogenesis\t85\t0.04\t0.015\t0.037\n",
      "30\tcardiac muscle tissue development\t30\t0.1\t0.02\t0.037\n",
      "14\textracellular muscle organization\t102\t0.05\t0.021\t0.037\n",
      "21\textracellular muscle organization\t49\t0.08\t0.021\t0.037\n",
      "19\tcellular respiration\t64\t0.04\t0.024\t0.038\n",
      "10\tmuscle system process\t110\t0.03\t0.03\t0.038\n",
      "13\tcarboxylic acid metabolic process\t63\t0.04\t0.033\t0.038\n",
      "11\ttranslation elongation\t34\t0.05\t0.033\t0.038\n",
      "34\tpeptide hormone processing\t19\t0.09\t0.038\t0.038\n",
      "46\tcarboxylic acid transport\t7\t0.2\t0.038\t0.038\n",
      "1\tspermatogenesis\t511\t0.02\t0.039\t0.038\n",
      "44\tsurfactant homeostasis\t7\t0.19\t0.049\t0.045\n",
      "\n",
      "we next asked if intron length coevolution is a general feature of genes that are coregulated. sets of genes that are coregulated include coexpressed genes. indeed, interacting proteins are often coexpressed. we constructed a gene coexpression network containing 60 clusters using publicly available gene expression data from the genotype-tissue expression (gtex) consortium (see supplementary table s2, supplementary material online, for a list of genes included in each coexpression cluster). although we allowed for overlapping clusters in our method (see “materials and methods” section), we found that the clusters identified were largely nonoverlapping and independent (supplementary fig. s1, supplementary material online). for this analysis, we only considered the 50 clusters for which at least three genes were present in our original 9,396 gene dataset. to first test if intron length coevolution could be detected across the gene coexpression network as a whole, we applied the randomization test used for protein complexes to the 50 clusters. similarly to protein complexes, we found that coexpressed genes were significantly more likely to be coevolving compared to sets of random genes (p < 0.0001; fig. 2b). we next applied this test to each cluster individually to determine which clusters were most enriched for intron length coevolution. we found 17 clusters with significant evidence of coevolution (p < 0.05; uncorrected p-values; table 2). π0 was estimated to be 0.3, suggesting that 70% of tests were not consistent with the null hypothesis (fig. 3b). all 17 tests remained significant after correction for multiple testing using the q-value method (q-value < 0.05; table 2). these clusters are highlighted in the gene coexpression network in supplementary figure s1, supplementary material online. \n",
      "it is possible that the intron content coevolution we observed for coexpressed genes or members of protein complexes reflects similar patterns of intron length change in different expression classes (e.g., highly or broadly expressed genes). to explore this possibility, we repeated the randomization tests, matching on median, or maximum gene expression across gtex tissues in addition to intron length (see “materials and methods” section). the randomization test remained significant for both the protein complexes (p = 0.0016 and p = 0.0001, matching on median or maximum expression, respectively) and gene coexpression clusters (p = 0.0002 and p < 0.0001). thus, we did not find evidence that expression class was the driver of intron content coevolution of members of protein complexes or coexpression clusters.\n",
      "we also considered the possibility that correlated change in intron length of coexpressed genes and members of protein complexes results from genomic colocation. it has previously been shown that genes in close proximity to each other tend to have similar patterns of gene expression and similarly sized introns. to investigate the possible impact of this on our analysis, we repeated our randomization test but this time excluding all gene pairs within a genomic distance of 1 mb. we found that this did not significantly alter our results, as both protein complexes and coexpressed genes were still found to be significant after the exclusion of these gene pairs (p < 0.0001 for both the protein complex and coexpressed genes). therefore, we conclude that chromosomal location or linkage does not significantly contribute to the patterns of intron length coevolution we observed.\n",
      "functional enrichment in coevolving gene modules\n",
      "heatmap showing the results of hierarchical clustering of the simpson coefficient. gene modules identified using the dynamictreecut method are indicated by colored bars along the x- and y-axes. four modules that were significantly enriched for specific biological processes are colored red, blue, green, and purple. modules that were tested, but not significantly enriched are shown in grey.\n",
      "considering all genes, we constructed a network of genes with edges corresponding to gene pairs that showed highly correlated change in intron content across the phylogeny (partial pearson correlation > 0.7; see “materials and methods” section for details), and then calculated the simpson coefficient for each pair of genes in the resulting network. hierarchical clustering of this matrix identified 14 distinct modules (fig. 4). to assess the sensitivity of this method to the choice of correlation threshold, we repeated this analysis at various thresholds ranging from 0.1 to 0.9. we found that this did not significantly alter the structure or content of these gene modules (supplementary fig. s2, supplementary material online), indicating that this analysis is robust to choice of threshold.  \n",
      "go biological processes most significantly enriched in gene coevolution modules.\n",
      "module number\tgo term\tdescription\tsize of module\texpected count\tcount\tsize of term\tlog odds ratio\tfdr\n",
      "1\tgo:0048513\torgan development\t2355\t324.8\t417\t2601\t0.39\t3.3 × 10−5\n",
      "2\tgo:0048856\tanatomical structure development\t661\t147.4\t205\t4374\t0.48\t0.001\n",
      "3\tgo:0006396\trna processing\t264\t9.9\t27\t762\t1.09\t0.035\n",
      "4\tgo:0032879\tregulation of localization\t251\t20.7\t46\t1721\t0.95\t0.003\n",
      "\n",
      "four of these modules were functionally enriched for biological processes (fdr < 0.05; see supplementary table s3, supplementary material online, for a list of genes in each of these coevolution modules). the most significantly enriched biological process for each of these modules is shown in table 3 (for a complete list of enriched processes, see supplementary table s4, supplementary material online). module 1 was most significantly enriched for organ development. interestingly, both modules 1 and 2 were enriched for biological processes related to development, in particular brain development (supplementary table s4, supplementary material online), suggesting that coevolution of intron length may be particularly important in this process. \n",
      "next, we examined the patterns of change in intron content across the phylogeny for each of the four coevolving modules that were enriched for biological processes (supplementary fig. s3, supplementary material online; see “materials and methods” section for details). we were interested to determine whether the coordinated change in intron length observed in these enriched modules reflected a trend towards longer or shorter intron lengths in these gene sets along specific lineages. in general, increasing intron length was observed more frequently than decreases in intron length, consistent with what has been reported previously. modules 1 and 3 showed some similarity in the rate of intron change across branches in the primate clade (supplementary fig. s3a and c, supplementary material online). in modules 2 and 4, much of the signal appeared to arise from rapid increase in intron length in the enriched gene sets on individual branches (supplementary fig. s3b and d, supplementary material online).\n",
      "in the case of pearson correlation, coordinated change in intron content along a single branch among members of a gene set may be sufficient to generate a signal of coevolution. in contrast, the rank-based spearman correlation would require coordinated changes on multiple branches. to identify signals of coevolution likely to result from coordinated changes in multiple branches, we repeated the analysis using spearman correlation (see “materials and methods” section for details). with a lower threshold in this case (ρ > 0.5), we were able to construct 16 modules from this data, four of which were functionally enriched (fdr < 0.05; supplementary fig. s4, supplementary material online; see supplementary table s5, supplementary material online, for a list of genes in each enriched module). these modules were enriched for developmental processes, including central nervous system development (modules s2 and s3; supplementary table s6, supplementary material online), consistent with our results from analysis of the partial pearson correlations. interestingly, we also found that these modules were enriched for several metabolic processes, as well as several processes related to the cell cycle (module s1; supplementary table s6, supplementary material online). there was also a significant overlap between these modules and modules identified using pearson correlation (supplementary table s7, supplementary material online). although some of the pearson modules overlapped significantly with more than one spearman module, the same themes emerged. in both cases, the enriched coevolving modules were enriched for developmental and cell cycle processes. clear modules were only visible at lower thresholds than for the pearson correlation, although traces of these modules persisted at higher thresholds (supplementary fig. s5, supplementary material online).\n",
      "effects of species choice\n",
      "the nine species in figure 1 were selected from the 15 species for which alignments were available as a tradeoff between maximizing the number of species and maximizing the number of genes for which an ortholog was found in all species. we also considered a second tree (supplementary fig. s6, supplementary material online), consisting of ten species and including a slightly reduced number (8719) of genes. results obtained using this set of species was similar to those obtained on the original data set. both protein complexes and coexpressed genes were significantly more likely to show evidence of intron length coevolution compared to sets of random genes with similarly sized introns (p = 0.0001 and p < 0.0001 for the corum protein complexes and gtex coexpressed genes, respectively), consistent with our previous findings from the nine species tree. we were able to identify 13 coevolving modules (supplementary fig. s7, supplementary material online), one of which was significantly enriched for developmental processes, though not, specifically, for brain development (supplementary tables s8 and s9, supplementary material online). we also tested the effects of the human-centric approach we adopted to defining gene models (see “materials and methods” section) and instead defined introns according to gene models annotated in cow. evidence for intron length coevolution remained significant for both the protein complexes and gtex coexpression clusters (p = 0.005 and p < 0.0001, respectively).\n",
      "coevolution of intergenic lengths\n",
      "although we hypothesized that genes that require precise coordination in the timing of their expression may show evidence of coevolution in intron content, this does not imply that selection acting on transcription time is the only possible cause of coevolution in intron content. for example, similarities in the genomic context of functionally related genes may result in similar patterns of insertion and deletion events across lineages that could result in correlated evolution. furthermore, introns, particularly first introns, contain regulatory elements, and functionally related genes may have similar regulatory complexity. changes in the selection pressures acting on functionally related sets of genes along distinct lineages could allow them to become more or less permissive to loss or gains of regulatory sites through deletions or insertions, potentially giving rise to correlated change in intron content across the phylogeny. as a test of these alternative and other similar potential sources of intron content coevolution, we redid the analysis but this time treating the length of the intergenic region as the property of interest. genes belonging to the same protein complex did not show evidence of coevolution of the length of intergenic regions (p = 0.72). coexpressed genes showed marginally significant evidence (p = 0.07) of coevolution of the lengths of intergenic regions, but the extent of coevolution for intergenic lengths was significantly less than for intron lengths (e.g., partial pearson correlation coefficients of genes belonging to the same gtex coexpression clusters were significantly higher in the case of introns than intergenic regions; p = 7.4 × 10−8). however, coevolution modules similar to the modules shown in figure 4 were also observed for intergenic lengths and these were also significantly enriched for genes associated with developmental processes, but not for cell cycle processes (supplementary fig. s8 and tables s10 and s11, supplementary material online). therefore, we conclude that there may be alternative explanations for coevolution of intron lengths of members of some of these modules; however, the coevolution of intron content only, in the case of members of protein complexes and the far stronger coevolution of intron content than of intergenic content for coexpressed gene clusters is consistent with a role for transcription time in intron length coevolution.\n",
      "discussion\n",
      "in mammals, introns comprise up to 95% of the length of primary protein coding transcripts, raising questions as to the evolution and functional consequences of such large introns. while some theories have argued that the expansion of introns in eukaryotes is largely a semi neutral process resulting from the failure to prevent insertions that have only a slightly deleterious effect on fitness in organisms with small effective population sizes, others have focused more on the functional roles of introns, particularly in gene regulation. the genome design theory suggests that intron length reflects gene function. others have argued that the shorter introns found in highly expressed genes are due to selection for transcriptional efficiency \n",
      "in this study, we carried out an investigation into intron length coevolution in mammals. we have shown that evolutionary changes in intron content are positively correlated among several sets of coregulated genes, including multi-protein complexes and coexpressed genes. transcriptional coregulation of protein complex subunits is conserved over evolutionary time. our results show that members of the same protein complex are significantly more likely to show coevolution of intron length compared to sets of randomly sampled genes with similar intron content. coordinated expression of the components of protein complexes is important for the proper formation of the complex. the time that it takes to transcribe genes that encode members of protein complexes places a constraint on the evolution of the intron content of coregulated genes, particularly for genes such as those involved in development or formation of certain protein complexes that are sensitive to small changes in expression timing.\n",
      "coevolution in intron content was more apparent for some protein complexes than for others. for example, the ncor1 complex (table 2), a complex that is involved in chromatin modification. interestingly, four of the top five complexes, including ncor1, with the strongest evidence of intron length coevolution are annotated in corum as being involved in the cell cycle, a time sensitive process. these are ncor1, cenp-a ncor/cad complex, dmap1 associated complex, and the linc complex. strikingly, our analysis of gene coexpression clusters is in agreement with this result, as coexpression clusters that were enriched for cell cycle genes were also found to be significantly coevolving (clusters 1 and 5; supplementary table s12, supplementary material online). the cell cycle is a dynamic process that requires precise temporal regulation of protein complex components at various time points. the presence of cell cycle processes in the set of genes associated with high levels of intron length coevolution is consistent with our initial hypothesis that genes that require precise coordination in the timing of their expression may show signs of intron length coevolution. we propose that for a substantial proportion of such genes large changes in intron length may have deleterious consequences.\n",
      "delays in gene expression that result from the transcription of introns are thought to be important in embryonic development. we have previously found that genes with the greatest conservation in intron length are highly enriched for processes relating to developmental patterning. although, we found little evidence for conserved sequences in the introns of these genes, we could not exclude that the conservation of intron lengths was the result of conserved functional elements within these introns that limited the scope for changes in length through insertion and deletion. the present study takes our original observation further by showing that, not only is the intron content of genes involved in developmental processes conserved, but when changes are observed in the intron content they tend to be shared among coexpressed sets of genes involved in these processes.\n",
      "our results suggest that constraints acting on transcription time are not likely to be the only source of intron content coevolution. our initial hypothesis was that genes that are particularly sensitive to changes in expression timing (e.g., genes involved in developmental patterning) may be particularly sensitive to changes in intron content and that sets of genes with coordinated expression timing (e.g., members of tightly regulated protein complexes) may show evidence of coevolution of intron content. however, this does not preclude that other factors may also cause coevolution of intron content. for example, functionally related genes may sometimes occur in similar genomic contexts that may experience similar mutational patterns across different lineages, setting up correlation in intron length. as a means of exploring these alternatives, we compared the coevolution of intron content to the coevolution of the lengths of intergenic sequences. in the case of members of protein complexes and coexpressed gene pairs, the correlation in intron content was either not found or found to a much lesser extent when we considered intergenic length instead of intronic content, reducing the plausibility of shared genomic context or selective pressures acting to preserve gene regulation as the cause of correlated intron content of these genes. some of the functionally enriched clusters with coevolving intron content could be reproduced with intergenic lengths rather than intron content, suggesting that these alternative explanations may apply in these cases. this included clusters that were enriched for brain development and consequently we cannot conclude that the coevolution of the intron content of genes involved in brain development reflects the effects of constraints acting on transcription timing.\n",
      "in summary, we hypothesized that the primary transcript length of sets of genes that require precise coordination in expression timing coevolves. in support of this hypothesis, we found that, in general, genes belonging to the same protein complex and pairs of coexpressed genes are significantly more likely to have coevolving intron length than randomly sampled gene pairs, matched for intron content. we found that some biological processes that are particularly time-sensitive, such as the cell cycle, are particularly enriched for genes with coevolving intron content. our results suggest that for some sets of genes, intron length is functionally relevant and evolves under natural selection, representing a novel aspect of intron evolution and of the coevolutionary dynamics of coregulated genes.\n",
      "materials and methods\n",
      "data\n",
      "gene models for human protein coding genes were downloaded from ensembl release 74 via biomart. to identify insertion and deletion events (indels) in the introns of these genes, we downloaded whole genome alignments of 15 eutherian mammals from ensembl compara release 74. this data included ancestral sequences, reconstructed using the enredo-pecan-ortheus (epo) pipeline, at each branch of the phylogenetic tree, allowing indels to be placed on the branch on which they are inferred to have occurred. we selected nine species for which complete ortholog sets were available for a large number of genes. this selection was made on the basis of a tradeoff between maximizing the number of species included in the tree and maximizing the number of genes for which orthologs were available for each species in the tree. the latter consideration was important for our analyses, particularly in order to have good coverage of protein complexes. these species were felis catus, canis familiaris, equus caballus, ovis aries, bos taurus, pan troglodytes, homo sapiens, gorilla gorilla, and pongo abelii. only alignments containing all nine species were used for analysis, resulting in a dataset consisting of 9396 genes. the phylogenetic tree of these species is shown in figure 1.\n",
      "partial correlation of intron change\n",
      "a prediction of our hypothesis is that changes in intron length due to insertions and deletions should be correlated among sets of coevolving genes. to test this, we first calculated the intron content of each gene, defined as the sum of the lengths of all introns in the canonical transcript, using the positions of the intron/exon boundaries of the human gene as a reference. canonical transcripts were obtained directly from ensembl. for protein coding genes, these are defined as the transcript with the longest coding sequence without stop codons. using the indels inferred on the branches of the phylogeny, we also calculated the intron content at each ancestral node. this was done by first calculating the intron content of the last common ancestor of all nine species by subtracting the length of all indels along the human lineage from the intron content of the human gene, and then applying the set of all indels leading to each node. we then calculated the proportional change in the intron content along each branch of the phylogeny. partial pearson correlation of this proportional change was calculated for each pair of genes, controlling for the median (over genes) change in intron content along each branch. this allowed us to control for differences in the overall rate of evolution along different branches.\n",
      "identification of coevolving gene modules\n",
      "from the partial correlations calculated above, we constructed a network with edges corresponding to pairs of highly correlated genes (partial pearson coefficient > 0.7). this network comprised 9,286 genes with 1.2 × 106 edges. to extract modules of coevolving genes, we calculated the simpson coefficient for each pair of genes a and b using the formula\n",
      "where a and b are the sets of genes that share an edge with genes a and b respectively. we then converted this similarity measure to a distance (1 – simpson coefficient), and carried out average linkage hierarchical clustering of the resulting matrix in r (version 3.1.1). modules were extracted from the dendrogram using the dynamictreecut method (version 1.62). to investigate the functions associated with these modules, we carried out gene ontology (go) enrichment analysis using goseq (version 1.16.2). we included intron content as a covariate, to exclude the possibility that the observed functional enrichment was a consequence of shared intron content of genes in certain functional classes (with genes with larger or smaller intron content having had similar proportional changes in intron content over the phylogeny). each module was tested for enriched biological processes. correction for multiple testing of the p-values produced by goseq was done using the benjamini–hochberg method in r. to rank these processes in order of enrichment, we calculated the log odds ratio, and the 95% confidence interval of the log odds ratio for each enriched go term using logistic regression with intron length included as a covariate.\n",
      "visualization of intron length change\n",
      "to show the changes in intron length across the phylogeny of the gene coevolution modules, we carried out a linear regression of the proportional change in intron content against the median change along each branch, for each gene in the module. we then calculated the residuals for these regressions and plotted the mean value obtained on the phylogenetic tree of the nine species. this was carried out for all functionally enriched gene modules and is represented in supplementary figure s3, supplementary material online, using colors to denote the value of the residuals on each branch.\n",
      "test for significantly coevolving gene sets\n",
      "we developed a randomization-based test that can be used to determine if a given set of genes is significantly more likely to be correlated than expected, compared to random sets of genes with similarly sized introns. first, to account for the possibility that genes with similar intron content may evolve in a similar way, we divided the genes into five intron size classes (<5 kb, 5–20 kb, 20–50 kb, 50–100 kb and >100 kb). for each gene within a set of interest, we randomly selected another gene at random from among all genes of the same intron size class. we then calculated the mean partial pearson correlation of the random pairings. this was repeated 10,000 times and produced a p-value that represents the proportion of times the mean correlation of the random gene set was greater than or equal to that of the real data. a gene set was deemed to be significant if this p-value was less than 0.05. to control for the false discovery rate from these tests, we calculated π0, the expected proportion of tests that are consistent with the null hypothesis that coevolution in the gene set is no different to randomly selected genes. this was done using the q-value package in r (version 2.0.0).\n",
      "we applied this test to a set of human protein complexes obtained from the core set of the corum database of mammalian protein complexes, and to gene coexpression networks constructed using publicly available gene expression data obtained from the gtex consortium.\n",
      "to investigate a possible influence of gene expression level on intron length coevolution, we modified the above test to also control for gene expression level in addition to intron size class. to do this, we first calculated the median and maximum expression value for each gene across all tissues of the gtex gene expression dataset. genes were then placed into one of five categories based on their median or maximum expression, and during the randomization test, genes were randomly resampled from both their relevant intron size class and gene expression class.\n",
      "construction of gene coexpression networks\n",
      "we downloaded gene expression data from the gtex consortium (data release v4) via the gtex web portal (www.gtexportal.org). this data consists of reads per kilobase per million mapped reads (rpkm) expression values calculated from rna sequencing applied to 42 human tissues/cell-types, sampled from 190 individuals. to reduce noise, we only considered genes with rpkm > 0.5 in at least 10% of all samples (22,718 genes). pearson correlations were then calculated for each gene pair. we constructed a gene coexpression network with edges corresponding to correlated gene pairs (|r| > 0.7). gene clusters were then identified using the clusterone algorithm (version 1.0), which can identify densely connected and overlapping gene clusters within large biological networks. clusterone was run with a minimum cluster size of 10 and maximum matching coefficient between clusters of 0.5. clusters above this threshold were merged to form single clusters. all other parameters were set at default values. for unweighted networks, clusterone determines the statistical significance of the clusters using a one-sided mann–whitney u test conducted on the number of inward versus outward edges. a low p-value is obtained when the number of inward edges is significantly greater than the number of outward edges. a cluster was deemed significant if this p-value was less than 0.05. to relate these clusters to gene function, we annotated each cluster using the most significantly enriched biological process identified using the gostats package (version 2.30.0) in r. go terms with an fdr < 0.05 were deemed to be significantly enriched in the coexpression clusters. to visualize the coexpression network, we carried out hierarchical clustering of the adjacency matrix for the network in r. the results of this are shown as a heatmap in supplementary figure s1, supplementary material online.\n",
      "availability of software\n",
      "the analysis presented here was conducted primarily using custom perl scripts. in order to ensure reproducibility of our results, we have made these scripts available on github at https://github.com/petebio/intron_coevolution together with detailed usage instructions.\n",
      "supplementary material\n",
      "supplementary figures s1–s8, tables s1–s12, and references are available at molecular biology and evolution online (http://www.mbe.oxfordjournals.org/).\n",
      "references\n",
      "tracking rates of transcription and splicing in vivo\n",
      "transcript length mediates developmental timing of gene expression across drosophila\n",
      "chromatin remodelling is a major source of coexpression of linked genes in yeast\n",
      "selection for short introns in highly expressed genes\n",
      "evidence for gene length as a determinant of gene coexpression in protein complexes\n",
      "the function of introns\n",
      "ensembl 2015\n",
      "human housekeeping genes are compact\n",
      "using gostats to test gene lists for go term association\n",
      "the architecture of pre-mrnas affects mechanisms of splice-site pairing\n",
      "coevolution of gene expression among interacting proteins\n",
      "changes in exon–intron structure during vertebrate evolution affect the splicing pattern of exons\n",
      "ensembl 2009\n",
      "relating whole-genome expression data with protein-protein interactions\n",
      "rapidly regulated genes are intron poor\n",
      "\n",
      "dynamic complex formation during the yeast cell cycle\n",
      "the genotype-tissue expression (gtex) project\n",
      "intron evolution as a population-genetic process\n",
      "the origins of eukaryotic gene structure\n",
      "the evolution of controlled multitasked gene networks: the role of introns and other noncoding rnas in the development of complex organisms\n",
      "detecting overlapping protein complexes in protein-protein interaction networks\n",
      "dosage sensitivity and the evolution of gene families in yeast\n",
      "a conserved mammalian protein interaction network\n",
      "corum: the comprehensive resource of mammalian protein complexes\n",
      "gametophytic selection in arabidopsis thaliana supports the selective model of intron length reduction\n",
      "evidence for intron length conservation in a set of mammalian genes associated with embryonic development\n",
      "progression of the cell cycle through mitosis leads to abortion of nascent transcripts\n",
      "statistical significance for genomewide studies\n",
      "\n",
      "intron delays and transcriptional timing during development\n",
      "intronic delay is essential for oscillatory expression in the segmentation clock\n",
      "the human dystrophin gene requires 16 hours to be transcribed and is cotranscriptionally spliced\n",
      "a novel nuclear receptor corepressor complex, n-cor, contains components of the mammalian swi/snf complex and the corepressor kap-1\n",
      "the signature of selection mediated by expression on human genes\n",
      "rate of elongation by rna polymerase ii is associated with specific gene features and epigenetic modifications\n",
      "“genome design” model: evidence from conserved intronic sequence in human–mouse comparison\n",
      "the transcriptional regulation of protein complexes; a cross-species perspective\n",
      "gene ontology analysis for rna-seq: accounting for selection bias\n"
     ]
    }
   ],
   "source": [
    "paper = r.json()\n",
    "for passage in paper['documents'][0]['passages']:\n",
    "    print(passage['text'].lower())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "ac_knowl",
   "language": "python",
   "name": "ac_knowl"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
