{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import urllib\n",
    "import requests\n",
    "import re\n",
    "import glob\n",
    "import json"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Get the pubmed papers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/riddleta/ac_knowl/ac_knowl/lib/python3.7/site-packages/ipykernel_launcher.py:2: FutureWarning: read_table is deprecated, use read_csv instead.\n",
      "  \n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(598386, 4)"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#bioc-pmc files\n",
    "files_to_get = pd.read_table('/home/riddleta/ac_knowl/data/filelist.txt', sep='\\s+')\n",
    "files_to_get.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1479862, 5)"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "oa_comm_use_files = pd.read_csv('/home/riddleta/ac_knowl/data/oa_comm_use_file_list.txt', \n",
    "                                sep='\\\\t', engine='python', header=None, skiprows=1)\n",
    "oa_comm_use_files.columns = ['File', 'ref', 'PMCID', 'PMID', 'license']\n",
    "oa_comm_use_files.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/riddleta/ac_knowl/ac_knowl/lib/python3.7/site-packages/ipykernel_launcher.py:2: FutureWarning: read_table is deprecated, use read_csv instead.\n",
      "  \n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(2408932, 5)"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "oa_files = pd.read_table('/home/riddleta/ac_knowl/data/oa_file_list.txt', \n",
    "                         sep='\\\\t', engine='python', header=None, skiprows=1)\n",
    "oa_files.columns = ['File', 'ref', 'PMCID', 'PMID', 'license']\n",
    "oa_files.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "unique_ids = pd.concat([oa_files['PMCID'], oa_comm_use_files['PMCID'], files_to_get['PMCID']]).unique()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Main function to return data from fulltext search"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "def return_useful_data(r):\n",
    "    git_hits = 0\n",
    "    osf_hits = 0\n",
    "    nda_hits = 0\n",
    "    open_neuro = 0\n",
    "    fmri = 0\n",
    "    res = 0\n",
    "    r_out = r.json()\n",
    "    date = r_out['date']\n",
    "    src = r_out['source']\n",
    "    try:\n",
    "        doi = r_out['documents'][0]['passages'][0]['infons']['article-id_doi']\n",
    "    except:\n",
    "        doi = None\n",
    "    try:\n",
    "        yr = r_out['documents'][0]['passages'][0]['infons']['year']\n",
    "    except:\n",
    "        yr = None\n",
    "    if yr not in ['2008', '2009', '2010', '2011', '2012', '2013', \n",
    "                  '2014', '2015', '2016', '2017', '2018', '2019']:\n",
    "        return(date, src, doi, git_hits, osf_hits, nda_hits, open_neuro, fmri, res, yr)\n",
    "    else:\n",
    "        for passage in r_out['documents'][0]['passages']:\n",
    "            if re.search(re.compile('github'), passage['text'].lower()):\n",
    "                git_hits = git_hits + 1\n",
    "            if re.search(re.compile('osf\\.io'), passage['text'].lower()):\n",
    "                osf_hits = osf_hits + 1\n",
    "            if re.search(re.compile('nda\\.nih\\.gov'), passage['text'].lower()):\n",
    "                nda_hits = nda_hits + 1\n",
    "            if re.search(re.compile('openneuro'), passage['text'].lower()):\n",
    "                open_neuro = open_neuro + 1\n",
    "            if re.search(re.compile('fmri'), passage['text'].lower()):\n",
    "                fmri = fmri + 1\n",
    "            if re.search(re.compile('results'), passage['text'].lower()):\n",
    "                res = res + 1\n",
    "        \n",
    "        return(date, src, doi, git_hits, osf_hits, nda_hits, open_neuro, fmri, res, yr)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## loop to get everything\n",
    "realistically, this took about two weeks. I ran this in small batches and saved the files iteratively. The second code block gives an example. There ended up being 168 papers that we couldn't retrieve."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#git_links = []\n",
    "#missed = []\n",
    "\n",
    "#for i, j in enumerate(unique_ids):\n",
    "#    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + j + '/unicode'\n",
    "#    try:\n",
    "#        r = requests.get(pth)\n",
    "#        git_links.append(return_useful_data(r) + (j, i))\n",
    "#    except:\n",
    "#        missed.append([i,j])\n",
    "\n",
    "#df = pd.DataFrame(git_links, columns=['date', 'src', 'doi', 'git_hits', \n",
    "#                                      'osf_hits', 'nda_hits', 'open_neuro', \n",
    "#                                      'fmri', 'res', 'yr', 'pmcid', 'idx'])\n",
    "#df.to_csv('/home/riddleta/ac_knowl/output/bionlp_03.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# like this:\n",
    "\n",
    "#git_links = []\n",
    "#missed = []\n",
    "#progress = [i for i in range(0, 15000, 2500)]\n",
    "\n",
    "#for i, j in enumerate(unique_ids[625000:]):\n",
    "#    if i in progress:\n",
    "#        print(i)\n",
    "#    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + j + '/unicode'\n",
    "#    try:\n",
    "#        r = requests.get(pth)\n",
    "#        git_links.append(return_useful_data(r) + (j, i))\n",
    "#    except:\n",
    "#        missed.append([i,j])\n",
    "#    if i==15000:\n",
    "#        break"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## target the subset of papers I want for a full text download"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/riddleta/ac_knowl/ac_knowl/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3049: DtypeWarning: Columns (10) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    }
   ],
   "source": [
    "files = glob.glob('/home/riddleta/ac_knowl/output/bio*.csv')\n",
    "file_list = []\n",
    "for filename in files:\n",
    "    df = pd.read_csv(filename, index_col=None, header=0)\n",
    "    file_list.append(df)\n",
    "\n",
    "frame = pd.concat(file_list, axis=0, ignore_index=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Some of the years were formatted incorrectly or missed entirely. This resolves that."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "frame['yr_fixed'] = frame['yr'].replace({'2016;':'2016'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "28"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pmids = frame[pd.isna(frame.yr_fixed)].pmcid.tolist()\n",
    "len(pmids)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/riddleta/ac_knowl/ac_knowl/lib/python3.7/site-packages/ipykernel_launcher.py:9: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  if __name__ == '__main__':\n"
     ]
    }
   ],
   "source": [
    "for pmid in pmids:\n",
    "    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + pmid + '/unicode'\n",
    "    r = requests.get(pth)\n",
    "    r_out = r.json()\n",
    "    for i in range(0, 5):\n",
    "        if 'year' in r_out['documents'][0]['passages'][i]['infons'].keys():\n",
    "            year = r_out['documents'][0]['passages'][i]['infons']['year']\n",
    "    \n",
    "    frame['yr_fixed'][frame.pmcid==pmid] = year\n",
    "            "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "frame['yr_ints'] = frame.yr_fixed.astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_targets = frame[frame.yr_ints>2008]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(2600884, 15)"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_targets.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "def already_retrieved_papes(dirs):\n",
    "    pmcids = []\n",
    "    for d in dirs:\n",
    "\n",
    "        with open(d) as infile:\n",
    "            dat = json.load(infile)\n",
    "            for paper in dat:\n",
    "                pmcids.append(paper['documents'][0]['id'])\n",
    "    \n",
    "    return(['PMC'+i for i in pmcids])\n",
    "\n",
    "def missed_papes(dirs):\n",
    "    pmcids = []\n",
    "    for d in dirs:\n",
    "        with open(d, 'r') as infile:\n",
    "            dat = infile.readlines()\n",
    "            for paper in dat:\n",
    "                pmcids.append(paper[0:10])\n",
    "    return(pmcids)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "already_retrieved = glob.glob('/home/riddleta/ac_knowl/output/full_texts/papes*')\n",
    "\n",
    "gots = already_retrieved_papes(already_retrieved)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "15710"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "missed = glob.glob('/home/riddleta/ac_knowl/output/full_texts/misses*')\n",
    "\n",
    "misses = missed_papes(missed)\n",
    "len(misses)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1408, 15)"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_targets_sub = df_targets[~df_targets.pmcid.isin(gots)]\n",
    "df_targets_sub.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "def already_retrieved_papes(dirs):\n",
    "    pmcids = []\n",
    "    for d in dirs:\n",
    "\n",
    "        with open(d) as infile:\n",
    "            dat = json.load(infile)\n",
    "            for paper in dat:\n",
    "                pmcids.append(paper['documents'][0]['id'])\n",
    "    \n",
    "    return(['PMC'+i for i in pmcids])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "already_retrieved = glob.glob('/home/riddleta/ac_knowl/output/full_texts/papes*')\n",
    "\n",
    "gots = already_retrieved_papes(already_retrieved)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(791283, 15)"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_targets_sub = df_targets[~df_targets.pmcid.isin(gots)]\n",
    "df_targets_sub.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "file_writes = [i for i in range(1827500, 2700000, 2500)]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "outdat = []\n",
    "missed = []\n",
    "for i, pmcid in enumerate(df_targets_sub.pmcid):\n",
    "    try:\n",
    "        pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + pmcid + '/unicode'\n",
    "        r = requests.get(pth)\n",
    "        r_out = r.json()\n",
    "        outdat.append(r_out)\n",
    "    except:\n",
    "        missed.append(pmcid)\n",
    "    if i+1827500 in file_writes: \n",
    "        with open('/home/riddleta/ac_knowl/output/full_texts/papes_'+str(i+1827500)+'.txt', 'w') as fout:\n",
    "            json.dump(outdat, fout)\n",
    "        if len(missed)>0:\n",
    "            with open('/home/riddleta/ac_knowl/output/full_texts/misses_'+str(i+1827500)+'.txt', 'w') as f:\n",
    "                for item in missed:\n",
    "                    f.write(\"%s\\n\" % item)\n",
    "        outdat = []\n",
    "        missed = []\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0***of***16988\n",
      "1000***of***16988\n",
      "2000***of***16988\n",
      "3000***of***16988\n",
      "4000***of***16988\n",
      "5000***of***16988\n",
      "6000***of***16988\n",
      "7000***of***16988\n",
      "8000***of***16988\n",
      "9000***of***16988\n",
      "10000***of***16988\n",
      "11000***of***16988\n",
      "12000***of***16988\n",
      "13000***of***16988\n",
      "14000***of***16988\n",
      "15000***of***16988\n",
      "16000***of***16988\n"
     ]
    }
   ],
   "source": [
    "with open('/home/riddleta/ac_knowl/output/full_texts/papes_7500.txt') as infile:\n",
    "    dat = json.load(infile)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'date': '20190111',\n",
       " 'source': 'PMC',\n",
       " 'infons': {},\n",
       " 'documents': [{'passages': [{'text': 'First evidence of hybridization between golden jackal (Canis aureus) and domestic dog (Canis familiaris) as revealed by genetic markers',\n",
       "     'offset': 0,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Caniglia;given-names:Romolo',\n",
       "      'name_3': 'surname:Arbanasić;given-names:Haidi',\n",
       "      'name_0': 'surname:Galov;given-names:Ana',\n",
       "      'name_1': 'surname:Fabbri;given-names:Elena',\n",
       "      'name_6': 'surname:Bošković;given-names:Ivica',\n",
       "      'alt-title': 'Golden jackal - dog hybrids',\n",
       "      'article-id_publisher-id': 'rsos150450',\n",
       "      'name_5': 'surname:Florijančić;given-names:Tihomir',\n",
       "      'name_8': 'surname:Randi;given-names:Ettore',\n",
       "      'year': '2015',\n",
       "      'kwd': 'Canis interspecific hybridization gene introgression major histocompatibility complex melanism β-defensin CDB103',\n",
       "      'name_4': 'surname:Lapalombella;given-names:Silvana',\n",
       "      'article-id_doi': '10.1098/rsos.150450',\n",
       "      'section_type': 'TITLE',\n",
       "      'article-id_pmid': '27019731',\n",
       "      'name_7': 'surname:Galaverni;given-names:Marco',\n",
       "      'issue': '12',\n",
       "      'elocation-id': '150450',\n",
       "      'volume': '2',\n",
       "      'article-id_pmc': '4807452',\n",
       "      'license': '© 2015 The Authors. Published by the Royal Society under the terms of the Creative Commons Attribution License http://creativecommons.org/licenses/by/4.0/, which permits unrestricted use, provided the original author and source are credited.',\n",
       "      'type': 'front'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Interspecific hybridization is relatively frequent in nature and numerous cases of hybridization between wild canids and domestic dogs have been recorded. However, hybrids between golden jackals (Canis aureus) and other canids have not been described before. In this study, we combined the use of biparental (15 autosomal microsatellites and three major histocompatibility complex (MHC) loci) and uniparental (mtDNA control region and a Y-linked Zfy intron) genetic markers to assess the admixed origin of three wild-living canids showing anomalous phenotypic traits. Results indicated that these canids were hybrids between golden jackals and domestic dogs. One of them was a backcross to jackal and another one was a backcross to dog, confirming that golden jackal–domestic dog hybrids are fertile. The uniparental markers showed that the direction of hybridization, namely females of the wild species hybridizing with male domestic dogs, was common to most cases of canid hybridization. A melanistic 3bp-deletion at the K locus (β-defensin CDB103 gene), that was absent in reference golden jackal samples, but was found in a backcross to jackal with anomalous black coat, suggested its introgression from dogs via hybridization. Moreover, we demonstrated that MHC sequences, although rarely used as markers of hybridization, can be also suitable for the identification of hybrids, as long as haplotypes are exclusive for the parental species.',\n",
       "     'offset': 136,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'abstract', 'section_type': 'ABSTRACT'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Introduction',\n",
       "     'offset': 1585,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'title_1', 'section_type': 'INTRO'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Interspecific hybridization is relatively frequent in nature, occurring not only in plants but also in animals, where at least 10% of species are involved in admixture and potential introgression. Interspecific hybridization in Canis has been described in a number of studies in North America (e.g. coyote–grey wolf). Moreover, cases of anthropogenic hybridization between wild canids and widespread free-ranging domestic dogs are particularly alarming, because they may threaten the survival of endangered species (e.g. the Ethiopian wolf, Canis simensis), or may deeply change the genetic make-up of wild populations in human-dominated landscapes (e.g. grey wolf).',\n",
       "     'offset': 1598,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'INTRO'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'The golden jackal (Canis aureus) is a medium-sized species, currently distributed in northern and northeastern Africa, southeastern Europe and large parts of southern Asia. European golden jackals were first reported in 1491 in the coastal region of southern Dalmatia, where they still occur. After suffering a severe decline in the first half of the twentieth century, the European population has recovered and has been expanding since the early 1980s, especially in the Balkan regions.',\n",
       "     'offset': 2267,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'INTRO'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'To our knowledge, there are no published cases of recent hybridization between golden jackals and any other canid species in nature, although genome-wide traces of ancestral admixture with wolves have been recently documented, and the assignment of the African Canis aureus lupaster to the wolf clade cannot rule out the hypothesis of ancient hybridization events. The only two documented cases of present-day hybrids were questionable, since they were only based on morphological measurements: in Romania a putative jackal–wolf hybrid was shot in 1942; in Hungary, the skull of a suspected jackal–dog was discovered in 1983, but erroneously reported as the first jackal shot in the country after 41 years.',\n",
       "     'offset': 2755,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'INTRO'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Molecular techniques are routinely used for the identification of closely related species and their hybrids within the first two to three generations of admixture. The combined use of maternal and paternal markers may also reveal the hybridization direction. The mitochondrial DNA control region (mtDNA CR) can identify the maternal ancestry in putative hybrids. A single mtDNA CR haplotype has been found so far in European golden jackal populations, facilitating the identification of golden jackal–dog hybrids through the maternal line. Paternal ancestry of hybrids may be revealed by species-specific Y chromosome markers developed to discriminate between golden jackal and domestic dog males, based on an insertion found in a dog Zfy intron, but not in golden jackal. In addition, putatively neutral biparental markers such as microsatellite loci (STR), originally developed for the domestic dog, can efficiently cross-amplify and identify golden jackal genotypes, thus providing estimates of the proportion of neutral admixture inherited from the mixing parental populations (e.g.).',\n",
       "     'offset': 3466,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'INTRO'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Functional markers could further contribute to the investigation of hybridization and introgression. The major histocompatibility complex (MHC) is a multigene family that is commonly used in adaptive variation studies. MHC genes encode cell-surface glycoproteins that bind and present antigens to T cells, which trigger an appropriate immune response, thus playing a pivotal role in the vertebrate immune system. However, to our knowledge, MHC genes have been used as markers to detect hybridization in only two vertebrate species: a case of hybridization between Iberian ibex and domestic goat, and a study of introgression of dog MHC alleles in wild-living Italian wolves. Other functional mutations that could be used in determining phenotypical variation have been discovered in recent genomic studies, such as a dominant three-nucleotide deletion in the β-defensin CBD103 gene (the K locus) correlated to black coat colour in canids, which could have been introduced from dogs into wild-living wolves in North America and in Italy via hybridization (hereafter referred to as ‘melanistic deletion’).',\n",
       "     'offset': 4557,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'INTRO'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Thus, the primary objective of this study was to document the potential occurrence and directionality of golden jackal–dog hybridization in an expanding jackal population, using combined genetic analyses of 15 microsatellite loci, mtDNA control region and a Y chromosome marker. Our secondary objectives were to assess the presence of a functional melanistic deletion at the β-defensin CBD103 gene in an individual with black coat coloration and to test the applicability of coding markers such as MHC genes to detect hybridization between closely related species, such as the dog and the golden jackal.',\n",
       "     'offset': 5666,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'INTRO'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Material and methods',\n",
       "     'offset': 6273,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'title_1', 'section_type': 'METHODS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Putative golden jackal–dog hybrid samples',\n",
       "     'offset': 6294,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'title_2', 'section_type': 'METHODS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Female golden jackal–dog hybrid (S21) (a) and its forelimb with notable digital pad depigmentation (dog characteristic) and partially joined digital pads of the middle fingers (golden jackal characteristic) (b).',\n",
       "     'offset': 6338,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'fig_caption',\n",
       "      'section_type': 'FIG',\n",
       "      'id': 'RSOS150450F1',\n",
       "      'file': 'rsos150450-g1.jpg'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Male golden jackal–dog hybrid (S22) (a) and its hind leg with dewclaw (b).',\n",
       "     'offset': 6552,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'fig_caption',\n",
       "      'section_type': 'FIG',\n",
       "      'id': 'RSOS150450F2',\n",
       "      'file': 'rsos150450-g2.jpg'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Male golden jackal–dog hybrid (60c) with black coat coloration (a) and ears with rounded tip (b) (dog characteristics), and forelimbs with partially joined digital pads of the middle fingers (golden jackal characteristics) (c).',\n",
       "     'offset': 6629,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'fig_caption',\n",
       "      'section_type': 'FIG',\n",
       "      'id': 'RSOS150450F3',\n",
       "      'file': 'rsos150450-g3.jpg'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'We collected muscle tissue samples from three putative golden jackal–dog hybrids legally harvested by hunters in Croatia. The putative hybrids were initially identified on the basis of their unusual morphological traits. The individual S21 (figure 1) was an adult female showing light coat colour, digital pad depigmentation and atypical long ears with rounded tip (whereas golden jackals have shorter triangle-shaped ears). However, the digital pads of the middle fingers were partially joined as commonly occurs in golden jackals, but not in dogs (I. Bošković 2012, unpublished data). The individual S22 (figure 2) was a juvenile male found and shot together with female S21, probably representing a mother–son pair. It displayed a dog-like morphology, particularly similar to the Istrian shorthaired hound breed (very short hair on the head, white coat colour with sparse patches of light brown, dewclaws on hind legs). However, the animal’s tail was shorter and thicker than typical for this dog breed and more similar to that of a golden jackal. Thus, we hypothesized it could have originated through a backcross between the putative hybrid female S21 and a male Istrian shorthaired hound dog. The third individual (60c) (figure 3) was a male that exhibited black coat coloration, atypical for golden jackals, and other dog characteristics such as ears with rounded tip. However, the digital pads on the middle fingers of its forelimbs were partially joined, as in golden jackals.   ',\n",
       "     'offset': 6859,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Tissue samples were stored in 96% ethanol at −20°C prior to analyses. DNA was extracted using a Wizard Genomic DNA Purification kit (Promega, USA).',\n",
       "     'offset': 8356,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Reference samples',\n",
       "     'offset': 8507,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'title_2', 'section_type': 'METHODS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'To correctly identify the origin and ancestry of the putative golden jackal–dog hybrids, we used as reference source populations 50 jackal samples from Croatia and 51 mixed breed dog samples from Croatia, previously genotyped at STR markers. Reference jackal samples were legally shot or road-killed in Croatia.',\n",
       "     'offset': 8525,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Mitochondrial DNA analysis',\n",
       "     'offset': 8839,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'title_2', 'section_type': 'METHODS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'We amplified the hypervariable left domain of the mtDNA CR using primers L-Pro and H-576 (electronic supplementary material, table S1). The polymerase chain reactions (PCRs) were carried out in 30\\u2009μl containing 1× Qiagen HotStarTaq Master Mix (Qiagen HotStarTaq Master Mix Kit, Qiagen, USA), 0.2\\u2009μM of forward and reverse primer and 3\\u2009μl template DNA. Cycling conditions were the following: 95°C for 15 min, 35 cycles of 40\\u2009s at 94°C, 50\\u2009s at 55°C, 1 min at 72°C, and 10\\u2009min final extension at 72°C. The amplification products were purified (Wizard SV Gel and PCR Clean-Up System, Promega) and sequenced using L-Pro primer. Sequences were aligned in BioEdit with the only golden jackal haplotype found so far and 12 dog haplotypes found among mixed breed dogs from Croatia.',\n",
       "     'offset': 8866,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Y chromosome analysis',\n",
       "     'offset': 9661,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'title_2', 'section_type': 'METHODS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'We analysed the two male samples S22 and 60c using a PCR-based Y chromosome marker method in which the dog DNA template produces two amplicons, whereas the golden jackal template produces only one (electronic supplementary material, table S1).',\n",
       "     'offset': 9683,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Autosomal microsatellite loci analysis',\n",
       "     'offset': 9927,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'title_2', 'section_type': 'METHODS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'We genotyped the reference populations and the putative hybrid samples at 15 unlinked autosomal canine STRs using the same procedure described in Fabbri et al. (electronic supplementary material, table S2). The average number of alleles (Na) and private alleles (Np), and the observed and expected heterozygosity (Ho, He) were estimated using GenAlEx v. 6.5 as measures of genetic diversity. Exact tests for Hardy–Weinberg equilibrium were computed using the Guo and Thompsons’ Markov chain method as implemented in the software Genpop v. 4.00. The sequential Bonferroni correction test for multiple comparisons was used to adjust significance levels. The 15-STR multilocus genotypes of the reference jackals and dogs, and of the three putative hybrids were used to distinguish species and to detect putative admixed individuals and their ancestry through two different methodologies: (i) a multivariate analysis: principal coordinates analysis (PCoA) of individual STR genotypes implemented in GenAlEx; (ii) a Bayesian clustering procedure implemented in Structure v. 2.3.4, which estimates the admixture proportion of each individual genotype.',\n",
       "     'offset': 9966,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'We used the Admixture model with independent allele frequencies (I-model) running five replicates of K from 1 to 5 using 5×105 iterations of MCMC following a burn-in period of 5×104 iterations. The optimal number of populations K was determined according to Evanno et al. independently from any prior non-genetic information (option usepopinfo not active). For each group, we assessed the average proportion of membership (Qi) to each different clusters, and individual assignment was consequently based on the proportions of membership (qi) estimated for every single individual. Based on these first Structure results, admixture analyses were performed again assuming two reference groups (jackal and dog) for the assignment of the putative jackal–dog hybrids (PHy). Structure was run with K=2, with the option ‘usepopinfo’ activated or not. In the former case, we assumed that reference jackals and dogs were a priori correctly identified and assigned to their own clusters (popflag=1), while the putative hybrids were left to be assigned (popflag=0).',\n",
       "     'offset': 11116,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'The software NewHybrids was then used to compute the posterior probability for each genotype to belong to each of the six following classes: jackal (J) and dog (D) parentals, F1 and F2, backcrosses of F1 with dogs (BC1D) and with jackals (BC1J). Posterior distributions were evaluated after 105 iterations of the Monte Carlo Markov chains, following a burn-in period of 104 iterations, without any individual or allele frequency prior information, with ‘Jeffreys-like’ or ‘Uniform’ priors for mixing both proportions and allele frequencies.',\n",
       "     'offset': 12179,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'In addition, we used Hybridlab to evaluate the power of the 15 STRs to correctly detect a priori known parentals, hybrids and backcrosses. We used the 50 reference golden jackals and the 51 reference dogs to simulate 50 genotypes for each of the following classes: first and second generation hybrids (F1, F2), first and second generation backcrosses with golden jackal and dog (BC1J, BC1D, BC2J, BC2D). The simulated genotypes were analysed in Structure and NewHybrids using the run parameters described before (‘Admixture’ and the ‘I’ models, without any prior population information).',\n",
       "     'offset': 12728,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'K locus analysis and major histocompatibility complex analyses',\n",
       "     'offset': 13324,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'title_2', 'section_type': 'METHODS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Since one putative hybrid showed a black coat (figure 3), we assessed the presence of the functional melanistic deletion at the β-defensin CBD103 gene (corresponding to the K locus), which determines black coat colours in dogs and wolves, following the amplification protocol described in Caniglia et al. (electronic supplementary material, table S1).',\n",
       "     'offset': 13387,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'The 50 golden jackal samples and the three putative hybrid samples were further analysed for MHC DLA-DRB1, DQA1 and DQB1 class II genes using cloning/sequencing method. The primers used to amplify exon 2 were: for DLA-DRB1, forward DRBF and reverse DRB1R; for DLA-DQA1, forward DQAin1 and reverse DQAIn2; for DLA-DQB1, forward DQB1BT7 and reverse DQBR3, ACCTGGGTGGGGAGCCCG (primer designed in this study based on the sequence published in Wagner et al.) (electronic supplementary material, table S1). Amplifications were carried out by PCR in a total volume of 25\\u2009μl containing 150–250\\u2009ng of genomic DNA, 1× QIAGEN HotStarTaq Master Mix (Qiagen, Hilden, Germany) (consisting of 1× PCR buffer, 200\\u2009μM of each dNTP and 2.5 units HotStartTaq DNA polymerase) and 0.2\\u2009μM of each primer. A negative control containing no DNA template was included in each amplification run to detect any contamination. All amplifications were performed using a standard touchdown PCR protocol as described in Kennedy et al.. PCR products were visualized on 1% agarose gels stained with SYBR Safe DNA gel stain (Invitrogen) and purified by the Wizard SV Gel and PCR Purification Clean-Up System (Promega). Sequencing for typing was performed using the same primers as for PCR in reverse direction for all three loci. Confirmation of new alleles was performed by sequencing in both directions and by further DNA cloning. The PCR products were ligated into vectors and transformed into bacteria using the pGEM-T Vector System II (Promega). Plasmid DNA from 8 to 12 positive clones per individual was isolated using the Promega Wizard Plus SV Miniprep DNA Purification System, and inserts were sequenced using the PCR primers described above. Sequence processing and analysis were performed with BioEdit. To identify alleles in heterozygous animals, we used the Applied Biosystems SeqScape® software, which is designed for analysis based on a locus-specific allele reference library, and previously described canid alleles that we obtained from the Immuno Polymorphism (IPD)—MHC Database (L. J. Kennedy 2013, personal communication), as described in Arbanasić et al.. Three-locus haplotypes were unambiguously identified in a sequential process and confirmed by Arlequin v. 3.11.',\n",
       "     'offset': 13742,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'METHODS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Results',\n",
       "     'offset': 16014,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'title_1', 'section_type': 'RESULTS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Mitochondrial DNA',\n",
       "     'offset': 16022,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'title_2', 'section_type': 'RESULTS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'We obtained fragments of 550\\u2009bp of the mtDNA CR from the three putative hybrid samples, which all carried sequences identical to the reference golden jackal mtDNA CR haplotype (GenBank accession no. KF588364), suggesting their golden jackal maternal ancestry.',\n",
       "     'offset': 16040,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Y chromosome marker',\n",
       "     'offset': 16302,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'title_2', 'section_type': 'RESULTS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Both male samples (S22 and 60c) produced two amplicons of the Zfy intron, characteristic of the domestic dog, suggesting their dog paternal ancestry.',\n",
       "     'offset': 16322,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Autosomal microsatellite loci',\n",
       "     'offset': 16472,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'title_2', 'section_type': 'RESULTS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'All the 15 STR loci were polymorphic in both reference populations, showing from three to 19 alleles in dog and from three to 10 alleles in jackal samples and an average number of alleles per locus of 8.2 (s.e. 0.98) and 4.3 (s.e. 0.50), respectively (electronic supplementary material, table S3). As expected, dogs showed an observed and expected heterozygosity higher than jackals and an average number of private alleles of 4.67 versus 1.07.',\n",
       "     'offset': 16502,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Principal coordinates analysis obtained by GenAlEx. The two principal axes (PC-I and PC-II) cumulatively explain 35.8% of the total genetic diversity. Blue dots represent the Croatian dog references, red dots the Croatian golden jackal references and triangles the golden jackal–dog hybrids: S21, S22 and 60c.',\n",
       "     'offset': 16947,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'fig_caption',\n",
       "      'section_type': 'FIG',\n",
       "      'id': 'RSOS150450F4',\n",
       "      'file': 'rsos150450-g4.jpg'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'The PCoA results are shown in figure 4, where the individual scores were plotted onto the two principal axes (PC-I and PC-II), which cumulatively explain 35.8% of the total genetic diversity. The PCoA split jackals and dogs into two clearly separate clusters, with all dogs and jackals correctly identified by their genotypes. The only exception is three putative hybrids that showed intermediate positions between the two clusters, suggesting an admixed origin. ',\n",
       "     'offset': 17259,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Bayesian analysis obtained by Structure using admixture models and K=2. Each individual is represented by a vertical bar fragmented in K sections of different length, according to their membership proportion in the two inferred genetic clusters: the red represent the dog component and green the jackal component. Dogs, Croatian dog references; jackals, Croatian golden jackal references; hybrids, golden jackal–dog hybrids; simulated genotypes by Hybridlab: F1 and F2, first and second generation hybrids, BC1J and BC2J, first and second backcrosses of F1 with golden jackals; BC1D and BC2D, first and second backcrosses of F1 with dogs.',\n",
       "     'offset': 17723,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'fig_caption',\n",
       "      'section_type': 'FIG',\n",
       "      'id': 'RSOS150450F5',\n",
       "      'file': 'rsos150450-g5.jpg'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Assignment of observed and simulated dog, golden jackal and admixed genotypes to reference parental populations (dogs, jackals), their first (F1) and second (F2) generation hybrids, their first (BC1) and second (BC2) backcrosses with jackals and dogs. The assignments of individual genotypes of samples S21, S22 and 60C are also reported. Qj, average proportion of membership to jackal cluster; qj, proportion of membership to jackal cluster of individual genotypes; P, mean posterior probability for each population to belong to each of the six classes (D, dog; J jackal; F1, first generation hybrid; F2, second generation hybrid; BCD, backcrosses of F1 with dogs; BCJ, backcrosses of F1 with jackal); p, individual posterior probability to belong to each of the six classes. The highest values are marked in italic.',\n",
       "     'offset': 18364,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'table_caption',\n",
       "      'section_type': 'TABLE',\n",
       "      'id': 'RSOS150450TB1',\n",
       "      'file': 'RSOS150450TB1.xml'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': '\\tqj (90% CI) Admixture model\\tqj (90% CI) UsePopInfo model\\tqj (90% CI) Admixture model\\tqj (90% CI) UsePopInfo model\\tpD\\tpJ\\tpF1\\tpF2\\tpBCD\\tpBCJ\\nS21\\t0.588 (0.396–0.770)\\t0.561 (0.403–0.717)\\t0.598 (0.416–0.772)\\t0.586 (0.413–0.750)\\t0.000\\t0.000\\t0.638\\t0.075\\t0.002\\t0.285\\nS22\\t0.227 (0.063–0.411)\\t0.319 (0.166–0.477)\\t0.255 (0.105–0.429)\\t0.275 (0.130–0.440)\\t0.000\\t0.000\\t0.000\\t0.018\\t0.982\\t0.000\\n60c\\t0.849 (0.687–1.000)\\t0.735 (0.584–0.875)\\t0.832 (0.680–0.954)\\t0.791 (0.646–0.912)\\t0.000\\t0.005\\t0.000\\t0.003\\t0.000\\t0.991\\n',\n",
       "     'offset': 19182,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'table',\n",
       "      'section_type': 'TABLE',\n",
       "      'id': 'RSOS150450TB1',\n",
       "      'File': 'RSOS150450TB1.xml'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'At K=2 (the optimal number of genetic clusters), results from Structure admixture analyses showed that all dogs were assigned to a single cluster with an average membership proportion QD\\u2009=\\u20090.998 and individual proportions of admixture qD ranging from 0.984 to 0.999. Jackals were assigned to the other cluster with Qj=0.999 and a qJ ranging from 0.994 to 0.999. The three putative hybrids S21, S22 and 60c showed qJ=0.588 (90% confidence intervals CI: 0.396–0.770), 0.227 (0.063–0.411) and 0.849 (0.687–1.000), respectively (figure 5 and table 1).  ',\n",
       "     'offset': 19706,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'When we assumed that reference jackals and dogs were a priori correctly identified and assigned to their own clusters (popflag=1), while the putative hybrids were left to be assigned (popflag=0), results obtained from five Structure runs using K=2, PopINFO and ‘I’ model were concordant with those obtained without any prior information (table 1).',\n",
       "     'offset': 20266,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Results of admixture analyses showed that 97.34% of the simulated admixed genotypes can be correctly identified as admixed at threshold qi=0.980 (after, we used as a threshold the minimum qi value observed in reference populations using only real data: 0.984). All the F1, F2, BC1J and BC1D were correctly identified as admixed using both Structure models (Admixture and PopInfo). Only 6% of BC2J and 10% of BC2D genotypes showed a qi>0.980 (to jackal cluster and to dog cluster, respectively). Using the Admixture model with selection Flag to reference populations and PopInfo model, all simulated were correctly identified as admixed (figure 5).',\n",
       "     'offset': 20618,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'In agreement with Structure results, the three putative hybrids were identified also by NewHybrids as admixed. S21 showed a posterior probability p=0.638 to belong to F1 class; S22 had p=0.982 to belong to backcross with dog and 60c had p=0.991 to be a backcross with jackal (table 1).',\n",
       "     'offset': 21266,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Relatedness analysis of female S21 and juvenile male S22 revealed that they shared at least one allele on each STR locus, confirming their mother–son relationship (electronic supplementary material, table S4).',\n",
       "     'offset': 21552,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'K locus and major histocompatibility complex',\n",
       "     'offset': 21764,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'title_2', 'section_type': 'RESULTS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Number and frequency (in parenthesis) of genotypes at the β-defensin CBD103 gene: K+/K+, homozygotes wild-type (no deletion); K+/KB, heterozygotes for the KB melanistic deletion; KB/KB, homozygotes for the KB melanistic deletion. Na, number of alleles; Np, number of private alleles; Ho and He, observed and expected heterozygosity; HWE prob., probability test for Hardy–Weinberg equilibrium.',\n",
       "     'offset': 21809,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'table_caption',\n",
       "      'section_type': 'TABLE',\n",
       "      'id': 'RSOS150450TB2',\n",
       "      'file': 'RSOS150450TB2.xml'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'ref. pop (N)\\tNa (s.e.)\\tNp (s.e.)\\tHo (s.e.)\\tHe (s.e.)\\tHWE prob.\\tK+/K+\\tK+/KB\\tKB/KB\\ndogs (51)\\t8.20 (0.99)\\t4.67 (0.80)\\t0.61 (0.05)\\t0.68 (0.05)\\t0.0000\\t36 (0.72)\\t11 (0.22)\\t3 (0.06)\\njackals (51)\\t4.33 (0.50)\\t1.07 (0.30)\\t0.46 (0.04)\\t0.54 (0.05)\\t0.0028\\t51 (1.0)\\t0\\t0\\n',\n",
       "     'offset': 22205,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'table',\n",
       "      'section_type': 'TABLE',\n",
       "      'id': 'RSOS150450TB2',\n",
       "      'File': 'RSOS150450TB2.xml'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'The black hybrid 60c showed a heterozygote genotype at the K locus (K+/KB), whereas the deletion was absent in all reference jackals but present in 14 of the 51 reference dogs (table 2). ',\n",
       "     'offset': 22462,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'aL. J. Kennedy 2013, personal communication.',\n",
       "     'offset': 22650,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'table_footnote', 'section_type': 'TABLE'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'DLA-DRB1/DQA1/DQB1 haplotypes found in 50 golden jackals and genotypes found in three golden jackal–dog hybrids. Alleles in bold were found exclusively in golden jackal. Italicized haplotype was predominant in golden jackal population.',\n",
       "     'offset': 22695,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'table_caption',\n",
       "      'section_type': 'TABLE',\n",
       "      'id': 'RSOS150450TB3',\n",
       "      'file': 'RSOS150450TB3.xml'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': '\\tindividual\\tgenotypes identified in three golden jackal–dog hybrids\\thaplotype determination\\t\\ngolden jackal–dog hybrids\\tS21\\t00901\\t00402\\t02305\\tgolden jackal\\t\\n\\t\\t00803\\t00301\\t00401\\tCroatian sheepdog, border terriera\\t\\n\\tS22\\t00901\\t00402\\t02305\\tgolden jackal\\t\\n\\t\\t02001\\t00401\\t01303\\tmore than 25 dog breeds \\t\\n\\t60c\\t00901\\t00402\\t02305\\tgolden jackal\\t\\n\\t\\t00101\\t00101\\t00201\\t50 dog breeds \\t\\n',\n",
       "     'offset': 22933,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'table',\n",
       "      'section_type': 'TABLE',\n",
       "      'id': 'RSOS150450TB3',\n",
       "      'File': 'RSOS150450TB3.xml'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'All 50 golden jackals and three putative golden jackal–dog hybrids successfully amplified at all three MHC loci analysed. In golden jackals, we found four DRB1, two DQA1 and two DQB1 alleles (table 3). Of these, three DLA-DRB1 (DLA-DRB1*13001, 13101 and 04503), one DLA-DQA1 (DLA-DQA1*03001) and both DLA-DQB1 alleles (DLA-DQB1*02305 and 06801) were not identified before in any canid species. New alleles were assigned official names by the DLA Nomenclature Committee (GenBank accession numbers KT159767, KT182924–KT182928). Identified alleles combined to form four DLA-DRB1/DQA1/DQB1 three-locus haplotypes (table 3). In all three putative golden jackal–dog hybrids, we identified the same DLA-DRB1*00901/DQA1*00402/DQB1*02305 haplotype, which was exclusive to and the most frequent among golden jackals in our reference samples. Among the three alleles that constitute this haplotype, the allele DQB1*02305 appears to be specific for golden jackals, whereas alleles DRB1*00901 and DQA1*00402 were previously found in dogs, but not in the same haplotype (L. J. Kennedy 2013, personal communication). ',\n",
       "     'offset': 23308,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'The other haplotype and alleles present in hybrids were not seen in this golden jackal cohort, but are common in dogs. The haplotype DLA-DRB1*00803/DQA1*00301/DQB1*00401 found in putative hybrid S21 was previously detected in the Croatian sheepdog and border terrier (L. J. Kennedy 2013, personal communication), while haplotypes DLA-DRB1*02001/DQA1*00401/DQB1*01301 and DLA-DRB1*00101/DQA1*00101/DQB1*00201 detected in putative hybrids S22 and 60c, respectively, were found in numerous dog breeds (table 3).',\n",
       "     'offset': 24417,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'RESULTS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Discussion',\n",
       "     'offset': 24926,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'title_1', 'section_type': 'DISCUSS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Using genetic markers we confirmed that the three individuals with anomalous phenotypic characters were indeed interspecific hybrids, namely a first generation hybrid between golden jackal and domestic dog (female S21), a backcross to dog (juvenile male S22) and a backcross to jackal (male 60c). The existence of backcrosses confirms that golden jackal–dog hybrids are fertile. Although these two species are estimated to have diverged about 1.7\\u2009Ma, or 0.4\\u2009Ma according to Freedman et al., the occurrence of their hybrids and the fact that they are fertile do not come as a surprise, since reproductive isolation between pairs of geographically overlapping species evolves progressively and may need hundreds to millions of generations to complete. Golden jackals and dogs, together with grey wolves and coyotes, form a monophyletic clade and are more closely related than the golden jackal to two other jackal species, the side-striped jackal (Canis adustus) and the black-backed jackal (Canis mesomelas). Furthermore, Moura et al. also reported weak evidence for current hybridization between grey wolves and golden jackals, as they identified several Bulgarian wolves exhibiting mixed ancestry with the jackal cluster.',\n",
       "     'offset': 24937,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'The fact that golden jackal–dog hybrids were not recorded before in the wild might be a consequence of low research interest in golden jackals in the past, as they were not present in countries with the most active research communities. The golden jackal range has been expanding from southeastern Europe northwards and westwards in the last 30 years, thus a number of monitoring projects have been recently activated, increasing the possibility to record phenotypically abnormal individuals that might have always occurred in the populations. Moreover, before the advent of genetic tools, hybrids without a clear phenotypic signature could have gone undetected. On the other hand, the occurrence of golden jackal–dog hybrids might indeed be increasing because of several factors. First, higher population densities of golden jackals due to their recent expansion in the Balkan Peninsula could enhance the encounters with stray and free-ranging dogs, which are quite abundant in Croatia and occur in the same areas where jackals live, hunt and gather food (I. Bošković 2012, unpublished data). It is known that the golden jackal is opportunistic in nature, primarily uses easily accessible human-derived food and greatly benefits from the presence of agricultural surroundings. Second, high mortality rates associated with jackal culling in Croatia, which mainly takes place from November to January and partially coincides with the jackal breeding season, can disrupt the social structure and promote hybridization with dogs, as suggested for wolf and coyote or wolf and dog. Yet, our results indicate that at least in Croatia introgressive hybridization between the two species is not a widespread phenomenon. We showed that Croatian golden jackals and dogs remain separated, forming two well-differentiated genetic entities where individuals are assigned to their respective cluster (figures 4 and 5), without significant ancestry in the other cluster.',\n",
       "     'offset': 26166,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Golden jackals exhibit lower genetic diversity measures than dogs (number of alleles, number of private alleles, observed and expected heterozygosity) (electronic supplementary material, table S3). This is in line with previous investigations (e.g.), which compared diversity measures between grey wolves and dogs and consistently reported higher values in dogs, probably due to the multiple events of domestication and introgression during dog history, or to an ascertainment bias. However, this pattern could be different when taking into account the genetic variability of single dog breeds, which show much less genetic variation than what can be observed across breeds or in village dogs.',\n",
       "     'offset': 28128,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'In order to evaluate the power of the STR loci in the determination of golden jackal–dog hybrids, we also used simulated genotypes for six different hybrid classes. The analysis of the simulation results revealed that these 15 STRs are variable enough to detect 100% of parentals and F1, F2, BC1J and BC1D hybrids using a threshold of 0.98. Only 6% and 10% of second backcrosses with jackals and dogs, respectively, were undetected using the Admixture model and no other a priori information in the Bayesian assignment procedure. Thus, the 15 STRs seem to be reliable and powerful enough to detect hybrids between golden jackals and dogs.',\n",
       "     'offset': 28822,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'The analysis of the mtDNA CR and of the Zfy intron on the Y chromosome appears to be diagnostic, since these markers proved to be fixed for different haplotypes in golden jackals and dogs (this study,). All the three anomalous individuals carried a golden jackal mtDNA CR haplotype, whereas the two males showed Y chromosome marker amplicons characteristic of dogs, enabling us to deduce their lines of descent (electronic supplementary material, figure S1): a female golden jackal mated with a male domestic dog to produce the hybrid F1 female S21, which further mated with a domestic dog producing the backcross S22. The father of the male 60c was a golden jackal–dog hybrid, whose mother was a golden jackal and whose father was a dog, and he further mated with a female golden jackal. Accordingly, in both cases of F1 hybrids (one documented—S21, and the other deduced—the father of 60c, which was not sampled), hybridization took place between female golden jackals and male dogs, which is congruent with the sexual asymmetry present in most hybridizations between domestic dogs and wild Canis species, e.g. grey wolf and Ethiopian wolf, though occasionally violated. The finding of a heterozygote genotype at the K locus (K+/KB) in the backcross to that of a golden jackal 60c with black coat coloration, the absence of the melanistic K locus mutation in all reference jackals and its presence in 14 of the dog samples (table 2) suggest that this hybrid could have received the K locus deletion from dogs. In this way, golden jackal could join the panel of canid species that possibly derived their melanistic K locus mutation through hybridization with domestic dogs, namely grey wolves and coyotes. Further, our findings cast doubts on the hypothesis of Ambarli & Bilgin, according to which melanism in the golden jackal they camera-trapped was due to an independent mutation instead of introgression from the domestic dog, and suggest that this individual might be another case of golden jackal–dog hybrid.',\n",
       "     'offset': 29463,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'The MHC loci further confirmed that the three anomalous individuals described here were hybrids. Three out of four DLA-DRB1 alleles, one out of two DLA-DQA1 alleles and both DLA-DQB1 alleles found in golden jackals in this research (table 3) have not been identified before in any canid species. Thus, they could represent golden jackal private alleles that can be used as species-diagnostic markers. However, as the three hybrids in our research did not possess any of those jackal private alleles on DRB1 and DQA1 loci, the species (and hence, hybridization) determination could not have been performed using either of those two loci separately. Notwithstanding, the combined use of alleles on three loci enabled us to reliably confirm golden jackal ancestry in all three individuals in our research, since all of them possessed a three-locus haplotype that is specific and exclusive for golden jackals (DRB1*00901/DQA1*00402/DQB1*02305, table 3) and has not been found in dogs (L. J. Kennedy 2013, personal communication). In other words, even when alleles on particular MHC loci are not species-specific, their two- or three-locus haplotypes might prove to be, due to high levels of linkage disequilibrium across large stretches of this genomic region, where particular combinations of alleles at neighbouring loci are maintained by selection. Further, the other haplotype possessed by each of the three individuals was characteristic of dogs and none of those was found in our reference jackal samples (table 3). In addition, none of the alleles comprising those haplotypes was found in jackals, indicating that even a single locus would suffice to confirm the dog ancestry in those samples. Likewise, for the confirmation of jackal ancestry in this research, the exclusive use of DQB1 locus would suffice since both alleles found in the reference jackal population are private. However, it would come as no surprise if additional alleles were found on that locus as more jackal samples become typed, and if some of them were to be shared between golden jackals and domestic dogs. This phenomenon, known as trans-species polymorphism, is characteristic of MHC genes, mainly occurs among closely related species and is a consequence of balancing selection, which acts on MHC genes over the long term and maintains ancestral polymorphism in descendant species. Indeed, when trans-species polymorphism is present, analysing three-locus haplotypes instead of individual MHC loci should add power to detect possible introgression. The procedure for analysing MHC markers is quite straightforward, and time and cost efficient even when analysing three loci. Albeit they were applied as markers to detect hybridization in only two studies up to now, our results indicate that MHC genes can be used as suitable molecular markers for the identification of vertebrate species and for the determination of hybridization events, at least when data on reference parental populations are available and they are not closely related.',\n",
       "     'offset': 31487,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Hybridization might play an important role in the process of animal evolution, especially in rapidly changing environments. Hybridizing with dogs, golden jackals might increase their variability and thus facilitate their long-term adaptation. In addition, adaptive introgression may be facilitated for genes evolving under multi-allelic balancing selection, such as the vertebrate MHC system, where increased resistance to infectious diseases from adapted MHC variants might be transferred to closely related recipient species, as long as fertile hybrids can be formed. This could be the case for the jackal, where standing variation to face new adaptive challenges, such as new parasites associated with human-related food sources, may be low but could be compensated by introgression from dogs, which have already adapted to human-related environments. Adaptive introgression of MHC genes was recently suggested between two closely related species of newts and from archaic to modern humans. Sexual selection may also facilitate introgression of dog MHC alleles into the genomes of golden jackals, potentially contributing to resistance against currently prevalent parasites, as was proposed for three-spined sticklebacks (Gasterosteus aculeatus). It is thought that MHC-based mate choice may allow genes to cross species boundaries if parasite selection is strong enough that the benefits of hybridization overcome its costs.',\n",
       "     'offset': 34510,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Conversely, hybridization and introgression may also have harmful effects on the fitness of animal populations in the wild, causing loss of genetic diversity due to genetic homogenization and/or outbreeding depression in local populations. In addition, possible cross-species transmission of canine diseases might pose another risk for the expanding golden jackal populations.',\n",
       "     'offset': 35939,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'In summary, in this paper we document the first occurrence of three cases of golden jackal–dog hybrids. However, the frequency of hybridization events, the extent of possible genetic introgression of dog genes into European golden jackal populations and the consequences on genetic diversity and population fitness (either beneficial or unfavourable) still remain to be investigated.',\n",
       "     'offset': 36316,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'DISCUSS'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Supplementary Material',\n",
       "     'offset': 36702,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'title_1', 'section_type': 'SUPPL'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Supplementary Material',\n",
       "     'offset': 36725,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'title_1', 'section_type': 'SUPPL'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Supplementary Material',\n",
       "     'offset': 36748,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'title_1', 'section_type': 'SUPPL'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Supplementary Material',\n",
       "     'offset': 36771,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'title_1', 'section_type': 'SUPPL'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Supplementary Material',\n",
       "     'offset': 36794,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'title_1', 'section_type': 'SUPPL'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Ethics',\n",
       "     'offset': 36817,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'title_1', 'section_type': 'SUPPL'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Tissue samples used in this study were obtained from golden jackal individuals legally harvested by hunters or road-killed. No animal was killed for the purpose of this study.',\n",
       "     'offset': 36824,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'SUPPL'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Data accessibility',\n",
       "     'offset': 37000,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'title_1', 'section_type': 'SUPPL'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'The dataset supporting this article has been uploaded as part of the electronic supplementary material.',\n",
       "     'offset': 37019,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'SUPPL'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': \"Author' contributions\",\n",
       "     'offset': 37123,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'title_1', 'section_type': 'AUTH_CONT'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'A.G., E.F., R.C., H.A. and E.R. designed the study; A.G., E.F., H.A., R.C., S.L. and M.G. carried out the molecular work and participated in data analysis; T.F. and I.B. conceived the study and collected field data; E.R. coordinated the study; A.G., E.F., R.C., H.A. and E.R. drafted the manuscript. All authors revised the manuscript and gave final approval for publication.',\n",
       "     'offset': 37145,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'AUTH_CONT'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Competing interests',\n",
       "     'offset': 37521,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'title_1', 'section_type': 'COMP_INT'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'We have no competing interests.',\n",
       "     'offset': 37541,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'COMP_INT'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Funding',\n",
       "     'offset': 37573,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'title_1', 'section_type': 'ACK_FUND'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'This research was partially funded by the Croatian Science Foundation (Project ‘Molecular epidemiology of selected parasitic diseases of wildlife’, code 3421).',\n",
       "     'offset': 37581,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'paragraph', 'section_type': 'ACK_FUND'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'References',\n",
       "     'offset': 37745,\n",
       "     'relations': [],\n",
       "     'infons': {'type': 'title', 'section_type': 'REF'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Hybridization as an invasion of the genome',\n",
       "     'offset': 37756,\n",
       "     'relations': [],\n",
       "     'infons': {'lpage': '237',\n",
       "      'name_0': 'surname:Mallet;given-names:J',\n",
       "      'pub-id_pmid': '16701374',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '20',\n",
       "      'source': 'Trends Ecol. Evol.',\n",
       "      'fpage': '229',\n",
       "      'year': '2005',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'A genome-wide perspective on the evolutionary history of enigmatic wolf-like canids',\n",
       "     'offset': 37799,\n",
       "     'relations': [],\n",
       "     'infons': {'lpage': '1305',\n",
       "      'name_0': 'surname:VonHoldt;given-names:BM',\n",
       "      'pub-id_pmid': '21566151',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '21',\n",
       "      'source': 'Genome Res.',\n",
       "      'fpage': '1294',\n",
       "      'year': '2011',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Molecular genetics of the most endangered canid: the Ethiopian wolf Canis simensis',\n",
       "     'offset': 37883,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Applebaum;given-names:GD',\n",
       "      'name_3': 'surname:Roy;given-names:MS',\n",
       "      'name_0': 'surname:Gottelli;given-names:D',\n",
       "      'name_1': 'surname:Sillero-Zubiri;given-names:C',\n",
       "      'name_6': 'surname:Ostrander;given-names:EA',\n",
       "      'name_7': 'surname:Wayne;given-names:RK',\n",
       "      'name_4': 'surname:Girman;given-names:DJ',\n",
       "      'name_5': 'surname:Garcia-Moreno;given-names:J',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '3',\n",
       "      'pub-id_pmid': '7921357',\n",
       "      'source': 'Mol. Ecol.',\n",
       "      'fpage': '301',\n",
       "      'year': '1994',\n",
       "      'lpage': '312',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Detecting rare introgression of domestic dog genes into wild wolf (Canis lupus) populations by Bayesian admixture analyses of microsatellite variation',\n",
       "     'offset': 37966,\n",
       "     'relations': [],\n",
       "     'infons': {'lpage': '45',\n",
       "      'name_0': 'surname:Randi;given-names:E',\n",
       "      'name_1': 'surname:Lucchini;given-names:V',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '3',\n",
       "      'source': 'Conserv. Genet.',\n",
       "      'fpage': '31',\n",
       "      'year': '2002',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Canid hybridization: contemporary evolution in human-modified landscapes',\n",
       "     'offset': 38117,\n",
       "     'relations': [],\n",
       "     'infons': {'lpage': '2140',\n",
       "      'name_0': 'surname:Stronen;given-names:AV',\n",
       "      'pub-id_pmid': '23139873',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '2',\n",
       "      'source': 'Ecol. Evol.',\n",
       "      'fpage': '2128',\n",
       "      'year': '2012',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Multilocus detection of wolf x dog hybridization in Italy, and guidelines for marker selection',\n",
       "     'offset': 38190,\n",
       "     'relations': [],\n",
       "     'infons': {'name_0': 'surname:Randi;given-names:E',\n",
       "      'pub-id_pmid': '24466077',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '9',\n",
       "      'source': 'PLoS ONE',\n",
       "      'fpage': 'e86409',\n",
       "      'year': '2014',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Canis aureus Linnaeus, 1758',\n",
       "     'offset': 38285,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Sillero-Zubiri;given-names:C',\n",
       "      'name_3': 'surname:Hoffman;given-names:M',\n",
       "      'name_0': 'surname:Jhala;given-names:Y',\n",
       "      'name_1': 'surname:Moehlman;given-names:P',\n",
       "      'name_4': 'surname:Macdonald;given-names:D',\n",
       "      'section_type': 'REF',\n",
       "      'lpage': '161',\n",
       "      'source': 'Canids: foxes, wolves, jackals and dogs status survey and conservation action plan',\n",
       "      'fpage': '156',\n",
       "      'year': '2004',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Current status and distribution of golden jackals Canis aureus in Europe',\n",
       "     'offset': 38313,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Heltai;given-names:M',\n",
       "      'name_3': 'surname:Murariu;given-names:D',\n",
       "      'name_0': 'surname:Arnold;given-names:J',\n",
       "      'name_1': 'surname:Humer;given-names:A',\n",
       "      'name_4': 'surname:Spassov;given-names:N',\n",
       "      'name_5': 'surname:Hackländer;given-names:K',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '42',\n",
       "      'source': 'Mamm. Rev.',\n",
       "      'fpage': '1',\n",
       "      'year': '2012',\n",
       "      'lpage': '11',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Genome sequencing highlights the dynamic early history of dogs',\n",
       "     'offset': 38386,\n",
       "     'relations': [],\n",
       "     'infons': {'name_0': 'surname:Freedman;given-names:AH',\n",
       "      'pub-id_pmid': '24453982',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '10',\n",
       "      'source': 'PLoS Genet.',\n",
       "      'fpage': 'e1004016',\n",
       "      'year': '2014',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'The cryptic African wolf: Canis aureus lupaster is not a golden jackal and is not endemic to Egypt',\n",
       "     'offset': 38449,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Sillero-Zubiri;given-names:C',\n",
       "      'name_3': 'surname:Macdonald;given-names:DW',\n",
       "      'name_0': 'surname:Rueness;given-names:EK',\n",
       "      'name_1': 'surname:Asmyhr;given-names:MG',\n",
       "      'name_6': 'surname:Stenseth;given-names:NC',\n",
       "      'name_4': 'surname:Bekele;given-names:A',\n",
       "      'name_5': 'surname:Atickem;given-names:A',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '6',\n",
       "      'pub-id_pmid': '21298107',\n",
       "      'source': 'PLoS ONE',\n",
       "      'fpage': 'e16385',\n",
       "      'year': '2011',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Reviving the African wolf Canis lupus lupaster in north and west Africa: a mitochondrial lineage ranging more than 6,000\\u2009km wide',\n",
       "     'offset': 38548,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Benyacoub;given-names:S',\n",
       "      'name_3': 'surname:Abdelhamid;given-names:A',\n",
       "      'name_0': 'surname:Gaubert;given-names:P',\n",
       "      'name_1': 'surname:Bloch;given-names:C',\n",
       "      'name_6': 'surname:Couloux;given-names:A',\n",
       "      'name_7': 'surname:Dufour;given-names:S',\n",
       "      'name_4': 'surname:Pagani;given-names:P',\n",
       "      'name_5': 'surname:Djagoun;given-names:CAMS',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '7',\n",
       "      'pub-id_pmid': '22900047',\n",
       "      'source': 'PLoS ONE',\n",
       "      'fpage': 'e42740',\n",
       "      'year': '2012',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Records of the golden jackal (Canis aureus Linnaeus, 1758) in Hungary from 1800th until 2007, based on a literature survey',\n",
       "     'offset': 38679,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Szucs;given-names:E',\n",
       "      'name_3': 'surname:Heltai;given-names:M',\n",
       "      'name_0': 'surname:Tóth;given-names:T',\n",
       "      'name_1': 'surname:Krecsák;given-names:L',\n",
       "      'name_4': 'surname:Huszár;given-names:G',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '5',\n",
       "      'source': 'North West. J. Zool.',\n",
       "      'fpage': '386',\n",
       "      'year': '2009',\n",
       "      'lpage': '405',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Identification of recent hybridization between gray wolves and domesticated dogs by SNP genotyping',\n",
       "     'offset': 38802,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Earl;given-names:DA',\n",
       "      'name_3': 'surname:Parker;given-names:HG',\n",
       "      'name_0': 'surname:VonHoldt;given-names:BM',\n",
       "      'name_1': 'surname:Pollinger;given-names:JP',\n",
       "      'name_4': 'surname:Ostrander;given-names:EA',\n",
       "      'name_5': 'surname:Wayne;given-names:RK',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '24',\n",
       "      'pub-id_pmid': '23064780',\n",
       "      'source': 'Mamm. Genome',\n",
       "      'fpage': '80',\n",
       "      'year': '2012',\n",
       "      'lpage': '88',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Combined use of maternal, paternal and bi-parental genetic markers for the identification of wolf–dog hybrids',\n",
       "     'offset': 38901,\n",
       "     'relations': [],\n",
       "     'infons': {'lpage': '24',\n",
       "      'name_0': 'surname:Vilà;given-names:C',\n",
       "      'pub-id_pmid': '12522421',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '90',\n",
       "      'source': 'Heredity',\n",
       "      'fpage': '17',\n",
       "      'year': '2003',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Bucking the trend in wolf–dog hybridization: first evidence from Europe of hybridization between female dogs and /male wolves',\n",
       "     'offset': 39013,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Ozolins;given-names:J',\n",
       "      'name_3': 'surname:Krzywinski;given-names:A',\n",
       "      'name_0': 'surname:Hindrikson;given-names:M',\n",
       "      'name_1': 'surname:Männil;given-names:P',\n",
       "      'name_4': 'surname:Saarma;given-names:U',\n",
       "      'pub-id_pmid': '23056315',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '7',\n",
       "      'source': 'PLoS ONE',\n",
       "      'fpage': 'e46465',\n",
       "      'year': '2012',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Black coats in an admixed wolf × dog pack is melanism an indicator of hybridization in wolves?',\n",
       "     'offset': 39141,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Greco;given-names:C',\n",
       "      'name_3': 'surname:Galaverni;given-names:M',\n",
       "      'name_0': 'surname:Caniglia;given-names:R',\n",
       "      'name_1': 'surname:Fabbri;given-names:E',\n",
       "      'name_6': 'surname:Sforzi;given-names:A',\n",
       "      'name_7': 'surname:Randi;given-names:E',\n",
       "      'name_4': 'surname:Manghi;given-names:L',\n",
       "      'name_5': 'surname:Boitani;given-names:L',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '59',\n",
       "      'source': 'Eur. J. Wildl. Res.',\n",
       "      'fpage': '543',\n",
       "      'year': '2013',\n",
       "      'lpage': '555',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Hybridization between white-headed ducks and introduced ruddy ducks in Spain',\n",
       "     'offset': 39237,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Green;given-names:AJ',\n",
       "      'name_3': 'surname:Negro;given-names:JJ',\n",
       "      'name_0': 'surname:Muñoz-Fuentes;given-names:V',\n",
       "      'name_1': 'surname:Vilà;given-names:C',\n",
       "      'name_4': 'surname:Sorenson;given-names:MD',\n",
       "      'pub-id_pmid': '17257118',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '16',\n",
       "      'source': 'Mol. Ecol.',\n",
       "      'fpage': '629',\n",
       "      'year': '2007',\n",
       "      'lpage': '638',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Genetic variability, differentiation, and founder effect in golden jackals (Canis aureus) from Serbia as revealed by mitochondrial DNA and nuclear microsatellite loci',\n",
       "     'offset': 39314,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Kirschning;given-names:J',\n",
       "      'name_3': 'surname:Otto;given-names:M',\n",
       "      'name_0': 'surname:Zachos;given-names:FE',\n",
       "      'name_1': 'surname:Cirovic;given-names:D',\n",
       "      'name_6': 'surname:Honnen;given-names:AC',\n",
       "      'name_4': 'surname:Hartl;given-names:GB',\n",
       "      'name_5': 'surname:Petersen;given-names:B',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '47',\n",
       "      'pub-id_pmid': '19169806',\n",
       "      'source': 'Biochem. Genet.',\n",
       "      'fpage': '241',\n",
       "      'year': '2009',\n",
       "      'lpage': '250',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Genetic structure and expansion of golden jackals (Canis aureus) in the north-western distribution range (Croatia and eastern Italian Alps)',\n",
       "     'offset': 39481,\n",
       "     'relations': [],\n",
       "     'infons': {'lpage': '199',\n",
       "      'name_0': 'surname:Fabbri;given-names:E',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '15',\n",
       "      'source': 'Conserv. Genet.',\n",
       "      'fpage': '187',\n",
       "      'year': '2014',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'PCR-based Y chromosome marker for discriminating between golden jackal (Canis aureus) and domestic dog (Canis lupus familiaris) paternal ancestry',\n",
       "     'offset': 39621,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Gomerčić;given-names:T',\n",
       "      'name_3': 'surname:Arbanasić;given-names:H',\n",
       "      'name_0': 'surname:Galov;given-names:A',\n",
       "      'name_1': 'surname:Sindičić;given-names:M',\n",
       "      'name_6': 'surname:Florijančić;given-names:T',\n",
       "      'name_4': 'surname:Baburić;given-names:M',\n",
       "      'name_5': 'surname:Bošković;given-names:I',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '6',\n",
       "      'source': 'Conserv. Genet. Resour.',\n",
       "      'fpage': '275',\n",
       "      'year': '2014',\n",
       "      'lpage': '277',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Genetic characterization of populations of the golden jackal and the red fox in Israel',\n",
       "     'offset': 39767,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Dolev;given-names:A',\n",
       "      'name_3': 'surname:Boldo;given-names:A',\n",
       "      'name_0': 'surname:Magory Cohen;given-names:T',\n",
       "      'name_1': 'surname:King;given-names:R',\n",
       "      'name_4': 'surname:Lichter-Peled;given-names:A',\n",
       "      'name_5': 'surname:Kahila Bar-Gal;given-names:G',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '14',\n",
       "      'source': 'Conserv. Genet.',\n",
       "      'fpage': '55',\n",
       "      'year': '2013',\n",
       "      'lpage': '63',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Genetic analysis of hybridization and introgression between wild mongoose and brown lemurs',\n",
       "     'offset': 39854,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Curtis;given-names:DJ',\n",
       "      'name_3': 'surname:Nievergelt;given-names:CM',\n",
       "      'name_0': 'surname:Pastorini;given-names:J',\n",
       "      'name_1': 'surname:Zaramody;given-names:A',\n",
       "      'name_4': 'surname:Mundy;given-names:NI',\n",
       "      'pub-id_pmid': '19196458',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '9',\n",
       "      'source': 'BMC Evol. Biol.',\n",
       "      'fpage': '32',\n",
       "      'year': '2009',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'MHC studies in nonmodel vertebrates: what have we learned about natural selection in 15 years?',\n",
       "     'offset': 39945,\n",
       "     'relations': [],\n",
       "     'infons': {'lpage': '377',\n",
       "      'name_0': 'surname:Bernatchez;given-names:L',\n",
       "      'name_1': 'surname:Landry;given-names:C',\n",
       "      'pub-id_pmid': '14635837',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '16',\n",
       "      'source': 'J. Evol. Biol.',\n",
       "      'fpage': '363',\n",
       "      'year': '2003',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Applicability of major histocompatibility complex DRB1 alleles as markers to detect vertebrate hybridization: a case study from Iberian ibex x domestic goat in southern Spain',\n",
       "     'offset': 40040,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Rossi;given-names:L',\n",
       "      'name_3': 'surname:Sarasa;given-names:M',\n",
       "      'name_0': 'surname:Alasaad;given-names:S',\n",
       "      'name_1': 'surname:Fickel;given-names:J',\n",
       "      'name_6': 'surname:Soriguer;given-names:RC',\n",
       "      'name_4': 'surname:Benítez;given-names:CV',\n",
       "      'name_5': 'surname:Granados;given-names:JE',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '54',\n",
       "      'pub-id_pmid': '23006678',\n",
       "      'source': 'Acta Vet. Scand.',\n",
       "      'fpage': '56',\n",
       "      'year': '2012',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'MHC variability in an isolated wolf population in Italy',\n",
       "     'offset': 40215,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Fabbri;given-names:E',\n",
       "      'name_3': 'surname:Lapalombella;given-names:S',\n",
       "      'name_0': 'surname:Galaverni;given-names:M',\n",
       "      'name_1': 'surname:Caniglia;given-names:R',\n",
       "      'name_4': 'surname:Randi;given-names:E',\n",
       "      'pub-id_pmid': '23885092',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '104',\n",
       "      'source': 'J. Hered.',\n",
       "      'fpage': '601',\n",
       "      'year': '2013',\n",
       "      'lpage': '612',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Evolutionary genomics of dog domestication',\n",
       "     'offset': 40271,\n",
       "     'relations': [],\n",
       "     'infons': {'lpage': '18',\n",
       "      'name_0': 'surname:Wayne;given-names:RK',\n",
       "      'name_1': 'surname:Vonholdt;given-names:BM',\n",
       "      'pub-id_pmid': '22270221',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '23',\n",
       "      'source': 'Mamm. Genome',\n",
       "      'fpage': '3',\n",
       "      'year': '2012',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Molecular and evolutionary history of melanism in North American gray wolves',\n",
       "     'offset': 40314,\n",
       "     'relations': [],\n",
       "     'infons': {'lpage': '1343',\n",
       "      'name_0': 'surname:Anderson;given-names:TM',\n",
       "      'pub-id_pmid': '19197024',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '323',\n",
       "      'source': 'Science',\n",
       "      'fpage': '1339',\n",
       "      'year': '2009',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'The mitochondrial control region of Cervidae: evolutionary patterns and phylogenetic content',\n",
       "     'offset': 40391,\n",
       "     'relations': [],\n",
       "     'infons': {'lpage': '1166',\n",
       "      'name_0': 'surname:Douzery;given-names:E',\n",
       "      'name_1': 'surname:Randi;given-names:E',\n",
       "      'pub-id_pmid': '9364773',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '14',\n",
       "      'source': 'Mol. Biol. Evol.',\n",
       "      'fpage': '1154',\n",
       "      'year': '1997',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Mitochondrial DNA variability in Italian and east European wolves: detecting the consequences of small population size and hybridization',\n",
       "     'offset': 40484,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Christensen;given-names:MF',\n",
       "      'name_3': 'surname:Mucci;given-names:N',\n",
       "      'name_0': 'surname:Randi;given-names:E',\n",
       "      'name_1': 'surname:Lucchini;given-names:V',\n",
       "      'name_6': 'surname:Loeschcke;given-names:V',\n",
       "      'name_4': 'surname:Funk;given-names:SM',\n",
       "      'name_5': 'surname:Dolf;given-names:G',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '14',\n",
       "      'source': 'Conserv. Biol.',\n",
       "      'fpage': '464',\n",
       "      'year': '2000',\n",
       "      'lpage': '473',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'BioEdit: a user-friendly biological sequence alignment editor and analysis program for Windows 95/98/NT',\n",
       "     'offset': 40621,\n",
       "     'relations': [],\n",
       "     'infons': {'lpage': '98',\n",
       "      'name_0': 'surname:Hall;given-names:TA',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '41',\n",
       "      'source': 'Nucleic Acids Symp. Ser.',\n",
       "      'fpage': '95',\n",
       "      'year': '1999',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Mitochondrial DNA control region as a tool for species identification and distinction between wolves and dogs from Croatia',\n",
       "     'offset': 40725,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Galov;given-names:A',\n",
       "      'name_3': 'surname:Arbanasić;given-names:H',\n",
       "      'name_0': 'surname:Sindičić;given-names:M',\n",
       "      'name_1': 'surname:Gomerčić;given-names:T',\n",
       "      'name_6': 'surname:Huber;given-names:dj',\n",
       "      'name_4': 'surname:Kusak;given-names:J',\n",
       "      'name_5': 'surname:Slavica;given-names:A',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '81',\n",
       "      'source': 'Vet. Arh.',\n",
       "      'fpage': '249',\n",
       "      'year': '2011',\n",
       "      'lpage': '258',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'GenALEx 6.5: genetic analysis in Excel. Population genetic software for teaching and research-an update',\n",
       "     'offset': 40848,\n",
       "     'relations': [],\n",
       "     'infons': {'lpage': '2539',\n",
       "      'name_0': 'surname:Peakall;given-names:R',\n",
       "      'name_1': 'surname:Smouse;given-names:PE',\n",
       "      'pub-id_pmid': '22820204',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '28',\n",
       "      'source': 'Bioinformatics',\n",
       "      'fpage': '2537',\n",
       "      'year': '2012',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'GENALEX 6: genetic analysis in Excel. Population genetic software for teaching and research',\n",
       "     'offset': 40952,\n",
       "     'relations': [],\n",
       "     'infons': {'lpage': '295',\n",
       "      'name_0': 'surname:Peakall;given-names:R',\n",
       "      'name_1': 'surname:Smouse;given-names:PE',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '6',\n",
       "      'source': 'Mol. Ecol. Notes',\n",
       "      'fpage': '288',\n",
       "      'year': '2006',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Performing the exact test of Hardy–Weinberg proportion for multiple alleles',\n",
       "     'offset': 41044,\n",
       "     'relations': [],\n",
       "     'infons': {'lpage': '372',\n",
       "      'name_0': 'surname:Guo;given-names:SW',\n",
       "      'name_1': 'surname:Thompson;given-names:EA',\n",
       "      'pub-id_pmid': '1637966',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '48',\n",
       "      'source': 'Biometrics',\n",
       "      'fpage': '361',\n",
       "      'year': '1992',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'GENEPOP’007: a complete re-implementation of the GENEPOP software for Windows and Linux',\n",
       "     'offset': 41122,\n",
       "     'relations': [],\n",
       "     'infons': {'lpage': '106',\n",
       "      'name_0': 'surname:Rousset;given-names:F',\n",
       "      'pub-id_pmid': '21585727',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '8',\n",
       "      'source': 'Mol. Ecol. Resour.',\n",
       "      'fpage': '103',\n",
       "      'year': '2008',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Analysing tables of statistical tests',\n",
       "     'offset': 41212,\n",
       "     'relations': [],\n",
       "     'infons': {'lpage': '225',\n",
       "      'name_0': 'surname:Rice;given-names:WR',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '43',\n",
       "      'source': 'Evolution',\n",
       "      'fpage': '223',\n",
       "      'year': '1989',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': '',\n",
       "     'offset': 41250,\n",
       "     'relations': [],\n",
       "     'infons': {'source': 'Multivariate analysis in vegetation research',\n",
       "      'name_0': 'surname:Orloci;given-names:L',\n",
       "      'year': '1978',\n",
       "      'type': 'ref',\n",
       "      'section_type': 'REF'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Inference of population structure using multilocus genotype data',\n",
       "     'offset': 41251,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Donnelly;given-names:P',\n",
       "      'lpage': '959',\n",
       "      'name_0': 'surname:Pritchard;given-names:JK',\n",
       "      'name_1': 'surname:Stephens;given-names:M',\n",
       "      'pub-id_pmid': '10835412',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '155',\n",
       "      'source': 'Genetics',\n",
       "      'fpage': '945',\n",
       "      'year': '2000',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Inference of population structure using multilocus genotype data: linked loci and correlated allele frequencies',\n",
       "     'offset': 41316,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Pritchard;given-names:J',\n",
       "      'lpage': '1587',\n",
       "      'name_0': 'surname:Falush;given-names:D',\n",
       "      'name_1': 'surname:Stephens;given-names:M',\n",
       "      'pub-id_pmid': '12930761',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '164',\n",
       "      'source': 'Genetics',\n",
       "      'fpage': '1567',\n",
       "      'year': '2003',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Inference of population structure using multilocus genotype data: dominant markers and null alleles',\n",
       "     'offset': 41428,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Pritchard;given-names:JK',\n",
       "      'lpage': '578',\n",
       "      'name_0': 'surname:Falush;given-names:D',\n",
       "      'name_1': 'surname:Stephens;given-names:M',\n",
       "      'pub-id_pmid': '18784791',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '7',\n",
       "      'source': 'Mol. Ecol. Notes',\n",
       "      'fpage': '574',\n",
       "      'year': '2007',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Detecting the number of clusters of individuals using the software STRUCTURE: a simulation study',\n",
       "     'offset': 41528,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Goudet;given-names:J',\n",
       "      'lpage': '2620',\n",
       "      'name_0': 'surname:Evanno;given-names:G',\n",
       "      'name_1': 'surname:Regnaut;given-names:S',\n",
       "      'pub-id_pmid': '15969739',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '14',\n",
       "      'source': 'Mol. Ecol.',\n",
       "      'fpage': '2611',\n",
       "      'year': '2005',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'A model-based method for identifying species hybrids using multilocus data',\n",
       "     'offset': 41625,\n",
       "     'relations': [],\n",
       "     'infons': {'comment': '(doi:teststatistics;hybrids)',\n",
       "      'lpage': '1229',\n",
       "      'name_0': 'surname:Anderson;given-names:EC',\n",
       "      'name_1': 'surname:Thompson;given-names:EA',\n",
       "      'pub-id_pmid': '11901135',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '160',\n",
       "      'source': 'Genetics',\n",
       "      'fpage': '1217',\n",
       "      'year': '2002',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'HYBRIDLAB (version 1.0): a program for generating simulated hybrids from population samples',\n",
       "     'offset': 41700,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Kotlicki;given-names:P',\n",
       "      'lpage': '973',\n",
       "      'name_0': 'surname:Nielsen;given-names:EE',\n",
       "      'name_1': 'surname:Bach;given-names:LA',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '6',\n",
       "      'source': 'Mol. Ecol. Notes',\n",
       "      'fpage': '971',\n",
       "      'year': '2006',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'A -defensin mutation causes black coat color in domestic dogs',\n",
       "     'offset': 41792,\n",
       "     'relations': [],\n",
       "     'infons': {'lpage': '1423',\n",
       "      'name_0': 'surname:Candille;given-names:SI',\n",
       "      'pub-id_pmid': '17947548',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '318',\n",
       "      'source': 'Science',\n",
       "      'fpage': '1418',\n",
       "      'year': '2007',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'High-resolution characterization of the canine DLA-DRBI locus using reference strand-mediated conformational analysis',\n",
       "     'offset': 41854,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Fretwell;given-names:N',\n",
       "      'name_3': 'surname:Martin;given-names:AJ',\n",
       "      'name_0': 'surname:Kennedy;given-names:LJ',\n",
       "      'name_1': 'surname:Quarmby;given-names:S',\n",
       "      'name_6': 'surname:Ollier;given-names:WER',\n",
       "      'name_4': 'surname:Jones;given-names:PG',\n",
       "      'name_5': 'surname:Jones;given-names:CA',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '96',\n",
       "      'pub-id_pmid': '16251520',\n",
       "      'source': 'J. Hered.',\n",
       "      'fpage': '836',\n",
       "      'year': '2005',\n",
       "      'lpage': '842',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Molecular analysis of DLA-DRBB1 polymorphism',\n",
       "     'offset': 41972,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Works;given-names:JD',\n",
       "      'name_3': 'surname:Storb;given-names:R',\n",
       "      'name_0': 'surname:Wagner;given-names:JL',\n",
       "      'name_1': 'surname:Burnett;given-names:RC',\n",
       "      'pub-id_pmid': '8988538',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '48',\n",
       "      'source': 'Tissue Antigens',\n",
       "      'fpage': '554',\n",
       "      'year': '1996',\n",
       "      'lpage': '561',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Molecular analysis and polymorphism of the DLA-DQA gene',\n",
       "     'offset': 42017,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:DeRose;given-names:SA',\n",
       "      'name_3': 'surname:Storb;given-names:R',\n",
       "      'name_0': 'surname:Wagner;given-names:JL',\n",
       "      'name_1': 'surname:Burnett;given-names:RC',\n",
       "      'pub-id_pmid': '8896179',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '48',\n",
       "      'source': 'Tissue Antigens',\n",
       "      'fpage': '199',\n",
       "      'year': '1996',\n",
       "      'lpage': '204',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Molecular analysis and polymorphism of the DLA-DQB genes',\n",
       "     'offset': 42073,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Works;given-names:JD',\n",
       "      'name_3': 'surname:Storb;given-names:R',\n",
       "      'name_0': 'surname:Wagner;given-names:JL',\n",
       "      'name_1': 'surname:Hayes-Lattin;given-names:B',\n",
       "      'pub-id_pmid': '9802604',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '52',\n",
       "      'source': 'Tissue Antigens',\n",
       "      'fpage': '242',\n",
       "      'year': '1998',\n",
       "      'lpage': '250',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'DLA-DRB, DQAI, and DQBI alleles and haplotypes in North American gray wolves',\n",
       "     'offset': 42130,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Barnes;given-names:A',\n",
       "      'name_3': 'surname:Carmichael;given-names:LE',\n",
       "      'name_0': 'surname:Kennedy;given-names:LJ',\n",
       "      'name_1': 'surname:Angles;given-names:JM',\n",
       "      'name_6': 'surname:Happ;given-names:GM',\n",
       "      'name_4': 'surname:Radford;given-names:AD',\n",
       "      'name_5': 'surname:Ollier;given-names:WER',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '98',\n",
       "      'pub-id_pmid': '17611255',\n",
       "      'source': 'J. Hered.',\n",
       "      'fpage': '491',\n",
       "      'year': '2007',\n",
       "      'lpage': '499',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Extensive polymorphism and evidence of selection pressure on major histocompatibility complex DLA-DRB1, DQA1 and DQB1 class II genes in Croatian grey wolves',\n",
       "     'offset': 42207,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Kusak;given-names:J',\n",
       "      'name_3': 'surname:Gomerčić;given-names:T',\n",
       "      'name_0': 'surname:Arbanasić;given-names:H',\n",
       "      'name_1': 'surname:Huber;given-names:D',\n",
       "      'name_4': 'surname:Hrenović;given-names:J',\n",
       "      'name_5': 'surname:Galov;given-names:A',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '81',\n",
       "      'pub-id_pmid': '23134500',\n",
       "      'source': 'Tissue Antigens',\n",
       "      'fpage': '19',\n",
       "      'year': '2013',\n",
       "      'lpage': '27',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Extensive interbreed, but minimal intrabreed, variation of DLA class II alleles and haplotypes in dogs',\n",
       "     'offset': 42364,\n",
       "     'relations': [],\n",
       "     'infons': {'lpage': '204',\n",
       "      'name_0': 'surname:Kennedy;given-names:LJ',\n",
       "      'pub-id_pmid': '12074709',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '59',\n",
       "      'source': 'Tissue Antigens',\n",
       "      'fpage': '194',\n",
       "      'year': '2002',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Arlequin (version 3.0): an integrated software package for population genetics data analysis',\n",
       "     'offset': 42467,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Schneider;given-names:S',\n",
       "      'lpage': '50',\n",
       "      'name_0': 'surname:Excoffier;given-names:L',\n",
       "      'name_1': 'surname:Laval;given-names:G',\n",
       "      'pub-id_pmid': '19325852',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '1',\n",
       "      'source': 'Evol. Bioinform. Online',\n",
       "      'fpage': '47',\n",
       "      'year': '2005',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Molecular analysis of hybridisation between wild and domestic cats (Felis silvestris) in Portugal: implications for conservation',\n",
       "     'offset': 42560,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Randi;given-names:E',\n",
       "      'name_3': 'surname:Ferrand;given-names:N',\n",
       "      'name_0': 'surname:Oliveira;given-names:R',\n",
       "      'name_1': 'surname:Godinho;given-names:R',\n",
       "      'name_4': 'surname:Alves;given-names:PC',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '9',\n",
       "      'source': 'Conserv. Genet.',\n",
       "      'fpage': '1',\n",
       "      'year': '2008',\n",
       "      'lpage': '11',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Canine DLA diversity: 1. New alleles and haplotypes',\n",
       "     'offset': 42689,\n",
       "     'relations': [],\n",
       "     'infons': {'lpage': '288',\n",
       "      'name_0': 'surname:Kennedy;given-names:LJ',\n",
       "      'pub-id_pmid': '17445218',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '69',\n",
       "      'source': 'Tissue Antigens',\n",
       "      'fpage': '272',\n",
       "      'year': '2007',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Hybridization and speciation',\n",
       "     'offset': 42741,\n",
       "     'relations': [],\n",
       "     'infons': {'lpage': '246',\n",
       "      'name_0': 'surname:Abbott;given-names:R',\n",
       "      'pub-id_pmid': '23323997',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '26',\n",
       "      'source': 'J. Evol. Biol.',\n",
       "      'fpage': '229',\n",
       "      'year': '2013',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Unregulated hunting and genetic recovery from a severe population decline: the cautionary case of Bulgarian wolves',\n",
       "     'offset': 42770,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Dabrowski;given-names:MJ',\n",
       "      'name_3': 'surname:Czarnomska;given-names:SD',\n",
       "      'name_0': 'surname:Moura;given-names:AE',\n",
       "      'name_1': 'surname:Tsingarska;given-names:E',\n",
       "      'name_4': 'surname:Jedrzejewska;given-names:B',\n",
       "      'name_5': 'surname:Pilot;given-names:M',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '15',\n",
       "      'source': 'Conserv. Genet.',\n",
       "      'fpage': '405',\n",
       "      'year': '2014',\n",
       "      'lpage': '417',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Late autumn trophic flexibility of the golden jackal Canis aureus',\n",
       "     'offset': 42885,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Dolev;given-names:A',\n",
       "      'name_3': 'surname:Bino;given-names:G',\n",
       "      'name_0': 'surname:Lanszki;given-names:J',\n",
       "      'name_1': 'surname:Giannatos;given-names:G',\n",
       "      'name_4': 'surname:Heltai;given-names:M',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '55',\n",
       "      'source': 'Acta Theriol. (Warsz)',\n",
       "      'fpage': '361',\n",
       "      'year': '2010',\n",
       "      'lpage': '370',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Diet composition of golden jackals in Israel',\n",
       "     'offset': 42951,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Manor;given-names:R',\n",
       "      'lpage': '118',\n",
       "      'name_0': 'surname:Borkowski;given-names:J',\n",
       "      'name_1': 'surname:Zalewski;given-names:A',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '48',\n",
       "      'source': 'Ann. Zool. Fennici',\n",
       "      'fpage': '108',\n",
       "      'year': '2011',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': '',\n",
       "     'offset': 42996,\n",
       "     'relations': [],\n",
       "     'infons': {'comment': 'Golden jackal (Canis aureus) distribution in Croatia. In Abstracts of VI European Congress of Mammalogy, 19-23 July 2011, Université P. et M. Curie – Muséum national d’Histoire naturelle, Paris, France (ed. C Denys)',\n",
       "      'name_2': 'surname:Sindičić;given-names:M',\n",
       "      'name_0': 'surname:Selanec;given-names:I',\n",
       "      'name_1': 'surname:Lauš;given-names:B',\n",
       "      'section_type': 'REF',\n",
       "      'year': '2011',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Intense harvesting of eastern wolves facilitated hybridization with coyotes',\n",
       "     'offset': 42997,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Row;given-names:JR',\n",
       "      'name_3': 'surname:Patterson;given-names:BR',\n",
       "      'name_0': 'surname:Rutledge;given-names:LY',\n",
       "      'name_1': 'surname:White;given-names:BN',\n",
       "      'pub-id_pmid': '22408723',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '2',\n",
       "      'source': 'Ecol. Evol.',\n",
       "      'fpage': '19',\n",
       "      'year': '2012',\n",
       "      'lpage': '33',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Genetic evidence for multiple events of hybridization between wolves and domestic dogs in the Iberian Peninsula',\n",
       "     'offset': 43073,\n",
       "     'relations': [],\n",
       "     'infons': {'lpage': '5166',\n",
       "      'name_0': 'surname:Godinho;given-names:R',\n",
       "      'pub-id_pmid': '22066758',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '20',\n",
       "      'source': 'Mol. Ecol.',\n",
       "      'fpage': '5154',\n",
       "      'year': '2011',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Single nucleotide polymorphism (SNP) variation of wolves (Canis lupus) in Southeast Alaska and comparison with wolves, dogs, and coyotes in North America',\n",
       "     'offset': 43185,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Bannasch;given-names:DL',\n",
       "      'name_3': 'surname:Oberbauer;given-names:AM',\n",
       "      'name_0': 'surname:Cronin;given-names:MA',\n",
       "      'name_1': 'surname:Canovas;given-names:A',\n",
       "      'name_4': 'surname:Medrano;given-names:JF',\n",
       "      'pub-id_pmid': '25429025',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '106',\n",
       "      'source': 'J. Hered.',\n",
       "      'fpage': '26',\n",
       "      'year': '2014',\n",
       "      'lpage': '36',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Complex population structure in African village dogs and its implications for inferring dog domestication history',\n",
       "     'offset': 43339,\n",
       "     'relations': [],\n",
       "     'infons': {'lpage': '13\\u2009908',\n",
       "      'name_0': 'surname:Boyko;given-names:AR',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '106',\n",
       "      'source': 'Proc. Natl Acad. Sci. USA',\n",
       "      'fpage': '13\\u2009903',\n",
       "      'year': '2009',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'LUPA: a European initiative taking advantage of the canine genome architecture for unravelling complex disorders in both human and dogs',\n",
       "     'offset': 43453,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:André;given-names:C',\n",
       "      'name_3': 'surname:Fredholm;given-names:M',\n",
       "      'name_0': 'surname:Lequarré;given-names:AS',\n",
       "      'name_1': 'surname:Andersson;given-names:L',\n",
       "      'name_6': 'surname:Lohi;given-names:H',\n",
       "      'name_7': 'surname:Lindblad-Toh;given-names:K',\n",
       "      'name_4': 'surname:Hitte;given-names:C',\n",
       "      'name_5': 'surname:Leeb;given-names:T',\n",
       "      'name_8': 'surname:Georges;given-names:M',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '189',\n",
       "      'pub-id_pmid': '21752675',\n",
       "      'source': 'Vet. J.',\n",
       "      'fpage': '155',\n",
       "      'year': '2011',\n",
       "      'lpage': '159',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Identification of genomic regions associated with phenotypic variation between dog breeds using selection mapping',\n",
       "     'offset': 43589,\n",
       "     'relations': [],\n",
       "     'infons': {'name_0': 'surname:Vaysse;given-names:A',\n",
       "      'pub-id_pmid': '22022279',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '7',\n",
       "      'source': 'PLoS Genet.',\n",
       "      'fpage': 'e1002316',\n",
       "      'year': '2011',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'First record of a melanistic golden jackal (Canis aureus, Canidae) from Turkey',\n",
       "     'offset': 43703,\n",
       "     'relations': [],\n",
       "     'infons': {'lpage': '222',\n",
       "      'name_0': 'surname:Ambarli;given-names:H',\n",
       "      'name_1': 'surname:Bilgin;given-names:CC',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '77',\n",
       "      'source': 'Mammalia',\n",
       "      'fpage': '219',\n",
       "      'year': '2013',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Perspective: detecting adaptive molecular polymorphism: lessons from the MHC',\n",
       "     'offset': 43782,\n",
       "     'relations': [],\n",
       "     'infons': {'lpage': '1722',\n",
       "      'name_0': 'surname:Garrigan;given-names:D',\n",
       "      'name_1': 'surname:Hedrick;given-names:PW',\n",
       "      'pub-id_pmid': '14503614',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '57',\n",
       "      'source': 'Evolution',\n",
       "      'fpage': '1707',\n",
       "      'year': '2003',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'The evolutionary ecology of the major histocompatibility complex',\n",
       "     'offset': 43859,\n",
       "     'relations': [],\n",
       "     'infons': {'lpage': '21',\n",
       "      'name_0': 'surname:Piertney;given-names:SB',\n",
       "      'name_1': 'surname:Oliver;given-names:MK',\n",
       "      'pub-id_pmid': '16094301',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '96',\n",
       "      'source': 'Heredity',\n",
       "      'fpage': '7',\n",
       "      'year': '2006',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Repeated adaptive introgression at a gene under multiallelic balancing selection',\n",
       "     'offset': 43924,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Schierup;given-names:MH',\n",
       "      'name_3': 'surname:Vekemans;given-names:X',\n",
       "      'name_0': 'surname:Castric;given-names:V',\n",
       "      'name_1': 'surname:Bechsgaard;given-names:J',\n",
       "      'pub-id_pmid': '18769722',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '4',\n",
       "      'source': 'PLoS Genet.',\n",
       "      'fpage': 'e1000168',\n",
       "      'year': '2008',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Adaptive introgression in animals: examples and comparison to new mutation and standing variation as sources of adaptive variation',\n",
       "     'offset': 44005,\n",
       "     'relations': [],\n",
       "     'infons': {'lpage': '4618',\n",
       "      'name_0': 'surname:Hedrick;given-names:PW',\n",
       "      'pub-id_pmid': '23906376',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '22',\n",
       "      'source': 'Mol. Ecol.',\n",
       "      'fpage': '4606',\n",
       "      'year': '2013',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Interspecific hybridization increases MHC class II diversity in two sister species of newts',\n",
       "     'offset': 44136,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Radwan;given-names:J',\n",
       "      'name_3': 'surname:Babik;given-names:W',\n",
       "      'name_0': 'surname:Nadachowska-Brzyska;given-names:K',\n",
       "      'name_1': 'surname:Zieliński;given-names:P',\n",
       "      'pub-id_pmid': '22066802',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '21',\n",
       "      'source': 'Mol. Ecol.',\n",
       "      'fpage': '887',\n",
       "      'year': '2012',\n",
       "      'lpage': '906',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'The shaping of modern human immune systems by multiregional admixture with archaic humans',\n",
       "     'offset': 44228,\n",
       "     'relations': [],\n",
       "     'infons': {'lpage': '94',\n",
       "      'name_0': 'surname:Abi-Rached;given-names:L',\n",
       "      'pub-id_pmid': '21868630',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '334',\n",
       "      'source': 'Science',\n",
       "      'fpage': '89',\n",
       "      'year': '2011',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'MHC-based mate choice combines good genes and maintenance of MHC polymorphism',\n",
       "     'offset': 44318,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Lenz;given-names:TL',\n",
       "      'name_3': 'surname:Kalbe;given-names:M',\n",
       "      'name_0': 'surname:Eizaguirre;given-names:C',\n",
       "      'name_1': 'surname:Yeates;given-names:SE',\n",
       "      'name_4': 'surname:Milinski;given-names:M',\n",
       "      'pub-id_pmid': '19523111',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '18',\n",
       "      'source': 'Mol. Ecol.',\n",
       "      'fpage': '3316',\n",
       "      'year': '2009',\n",
       "      'lpage': '3329',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'New(t)s and views from hybridizing MHC genes: introgression rather than trans-species polymorphism may shape allelic repertoires',\n",
       "     'offset': 44396,\n",
       "     'relations': [],\n",
       "     'infons': {'lpage': '781',\n",
       "      'name_0': 'surname:Wegner;given-names:KM',\n",
       "      'name_1': 'surname:Eizaguirre;given-names:C',\n",
       "      'pub-id_pmid': '22268859',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '21',\n",
       "      'source': 'Mol. Ecol.',\n",
       "      'fpage': '779',\n",
       "      'year': '2012',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Detecting hybridization between wild species and their domesticated relatives',\n",
       "     'offset': 44525,\n",
       "     'relations': [],\n",
       "     'infons': {'lpage': '293',\n",
       "      'name_0': 'surname:Randi;given-names:E',\n",
       "      'pub-id_pmid': '18173502',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '17',\n",
       "      'source': 'Mol. Ecol.',\n",
       "      'fpage': '285',\n",
       "      'year': '2008',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []},\n",
       "    {'text': 'Canine parvovirus enteritis, canine distemper, and major histocompatibility complex genetic variation in Mexican wolves',\n",
       "     'offset': 44603,\n",
       "     'relations': [],\n",
       "     'infons': {'name_2': 'surname:Buchanan;given-names:C',\n",
       "      'lpage': '913',\n",
       "      'name_0': 'surname:Hedrick;given-names:PW',\n",
       "      'name_1': 'surname:Lee;given-names:RN',\n",
       "      'pub-id_pmid': '14733289',\n",
       "      'section_type': 'REF',\n",
       "      'volume': '39',\n",
       "      'source': 'J. Wildl. Dis.',\n",
       "      'fpage': '909',\n",
       "      'year': '2003',\n",
       "      'type': 'ref'},\n",
       "     'sentences': [],\n",
       "     'annotations': []}],\n",
       "   'infons': {'license': 'CC BY'},\n",
       "   'id': '4807452',\n",
       "   'relations': []}],\n",
       " 'key': 'pmc.key'}"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "outdat[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "pmid = df_targets.pmcid.iloc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + pmid + '/unicode'\n",
    "r = requests.get(pth)\n",
    "r_out = r.json()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'PMC5139083'"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pmid"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'5139083'"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#r_out['documents'][0]['passages'][0]['infons']['year']\n",
    "r_out['documents'][0]['id']# that's the pmcid"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1996      3558\n",
       "1994      3383\n",
       "1995      3292\n",
       "1993      3168\n",
       "1992      2730\n",
       "1991      2724\n",
       "1990      2675\n",
       "1989      2554\n",
       "2015      2220\n",
       "1988      2170\n",
       "1899      2120\n",
       "1983      2101\n",
       "1898      2083\n",
       "1910.0    2073\n",
       "1894      2069\n",
       "1982      2058\n",
       "1920.0    2036\n",
       "1902.0    2034\n",
       "1986      2026\n",
       "1987      2019\n",
       "1904      2004\n",
       "1985      1959\n",
       "1911      1959\n",
       "1984      1948\n",
       "1909.0    1942\n",
       "Name: yr_fixed, dtype: int64"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "yr_counts = frame.yr_fixed.value_counts()\n",
    "yr_counts.iloc[25:50]"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "ac_knowl",
   "language": "python",
   "name": "ac_knowl"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
