{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from SPARQLWrapper import SPARQLWrapper, JSON\n",
    "import pandas as pd\n",
    "import urllib\n",
    "import requests\n",
    "import re"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Ongoing problems:\n",
    "I'm not exactly sure what was causing some of my previous sparql queries to fail, but I think I have pulled down basic information for all the papers in the ma graph database. I'm not sure how useful this is going to be. I'm now getting some details from around 3M pubmed papers that are hosted by a research group at [NCBI](https://www.ncbi.nlm.nih.gov/research/bionlp/APIs/BioC-PMC/). I would ideally like to filter to just papers after 2008 or so (when github was founded), but I'm not sure that is possible. So instead, I'm doing one pass through all the listed files collecting minimal information. Then, any further data needed can be queried on the subset that I'm interested in.\n",
    "\n",
    "## Data description\n",
    "Here, I'm pulling data out of a sparql database that includes information from microsoft academic knowledge graph. The website for this data is here: http://ma-graph.org/\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "sparql = SPARQLWrapper('http://ma-graph.org/sparql')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [],
   "source": [
    "sparql.setQuery(\"\"\"\n",
    "PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>\n",
    "PREFIX magp: <http://ma-graph.org/property/>\n",
    "PREFIX mag: <http://ma-graph.org/>\n",
    "PREFIX dcterms: <http://purl.org/dc/terms/>\n",
    "PREFIX foaf: <http://xmlns.com/foaf/0.1/>\n",
    "PREFIX fabio: <http://purl.org/spar/fabio/>\n",
    "PREFIX org: <http://www.w3.org/ns/org#>\n",
    "PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>\n",
    "PREFIX cito: <http://purl.org/spar/cito/>\n",
    "PREFIX c4o: <http://purl.org/spar/c4o/>\n",
    "PREFIX prism: <http://prismstandard.org/namespaces/1.2/basic/>\n",
    "PREFIX datacite: <http://purl.org/spar/datacite/>\n",
    "\n",
    "SELECT ?nihauthname ?otherauths ?papertitle ?discname ?dt ?doi ?abst\n",
    "WHERE {\n",
    "?inst rdf:type <http://ma-graph.org/class/Affiliation> .\n",
    "?inst foaf:name \"National Institutes of Health\"^^xsd:string .\n",
    "?auth org:memberOf ?inst .\n",
    "?paper dcterms:creator ?auth .\n",
    "?auth foaf:name ?nihauthname .\n",
    "?paper dcterms:creator ?allauths .\n",
    "?allauths foaf:name ?otherauths .\n",
    "?paper dcterms:title ?papertitle .\n",
    "?paper prism:publicationDate ?dt .\n",
    "?paper fabio:hasDiscipline ?disc .\n",
    "?paper datacite:doi ?doi .\n",
    "?paper dcterms:abstract ?abst .\n",
    "?disc foaf:name ?discname .\n",
    "}\n",
    "\"\"\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [],
   "source": [
    "#?paper cito:hasCitedEntity ?cit .\n",
    "#?cit c4o:hasContext ?citcont ."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [],
   "source": [
    "sparql.setReturnFormat(JSON)\n",
    "results = sparql.query().convert()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1048576"
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(results['results']['bindings'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'nihauthname': {'type': 'typed-literal',\n",
       "  'datatype': 'http://www.w3.org/2001/XMLSchema#string',\n",
       "  'value': 'Scott R. Budihas'},\n",
       " 'otherauths': {'type': 'typed-literal',\n",
       "  'datatype': 'http://www.w3.org/2001/XMLSchema#string',\n",
       "  'value': 'Michael A. Parniak'},\n",
       " 'papertitle': {'type': 'typed-literal',\n",
       "  'datatype': 'http://www.w3.org/2001/XMLSchema#string',\n",
       "  'value': 'A capillary electrophoretic assay for ribonuclease H activity.'},\n",
       " 'discname': {'type': 'typed-literal',\n",
       "  'datatype': 'http://www.w3.org/2001/XMLSchema#string',\n",
       "  'value': 'Biochemistry'},\n",
       " 'dt': {'type': 'typed-literal',\n",
       "  'datatype': 'http://www.w3.org/2001/XMLSchema#date',\n",
       "  'value': '2004-08-01'},\n",
       " 'doi': {'type': 'typed-literal',\n",
       "  'datatype': 'http://www.w3.org/2001/XMLSchema#string',\n",
       "  'value': '10.1016/j.ab.2004.05.017'},\n",
       " 'abst': {'type': 'typed-literal',\n",
       "  'datatype': 'http://www.w3.org/2001/XMLSchema#string',\n",
       "  'value': 'Abstract A capillary electrophoretic assay was developed to measure the ribonuclease (RNase) H activity of human immunodeficiency virus (HIV) type 1 reverse transcriptase. Cleavage of a fluorescein-labeled RNA\\x13DNA heteroduplex was monitored by capillary electrophoresis. This new assay was used as a secondary assay to confirm hits from a high-throughput screening program. Since autofluorescent compounds in samples migrated differently from both substrate and product in most cases, the assay was extremely robust for assaying enzymatic inhibition of such samples, in contrast to a simple well-based approach. The assay was broadly applicable to other RNases H, specifically those from human, Escherichia coli , and HIV-2, although product profiles varied for each enzyme.'}}"
      ]
     },
     "execution_count": 106,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "results['results']['bindings'][10000]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [],
   "source": [
    "outdat = []\n",
    "for i in results['results']['bindings']:\n",
    "    outdat.append([i['papertitle']['value'], i['authname']['value'], \n",
    "                   i['discname']['value'], i['dt']['value'], i['doi']['value']])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [],
   "source": [
    "dat = pd.DataFrame(outdat, columns = ['title', 'author', 'field', 'date', 'doi'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>author</th>\n",
       "      <th>field</th>\n",
       "      <th>date</th>\n",
       "      <th>doi</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Molecular Signature of the Ebola Virus Associa...</td>\n",
       "      <td>Concetta Castilletti</td>\n",
       "      <td>Ebola virus</td>\n",
       "      <td>2015-10-29</td>\n",
       "      <td>10.1128/genomeA.01093-15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Molecular Signature of the Ebola Virus Associa...</td>\n",
       "      <td>Concetta Castilletti</td>\n",
       "      <td>Virology</td>\n",
       "      <td>2015-10-29</td>\n",
       "      <td>10.1128/genomeA.01093-15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Molecular Signature of the Ebola Virus Associa...</td>\n",
       "      <td>Concetta Castilletti</td>\n",
       "      <td>Subclade</td>\n",
       "      <td>2015-10-29</td>\n",
       "      <td>10.1128/genomeA.01093-15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Molecular Signature of the Ebola Virus Associa...</td>\n",
       "      <td>Concetta Castilletti</td>\n",
       "      <td>Outbreak</td>\n",
       "      <td>2015-10-29</td>\n",
       "      <td>10.1128/genomeA.01093-15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Molecular Signature of the Ebola Virus Associa...</td>\n",
       "      <td>Concetta Castilletti</td>\n",
       "      <td>Bioinformatics</td>\n",
       "      <td>2015-10-29</td>\n",
       "      <td>10.1128/genomeA.01093-15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Molecular Signature of the Ebola Virus Associa...</td>\n",
       "      <td>Concetta Castilletti</td>\n",
       "      <td>Biology</td>\n",
       "      <td>2015-10-29</td>\n",
       "      <td>10.1128/genomeA.01093-15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160224</th>\n",
       "      <td>Molecular Signature of the Ebola Virus Associa...</td>\n",
       "      <td>Silvia Meschi</td>\n",
       "      <td>Ebola virus</td>\n",
       "      <td>2015-10-29</td>\n",
       "      <td>10.1128/genomeA.01093-15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160225</th>\n",
       "      <td>Molecular Signature of the Ebola Virus Associa...</td>\n",
       "      <td>Silvia Meschi</td>\n",
       "      <td>Virology</td>\n",
       "      <td>2015-10-29</td>\n",
       "      <td>10.1128/genomeA.01093-15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160226</th>\n",
       "      <td>Molecular Signature of the Ebola Virus Associa...</td>\n",
       "      <td>Silvia Meschi</td>\n",
       "      <td>Subclade</td>\n",
       "      <td>2015-10-29</td>\n",
       "      <td>10.1128/genomeA.01093-15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160227</th>\n",
       "      <td>Molecular Signature of the Ebola Virus Associa...</td>\n",
       "      <td>Silvia Meschi</td>\n",
       "      <td>Outbreak</td>\n",
       "      <td>2015-10-29</td>\n",
       "      <td>10.1128/genomeA.01093-15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160228</th>\n",
       "      <td>Molecular Signature of the Ebola Virus Associa...</td>\n",
       "      <td>Silvia Meschi</td>\n",
       "      <td>Bioinformatics</td>\n",
       "      <td>2015-10-29</td>\n",
       "      <td>10.1128/genomeA.01093-15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160229</th>\n",
       "      <td>Molecular Signature of the Ebola Virus Associa...</td>\n",
       "      <td>Silvia Meschi</td>\n",
       "      <td>Biology</td>\n",
       "      <td>2015-10-29</td>\n",
       "      <td>10.1128/genomeA.01093-15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>391311</th>\n",
       "      <td>Molecular Signature of the Ebola Virus Associa...</td>\n",
       "      <td>Antonino Di Caro</td>\n",
       "      <td>Ebola virus</td>\n",
       "      <td>2015-10-29</td>\n",
       "      <td>10.1128/genomeA.01093-15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>391312</th>\n",
       "      <td>Molecular Signature of the Ebola Virus Associa...</td>\n",
       "      <td>Antonino Di Caro</td>\n",
       "      <td>Virology</td>\n",
       "      <td>2015-10-29</td>\n",
       "      <td>10.1128/genomeA.01093-15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>391313</th>\n",
       "      <td>Molecular Signature of the Ebola Virus Associa...</td>\n",
       "      <td>Antonino Di Caro</td>\n",
       "      <td>Subclade</td>\n",
       "      <td>2015-10-29</td>\n",
       "      <td>10.1128/genomeA.01093-15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>391314</th>\n",
       "      <td>Molecular Signature of the Ebola Virus Associa...</td>\n",
       "      <td>Antonino Di Caro</td>\n",
       "      <td>Outbreak</td>\n",
       "      <td>2015-10-29</td>\n",
       "      <td>10.1128/genomeA.01093-15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>391315</th>\n",
       "      <td>Molecular Signature of the Ebola Virus Associa...</td>\n",
       "      <td>Antonino Di Caro</td>\n",
       "      <td>Bioinformatics</td>\n",
       "      <td>2015-10-29</td>\n",
       "      <td>10.1128/genomeA.01093-15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>391316</th>\n",
       "      <td>Molecular Signature of the Ebola Virus Associa...</td>\n",
       "      <td>Antonino Di Caro</td>\n",
       "      <td>Biology</td>\n",
       "      <td>2015-10-29</td>\n",
       "      <td>10.1128/genomeA.01093-15</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    title  \\\n",
       "0       Molecular Signature of the Ebola Virus Associa...   \n",
       "1       Molecular Signature of the Ebola Virus Associa...   \n",
       "2       Molecular Signature of the Ebola Virus Associa...   \n",
       "3       Molecular Signature of the Ebola Virus Associa...   \n",
       "4       Molecular Signature of the Ebola Virus Associa...   \n",
       "5       Molecular Signature of the Ebola Virus Associa...   \n",
       "160224  Molecular Signature of the Ebola Virus Associa...   \n",
       "160225  Molecular Signature of the Ebola Virus Associa...   \n",
       "160226  Molecular Signature of the Ebola Virus Associa...   \n",
       "160227  Molecular Signature of the Ebola Virus Associa...   \n",
       "160228  Molecular Signature of the Ebola Virus Associa...   \n",
       "160229  Molecular Signature of the Ebola Virus Associa...   \n",
       "391311  Molecular Signature of the Ebola Virus Associa...   \n",
       "391312  Molecular Signature of the Ebola Virus Associa...   \n",
       "391313  Molecular Signature of the Ebola Virus Associa...   \n",
       "391314  Molecular Signature of the Ebola Virus Associa...   \n",
       "391315  Molecular Signature of the Ebola Virus Associa...   \n",
       "391316  Molecular Signature of the Ebola Virus Associa...   \n",
       "\n",
       "                      author           field        date  \\\n",
       "0       Concetta Castilletti     Ebola virus  2015-10-29   \n",
       "1       Concetta Castilletti        Virology  2015-10-29   \n",
       "2       Concetta Castilletti        Subclade  2015-10-29   \n",
       "3       Concetta Castilletti        Outbreak  2015-10-29   \n",
       "4       Concetta Castilletti  Bioinformatics  2015-10-29   \n",
       "5       Concetta Castilletti         Biology  2015-10-29   \n",
       "160224         Silvia Meschi     Ebola virus  2015-10-29   \n",
       "160225         Silvia Meschi        Virology  2015-10-29   \n",
       "160226         Silvia Meschi        Subclade  2015-10-29   \n",
       "160227         Silvia Meschi        Outbreak  2015-10-29   \n",
       "160228         Silvia Meschi  Bioinformatics  2015-10-29   \n",
       "160229         Silvia Meschi         Biology  2015-10-29   \n",
       "391311      Antonino Di Caro     Ebola virus  2015-10-29   \n",
       "391312      Antonino Di Caro        Virology  2015-10-29   \n",
       "391313      Antonino Di Caro        Subclade  2015-10-29   \n",
       "391314      Antonino Di Caro        Outbreak  2015-10-29   \n",
       "391315      Antonino Di Caro  Bioinformatics  2015-10-29   \n",
       "391316      Antonino Di Caro         Biology  2015-10-29   \n",
       "\n",
       "                             doi  \n",
       "0       10.1128/genomeA.01093-15  \n",
       "1       10.1128/genomeA.01093-15  \n",
       "2       10.1128/genomeA.01093-15  \n",
       "3       10.1128/genomeA.01093-15  \n",
       "4       10.1128/genomeA.01093-15  \n",
       "5       10.1128/genomeA.01093-15  \n",
       "160224  10.1128/genomeA.01093-15  \n",
       "160225  10.1128/genomeA.01093-15  \n",
       "160226  10.1128/genomeA.01093-15  \n",
       "160227  10.1128/genomeA.01093-15  \n",
       "160228  10.1128/genomeA.01093-15  \n",
       "160229  10.1128/genomeA.01093-15  \n",
       "391311  10.1128/genomeA.01093-15  \n",
       "391312  10.1128/genomeA.01093-15  \n",
       "391313  10.1128/genomeA.01093-15  \n",
       "391314  10.1128/genomeA.01093-15  \n",
       "391315  10.1128/genomeA.01093-15  \n",
       "391316  10.1128/genomeA.01093-15  "
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dat[dat.doi=='10.1128/genomeA.01093-15']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>field</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Biology</th>\n",
       "      <td>56725</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Medicine</th>\n",
       "      <td>41832</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Molecular biology</th>\n",
       "      <td>23265</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Immunology</th>\n",
       "      <td>22135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Internal medicine</th>\n",
       "      <td>19022</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Virology</th>\n",
       "      <td>15039</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Biochemistry</th>\n",
       "      <td>14901</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Endocrinology</th>\n",
       "      <td>13882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genetics</th>\n",
       "      <td>13696</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Cancer</th>\n",
       "      <td>9490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Chemistry</th>\n",
       "      <td>8870</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pathology</th>\n",
       "      <td>8655</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Cancer research</th>\n",
       "      <td>7490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Diabetes mellitus</th>\n",
       "      <td>6095</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene</th>\n",
       "      <td>5912</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Psychology</th>\n",
       "      <td>5773</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Virus</th>\n",
       "      <td>5252</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Physical therapy</th>\n",
       "      <td>4983</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Immune system</th>\n",
       "      <td>4951</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Antibody</th>\n",
       "      <td>4856</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Disease</th>\n",
       "      <td>4786</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Cardiology</th>\n",
       "      <td>4611</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Cell biology</th>\n",
       "      <td>4440</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Antigen</th>\n",
       "      <td>4089</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Surgery</th>\n",
       "      <td>4022</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Receptor</th>\n",
       "      <td>3610</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Anatomy</th>\n",
       "      <td>3413</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Anesthesia</th>\n",
       "      <td>2886</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T cell</th>\n",
       "      <td>2751</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mutation</th>\n",
       "      <td>2588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Linear molecular geometry</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Linear no-threshold model</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Linear phase</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Linear predictor function</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ligneous conjunctivitis</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Lightness</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Light-harvesting complex</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Light-emitting diode</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Levinthal&amp;#039;s paradox</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Levoamphetamine</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Levodopa-induced dyskinesia</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Levorphanol</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Lewis acid catalysis</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Leydig cell hypoplasia</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Liberalism</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Libman\u0013Sacks endocarditis</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Library management</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Lichtheimia corymbifera</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Lidoflazine</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Lien</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Life skills</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Life support</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Lifestyle management</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Lift (force)</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ligature</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Light beam</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Light chain deposition disease</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Light field</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Light scattering by particles</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>xDNA</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>29873 rows Ã— 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                title\n",
       "field                                \n",
       "Biology                         56725\n",
       "Medicine                        41832\n",
       "Molecular biology               23265\n",
       "Immunology                      22135\n",
       "Internal medicine               19022\n",
       "Virology                        15039\n",
       "Biochemistry                    14901\n",
       "Endocrinology                   13882\n",
       "Genetics                        13696\n",
       "Cancer                           9490\n",
       "Chemistry                        8870\n",
       "Pathology                        8655\n",
       "Cancer research                  7490\n",
       "Diabetes mellitus                6095\n",
       "Gene                             5912\n",
       "Psychology                       5773\n",
       "Virus                            5252\n",
       "Physical therapy                 4983\n",
       "Immune system                    4951\n",
       "Antibody                         4856\n",
       "Disease                          4786\n",
       "Cardiology                       4611\n",
       "Cell biology                     4440\n",
       "Antigen                          4089\n",
       "Surgery                          4022\n",
       "Receptor                         3610\n",
       "Anatomy                          3413\n",
       "Anesthesia                       2886\n",
       "T cell                           2751\n",
       "Mutation                         2588\n",
       "...                               ...\n",
       "Linear molecular geometry           1\n",
       "Linear no-threshold model           1\n",
       "Linear phase                        1\n",
       "Linear predictor function           1\n",
       "Ligneous conjunctivitis             1\n",
       "Lightness                           1\n",
       "Light-harvesting complex            1\n",
       "Light-emitting diode                1\n",
       "Levinthal&#039;s paradox            1\n",
       "Levoamphetamine                     1\n",
       "Levodopa-induced dyskinesia         1\n",
       "Levorphanol                         1\n",
       "Lewis acid catalysis                1\n",
       "Leydig cell hypoplasia              1\n",
       "Liberalism                          1\n",
       "Libman\u0013Sacks endocarditis           1\n",
       "Library management                  1\n",
       "Lichtheimia corymbifera             1\n",
       "Lidoflazine                         1\n",
       "Lien                                1\n",
       "Life skills                         1\n",
       "Life support                        1\n",
       "Lifestyle management                1\n",
       "Lift (force)                        1\n",
       "Ligature                            1\n",
       "Light beam                          1\n",
       "Light chain deposition disease      1\n",
       "Light field                         1\n",
       "Light scattering by particles       1\n",
       "xDNA                                1\n",
       "\n",
       "[29873 rows x 1 columns]"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dat.groupby('field').count()[['title']].sort_values('title', ascending=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## URL presence\n",
    "\n",
    "To what extent have open-data repositories been adopted? Have some repositories been adopted more quickly than others? Does this vary by field?\n",
    "\n",
    "These questions have bearing on our understanding of the effects of open-data, and open science more generally. In principle, open science is designed to facilitate the participation of a wide range of people in the scientific community. If groups that have a disproportionate share of resources consistently publish their likely resource-intensive datasets, then these valuable datasets can be mined by a wider range of individuals for scientific insights. \n",
    "\n",
    "To this end, I will examine how frequently URL's to some prominent open science repositories are present in papers as urls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "sparql.setQuery(\"\"\"\n",
    "PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>\n",
    "PREFIX magp: <http://ma-graph.org/property/>\n",
    "PREFIX mag: <http://ma-graph.org/>\n",
    "PREFIX dcterms: <http://purl.org/dc/terms/>\n",
    "PREFIX foaf: <http://xmlns.com/foaf/0.1/>\n",
    "PREFIX fabio: <http://purl.org/spar/fabio/>\n",
    "PREFIX org: <http://www.w3.org/ns/org#>\n",
    "PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>\n",
    "PREFIX cito: <http://purl.org/spar/cito/>\n",
    "PREFIX datacite: <http://purl.org/spar/datacite/>\n",
    "PREFIX prism: <http://prismstandard.org/namespaces/1.2/basic/>\n",
    "PREFIX c4o: <http://purl.org/spar/c4o/>\n",
    "\n",
    "SELECT ?paper ?paptitle ?abst ?dt ?ln ?auth ?doi ?estCites\n",
    "WHERE {\n",
    "?paper rdf:type <http://ma-graph.org/class/Paper> .\n",
    "?paper dcterms:title ?paptitle .\n",
    "?paper dcterms:abstract ?abst .\n",
    "?paper prism:publicationDate ?dt .\n",
    "?paper dcterms:creator ?auth .\n",
    "?paper datacite:doi ?doi .\n",
    "?paper dcterms:language ?ln .\n",
    "?paper dcterms:creator ?auth .\n",
    "}\n",
    "\"\"\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "sparql.setReturnFormat(JSON)\n",
    "results = sparql.query().convert() "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1048576"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(results['results']['bindings'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'paper': {'type': 'uri', 'value': 'http://ma-graph.org/entity/4524729'},\n",
       " 'paptitle': {'type': 'typed-literal',\n",
       "  'datatype': 'http://www.w3.org/2001/XMLSchema#string',\n",
       "  'value': 'Fiber Optics Sensors'},\n",
       " 'abst': {'type': 'typed-literal',\n",
       "  'datatype': 'http://www.w3.org/2001/XMLSchema#string',\n",
       "  'value': 'Fiber optic technology started in 1970\\x19s, for long distance telecommunications, and it has experienced an exponential growth during the last four decades. Sensing applications are a small spin-off from this technology, taking profit of developments done for optoelectronic components and concepts. Optical fibers are cylindrical dielectric waveguides for the propagation of the light, made out from high purity, low loss optical materials, usually silica (Optical fibers made from polymers and other transparent materials are also commercially available). The refractive index of the core (about 1.46 for silica) is slightly higher than surrounding material or cladding, due to the presence of dopants. Light is confined at the core, because optical rays arriving to the interface core/cladding with an angle higher than the total reflectance angle, as defined by the Snell law, will follow total reflections and will remain confined at the core. Only when the fiber is locally bent with a sharp radius, the light may escape.'},\n",
       " 'dt': {'type': 'typed-literal',\n",
       "  'datatype': 'http://www.w3.org/2001/XMLSchema#date',\n",
       "  'value': '2013-01-01'},\n",
       " 'ln': {'type': 'typed-literal',\n",
       "  'datatype': 'http://www.w3.org/2001/XMLSchema#language',\n",
       "  'value': 'en'},\n",
       " 'doi': {'type': 'typed-literal',\n",
       "  'datatype': 'http://www.w3.org/2001/XMLSchema#string',\n",
       "  'value': '10.1007/978-3-7091-1390-5_5'}}"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "results['results']['bindings'][2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "outdat = []\n",
    "for i in results['results']['bindings']:\n",
    "    outdat.append([i['paper']['value'], i['paptitle']['value'], \n",
    "                   i['abst']['value'], i['dt']['value'], i['ln']['value'],\n",
    "                   i['doi']['value']])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "dat = pd.DataFrame(outdat, columns = ['paper', 'title', 'abstract', 'date',\n",
    "                                      'language', 'doi'])\n",
    "dat = dat[dat.lang=='en']\n",
    "dat['open_science'] = dat.abstract.str.contains('open science', case=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "dat.to_pickle('/home/riddleta/ac_knowl/output/testdat.pkl')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "sparql.setQuery(\"\"\"\n",
    "PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>\n",
    "PREFIX magp: <http://ma-graph.org/property/>\n",
    "PREFIX mag: <http://ma-graph.org/>\n",
    "PREFIX dcterms: <http://purl.org/dc/terms/>\n",
    "PREFIX foaf: <http://xmlns.com/foaf/0.1/>\n",
    "PREFIX fabio: <http://purl.org/spar/fabio/>\n",
    "PREFIX org: <http://www.w3.org/ns/org#>\n",
    "PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>\n",
    "PREFIX cito: <http://purl.org/spar/cito/>\n",
    "PREFIX datacite: <http://purl.org/spar/datacite/>\n",
    "PREFIX prism: <http://prismstandard.org/namespaces/1.2/basic/>\n",
    "PREFIX c4o: <http://purl.org/spar/c4o/>\n",
    "\n",
    "SELECT ?author ?dt ?cites ?papes ?authname ?rnk ?inst\n",
    "WHERE {\n",
    "?author rdf:type <http://ma-graph.org/class/Author> .\n",
    "}\n",
    "\"\"\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "sparql.setReturnFormat(JSON)\n",
    "results = sparql.query().convert() "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1048576"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(results['results']['bindings'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "imagine a matrix of field x institution/author, with the cells of the matrix representing either counts of papers, or binary occurrences. What fields have a diversity of authors/institutions? what are more monopolistics? what are the consequences of the differences between these especially over time?\n",
    "\n",
    "is there an association between data sharing and market concentration (e.g. hh index)?\n",
    "\n",
    "also, look at what aaron clausett has been working on\n",
    "\n",
    "http://www.semantic-web-journal.net/system/files/swj1495.pdf\n",
    "\n",
    "group differences in \"tenure rates, competency evaluations, remuneration, allocation of research facilities, and\n",
    "grant competitions\" (way, larremore & clauset, 2016)\n",
    "\n",
    "\n",
    "plots to make:\n",
    "\n",
    "prevalency of links to various data-sharing platforms over time (see the [google spreadsheet](https://docs.google.com/spreadsheets/d/14GSgOppnvjS-pxGz1__fEBuj3KUNgKB7ni5nukHT8Ps/edit#gid=2016420688) for links)\n",
    "\n",
    "market concentration by...author? institution?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Some people working in the bibliometric/meta science space that might be relevant:\n",
    "\n",
    "## Aaron Clauset:\n",
    "https://arxiv.org/abs/1612.08228\n",
    "\n",
    "https://arxiv.org/abs/1804.02760\n",
    "\n",
    "https://arxiv.org/abs/1805.09966\n",
    "\n",
    "## Dani Bassett:\n",
    "https://arxiv.org/abs/1810.10534\n",
    "\n",
    "https://arxiv.org/abs/1804.06434\n",
    "\n",
    "https://www.sciencedirect.com/science/article/pii/S105381191830778X?via%3Dihub\n",
    "\n",
    "## Jevin West:\n",
    "https://jevinwest.org/papers/West2013PLoSOne.pdf\n",
    "\n",
    "https://jevinwest.org/papers/West2014EconInquiry.pdf\n",
    "\n",
    "https://jevinwest.org/papers/Vilhena2014SocScience.pdf\n",
    "\n",
    "https://jevinwest.org/papers/Larsen2014HICSS.pdf\n",
    "\n",
    "https://jevinwest.org/papers/Portenoy2016WWW.pdf\n",
    "\n",
    "https://jevinwest.org/papers/Kim2017asa.pdf\n",
    "\n",
    "https://jevinwest.org/papers/Chow2017fisheries.pdf\n",
    "\n",
    "https://jevinwest.org/papers/King2017socius.pdf\n",
    "\n",
    "https://jevinwest.org/papers/Piwowar2018peerj.pdf\n",
    "\n",
    "\n",
    "## Michael Farber\n",
    "http://dbis.informatik.uni-freiburg.de/content/team/faerber/papers/arXiv_Dataset_BIR2019.pdf\n",
    "\n",
    "\n",
    "\n",
    "## Other:\n",
    "https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005962\n",
    "\n",
    "https://cdn.elifesciences.org/articles/32822/elife-32822-v2.pdf\n",
    "\n",
    "https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/btz070/5305021\n",
    "\n",
    "https://link.springer.com/article/10.1007/s11192-012-0900-9\n",
    "\n",
    "https://link.springer.com/chapter/10.1007/978-3-319-67056-0_18\n",
    "\n",
    "https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5761855/\n",
    "\n",
    "https://onlinelibrary.wiley.com/doi/full/10.1002/asi.23266\n",
    "\n",
    "https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0000308\n",
    "\n",
    "https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2990274/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# datasets\n",
    "# arxiv (downloaded)\n",
    "# https://www.ncbi.nlm.nih.gov/research/bionlp/APIs/BioC-PMC/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Pulling arxiv papers\n",
    "\n",
    "These arxiv papers are connected to the ma graph. See Michael Farber's paper for additional details"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '/Downloads/arxiv_v2.1/0704.0001.txt'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-8-78fa68d43234>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'/Downloads/arxiv_v2.1/0704.0001.txt'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'w'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mf\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '/Downloads/arxiv_v2.1/0704.0001.txt'"
     ]
    }
   ],
   "source": [
    "f = open('/Downloads/arxiv_v2.1/0704.0001.txt','w') \n",
    "print(f) "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Get the pubmed papers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/riddleta/ac_knowl/ac_knowl/lib/python3.7/site-packages/ipykernel_launcher.py:2: FutureWarning: read_table is deprecated, use read_csv instead.\n",
      "  \n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(598386, 4)"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#bioc-pmc files\n",
    "files_to_get = pd.read_table('/home/riddleta/ac_knowl/data/filelist.txt', sep='\\s+')\n",
    "files_to_get.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1479862, 5)"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "oa_comm_use_files = pd.read_csv('/home/riddleta/ac_knowl/data/oa_comm_use_file_list.txt', \n",
    "                                sep='\\\\t', engine='python', header=None, skiprows=1)\n",
    "oa_comm_use_files.columns = ['File', 'ref', 'PMCID', 'PMID', 'license']\n",
    "oa_comm_use_files.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/riddleta/ac_knowl/ac_knowl/lib/python3.7/site-packages/ipykernel_launcher.py:2: FutureWarning: read_table is deprecated, use read_csv instead.\n",
      "  \n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(2408932, 5)"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "oa_files = pd.read_table('/home/riddleta/ac_knowl/data/oa_file_list.txt', \n",
    "                         sep='\\\\t', engine='python', header=None, skiprows=1)\n",
    "oa_files.columns = ['File', 'ref', 'PMCID', 'PMID', 'license']\n",
    "oa_files.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "unique_ids = pd.concat([oa_files['PMCID'], oa_comm_use_files['PMCID'], files_to_get['PMCID']]).unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "def return_useful_data(r):\n",
    "    git_hits = 0\n",
    "    osf_hits = 0\n",
    "    nda_hits = 0\n",
    "    open_neuro = 0\n",
    "    fmri = 0\n",
    "    res = 0\n",
    "    r_out = r.json()\n",
    "    date = r_out['date']\n",
    "    src = r_out['source']\n",
    "    try:\n",
    "        doi = r_out['documents'][0]['passages'][0]['infons']['article-id_doi']\n",
    "    except:\n",
    "        doi = None\n",
    "    try:\n",
    "        yr = r_out['documents'][0]['passages'][0]['infons']['year']\n",
    "    except:\n",
    "        yr = None\n",
    "    if yr not in ['2008', '2009', '2010', '2011', '2012', '2013', \n",
    "                  '2014', '2015', '2016', '2017', '2018', '2019']:\n",
    "        return(date, src, doi, git_hits, osf_hits, nda_hits, open_neuro, fmri, res, yr)\n",
    "    else:\n",
    "        for passage in r_out['documents'][0]['passages']:\n",
    "            if re.search(re.compile('github'), passage['text'].lower()):\n",
    "                git_hits = git_hits + 1\n",
    "            if re.search(re.compile('osf\\.io'), passage['text'].lower()):\n",
    "                osf_hits = osf_hits + 1\n",
    "            if re.search(re.compile('nda\\.nih\\.gov'), passage['text'].lower()):\n",
    "                nda_hits = nda_hits + 1\n",
    "            if re.search(re.compile('openneuro'), passage['text'].lower()):\n",
    "                open_neuro = open_neuro + 1\n",
    "            if re.search(re.compile('fmri'), passage['text'].lower()):\n",
    "                fmri = fmri + 1\n",
    "            if re.search(re.compile('results'), passage['text'].lower()):\n",
    "                res = res + 1\n",
    "        \n",
    "        return(date, src, doi, git_hits, osf_hits, nda_hits, open_neuro, fmri, res, yr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [],
   "source": [
    "git_links = []\n",
    "for i, j in enumerate(unique_ids):\n",
    "    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + j + '/unicode'\n",
    "    r = requests.get(pth)\n",
    "    git_links.append(return_useful_data(r) + (j, i))\n",
    "    if i==10000:\n",
    "        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.DataFrame(git_links, columns=['date', 'src', 'doi', 'git_hits', \n",
    "                                      'osf_hits', 'nda_hits', 'open_neuro', \n",
    "                                      'fmri', 'res', 'yr', 'pmcid', 'idx'])\n",
    "df.to_csv('/home/riddleta/ac_knowl/output/bionlp_10k.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {},
   "outputs": [],
   "source": [
    "git_links = []\n",
    "for i, j in enumerate(unique_ids[10000:]):\n",
    "    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + j + '/unicode'\n",
    "    r = requests.get(pth)\n",
    "    git_links.append(return_useful_data(r) + (j, i))\n",
    "    if i==15000:\n",
    "        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.DataFrame(git_links, columns=['date', 'src', 'doi', 'git_hits', \n",
    "                                      'osf_hits', 'nda_hits', 'open_neuro', \n",
    "                                      'fmri', 'res', 'yr', 'pmcid', 'idx'])\n",
    "df.to_csv('/home/riddleta/ac_knowl/output/bionlp_02.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "git_links = []\n",
    "missed = []\n",
    "\n",
    "for i, j in enumerate(unique_ids[25000:]):\n",
    "    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + j + '/unicode'\n",
    "    try:\n",
    "        r = requests.get(pth)\n",
    "        git_links.append(return_useful_data(r) + (j, i))\n",
    "    except:\n",
    "        missed.append([i,j])\n",
    "    if i==20000:\n",
    "        break\n",
    "df = pd.DataFrame(git_links, columns=['date', 'src', 'doi', 'git_hits', \n",
    "                                      'osf_hits', 'nda_hits', 'open_neuro', \n",
    "                                      'fmri', 'res', 'yr', 'pmcid', 'idx'])\n",
    "df.to_csv('/home/riddleta/ac_knowl/output/bionlp_03.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "1000\n",
      "2000\n",
      "3000\n",
      "4000\n",
      "5000\n",
      "6000\n",
      "7000\n",
      "8000\n",
      "9000\n",
      "10000\n",
      "11000\n",
      "12000\n",
      "13000\n",
      "14000\n",
      "15000\n",
      "16000\n",
      "17000\n",
      "18000\n",
      "19000\n",
      "20000\n",
      "21000\n",
      "22000\n",
      "23000\n",
      "24000\n"
     ]
    }
   ],
   "source": [
    "git_links = []\n",
    "missed = []\n",
    "\n",
    "progress = [i for i in range(0, 25000, 1000)]\n",
    "\n",
    "for i, j in enumerate(unique_ids[45000:]):\n",
    "    if i in progress:\n",
    "        print(i)\n",
    "    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + j + '/unicode'\n",
    "    try:\n",
    "        r = requests.get(pth)\n",
    "        git_links.append(return_useful_data(r) + (j, i))\n",
    "    except:\n",
    "        missed.append([i,j])\n",
    "    if i==25000:\n",
    "        break\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "1000\n",
      "2000\n",
      "3000\n",
      "4000\n",
      "5000\n",
      "6000\n",
      "7000\n",
      "8000\n",
      "9000\n",
      "10000\n",
      "11000\n",
      "12000\n",
      "13000\n",
      "14000\n",
      "15000\n",
      "16000\n",
      "17000\n",
      "18000\n",
      "19000\n",
      "20000\n",
      "21000\n",
      "22000\n",
      "23000\n",
      "24000\n",
      "25000\n",
      "26000\n",
      "27000\n",
      "28000\n",
      "29000\n"
     ]
    }
   ],
   "source": [
    "git_links = []\n",
    "missed = []\n",
    "\n",
    "progress = [i for i in range(0, 30000, 1000)]\n",
    "\n",
    "for i, j in enumerate(unique_ids[70000:]):\n",
    "    if i in progress:\n",
    "        print(i)\n",
    "    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + j + '/unicode'\n",
    "    try:\n",
    "        r = requests.get(pth)\n",
    "        git_links.append(return_useful_data(r) + (j, i))\n",
    "    except:\n",
    "        missed.append([i,j])\n",
    "    if i==30000:\n",
    "        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.DataFrame(git_links, columns=['date', 'src', 'doi', 'git_hits', \n",
    "                                      'osf_hits', 'nda_hits', 'open_neuro', \n",
    "                                      'fmri', 'res', 'yr', 'pmcid', 'idx'])\n",
    "df.to_csv('/home/riddleta/ac_knowl/output/bionlp_05.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "2500\n",
      "5000\n",
      "7500\n",
      "10000\n",
      "12500\n",
      "15000\n",
      "17500\n",
      "20000\n",
      "22500\n",
      "25000\n",
      "27500\n",
      "30000\n",
      "32500\n",
      "35000\n",
      "37500\n",
      "40000\n",
      "42500\n",
      "45000\n",
      "47500\n",
      "50000\n",
      "52500\n",
      "55000\n",
      "57500\n",
      "60000\n",
      "62500\n",
      "65000\n",
      "67500\n",
      "70000\n",
      "72500\n",
      "75000\n",
      "77500\n",
      "80000\n",
      "82500\n",
      "85000\n",
      "87500\n",
      "90000\n",
      "92500\n",
      "95000\n",
      "97500\n"
     ]
    }
   ],
   "source": [
    "git_links = []\n",
    "missed = []\n",
    "\n",
    "progress = [i for i in range(0, 100000, 2500)]\n",
    "\n",
    "for i, j in enumerate(unique_ids[100000:]):\n",
    "    if i in progress:\n",
    "        print(i)\n",
    "    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + j + '/unicode'\n",
    "    try:\n",
    "        r = requests.get(pth)\n",
    "        git_links.append(return_useful_data(r) + (j, i))\n",
    "    except:\n",
    "        missed.append([i,j])\n",
    "    if i==100000:\n",
    "        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>date</th>\n",
       "      <th>src</th>\n",
       "      <th>doi</th>\n",
       "      <th>git_hits</th>\n",
       "      <th>osf_hits</th>\n",
       "      <th>nda_hits</th>\n",
       "      <th>open_neuro</th>\n",
       "      <th>fmri</th>\n",
       "      <th>res</th>\n",
       "      <th>yr</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>idx</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>20181110</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1083/jcb.200211071581251</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2002</td>\n",
       "      <td>PMC2173096</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20181110</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1083/jcb.200208058</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2002</td>\n",
       "      <td>PMC2173097</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20181110</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1083/jcb.200204153</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2002</td>\n",
       "      <td>PMC2173098</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20181110</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1083/jcb.200206024</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2002</td>\n",
       "      <td>PMC2173099</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20181117</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1083/jcb.200204102</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2002</td>\n",
       "      <td>PMC2173100</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       date  src                          doi  git_hits  osf_hits  nda_hits  \\\n",
       "0  20181110  PMC  10.1083/jcb.200211071581251         0         0         0   \n",
       "1  20181110  PMC        10.1083/jcb.200208058         0         0         0   \n",
       "2  20181110  PMC        10.1083/jcb.200204153         0         0         0   \n",
       "3  20181110  PMC        10.1083/jcb.200206024         0         0         0   \n",
       "4  20181117  PMC        10.1083/jcb.200204102         0         0         0   \n",
       "\n",
       "   open_neuro  fmri  res    yr       pmcid  idx  \n",
       "0           0     0    0  2002  PMC2173096    0  \n",
       "1           0     0    0  2002  PMC2173097    1  \n",
       "2           0     0    0  2002  PMC2173098    2  \n",
       "3           0     0    0  2002  PMC2173099    3  \n",
       "4           0     0    0  2002  PMC2173100    4  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.DataFrame(git_links, columns=['date', 'src', 'doi', 'git_hits', \n",
    "                                      'osf_hits', 'nda_hits', 'open_neuro', \n",
    "                                      'fmri', 'res', 'yr', 'pmcid', 'idx'])\n",
    "df.to_csv('/home/riddleta/ac_knowl/output/bionlp_06.csv')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "2500\n",
      "5000\n",
      "7500\n",
      "10000\n",
      "12500\n",
      "15000\n",
      "17500\n",
      "20000\n",
      "22500\n",
      "25000\n",
      "27500\n",
      "30000\n",
      "32500\n"
     ]
    }
   ],
   "source": [
    "git_links = []\n",
    "missed = []\n",
    "\n",
    "progress = [i for i in range(0, 35000, 2500)]\n",
    "\n",
    "for i, j in enumerate(unique_ids[200000:]):\n",
    "    if i in progress:\n",
    "        print(i)\n",
    "    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + j + '/unicode'\n",
    "    try:\n",
    "        r = requests.get(pth)\n",
    "        git_links.append(return_useful_data(r) + (j, i))\n",
    "    except:\n",
    "        missed.append([i,j])\n",
    "    if i==35000:\n",
    "        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>date</th>\n",
       "      <th>src</th>\n",
       "      <th>doi</th>\n",
       "      <th>git_hits</th>\n",
       "      <th>osf_hits</th>\n",
       "      <th>nda_hits</th>\n",
       "      <th>open_neuro</th>\n",
       "      <th>fmri</th>\n",
       "      <th>res</th>\n",
       "      <th>yr</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>idx</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>20181207</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1186/1297-9686-29-4-413</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1997</td>\n",
       "      <td>PMC2708220</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20181207</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1186/1297-9686-29-4-427</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1997</td>\n",
       "      <td>PMC2708221</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20181207</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1186/1297-9686-29-4-451</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1997</td>\n",
       "      <td>PMC2708222</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20181207</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1186/1297-9686-29-4-469</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1997</td>\n",
       "      <td>PMC2708223</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20181207</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1186/1297-9686-29-4-483</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1997</td>\n",
       "      <td>PMC2708224</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       date  src                         doi  git_hits  osf_hits  nda_hits  \\\n",
       "0  20181207  PMC  10.1186/1297-9686-29-4-413         0         0         0   \n",
       "1  20181207  PMC  10.1186/1297-9686-29-4-427         0         0         0   \n",
       "2  20181207  PMC  10.1186/1297-9686-29-4-451         0         0         0   \n",
       "3  20181207  PMC  10.1186/1297-9686-29-4-469         0         0         0   \n",
       "4  20181207  PMC  10.1186/1297-9686-29-4-483         0         0         0   \n",
       "\n",
       "   open_neuro  fmri  res    yr       pmcid  idx  \n",
       "0           0     0    0  1997  PMC2708220    0  \n",
       "1           0     0    0  1997  PMC2708221    1  \n",
       "2           0     0    0  1997  PMC2708222    2  \n",
       "3           0     0    0  1997  PMC2708223    3  \n",
       "4           0     0    0  1997  PMC2708224    4  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.DataFrame(git_links, columns=['date', 'src', 'doi', 'git_hits', \n",
    "                                      'osf_hits', 'nda_hits', 'open_neuro', \n",
    "                                      'fmri', 'res', 'yr', 'pmcid', 'idx'])\n",
    "df.to_csv('/home/riddleta/ac_knowl/output/bionlp_07.csv')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "2500\n",
      "5000\n",
      "7500\n",
      "10000\n",
      "12500\n",
      "15000\n",
      "17500\n",
      "20000\n",
      "22500\n",
      "25000\n",
      "27500\n",
      "30000\n",
      "32500\n",
      "35000\n",
      "37500\n"
     ]
    }
   ],
   "source": [
    "git_links = []\n",
    "progress = [i for i in range(0, 40000, 2500)]\n",
    "\n",
    "for i, j in enumerate(unique_ids[235000:]):\n",
    "    if i in progress:\n",
    "        print(i)\n",
    "    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + j + '/unicode'\n",
    "    try:\n",
    "        r = requests.get(pth)\n",
    "        git_links.append(return_useful_data(r) + (j, i))\n",
    "    except:\n",
    "        missed.append([i,j])\n",
    "    if i==40000:\n",
    "        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>date</th>\n",
       "      <th>src</th>\n",
       "      <th>doi</th>\n",
       "      <th>git_hits</th>\n",
       "      <th>osf_hits</th>\n",
       "      <th>nda_hits</th>\n",
       "      <th>open_neuro</th>\n",
       "      <th>fmri</th>\n",
       "      <th>res</th>\n",
       "      <th>yr</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>idx</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>20181124</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.4103/0970-2113.44123</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2008</td>\n",
       "      <td>PMC2822325</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20181207</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.4103/0970-2113.44127</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2008</td>\n",
       "      <td>PMC2822326</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20181124</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.4103/0970-2113.59590</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2008</td>\n",
       "      <td>PMC2822327</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20181124</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.4103/0970-2113.59591</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2008</td>\n",
       "      <td>PMC2822328</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20181124</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.4103/0970-2113.44122</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>2008</td>\n",
       "      <td>PMC2822329</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       date  src                      doi  git_hits  osf_hits  nda_hits  \\\n",
       "0  20181124  PMC  10.4103/0970-2113.44123         0         0         0   \n",
       "1  20181207  PMC  10.4103/0970-2113.44127         0         0         0   \n",
       "2  20181124  PMC  10.4103/0970-2113.59590         0         0         0   \n",
       "3  20181124  PMC  10.4103/0970-2113.59591         0         0         0   \n",
       "4  20181124  PMC  10.4103/0970-2113.44122         0         0         0   \n",
       "\n",
       "   open_neuro  fmri  res    yr       pmcid  idx  \n",
       "0           0     0    1  2008  PMC2822325    0  \n",
       "1           0     0    1  2008  PMC2822326    1  \n",
       "2           0     0    0  2008  PMC2822327    2  \n",
       "3           0     0    2  2008  PMC2822328    3  \n",
       "4           0     0    6  2008  PMC2822329    4  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.DataFrame(git_links, columns=['date', 'src', 'doi', 'git_hits', \n",
    "                                      'osf_hits', 'nda_hits', 'open_neuro', \n",
    "                                      'fmri', 'res', 'yr', 'pmcid', 'idx'])\n",
    "df.to_csv('/home/riddleta/ac_knowl/output/bionlp_08.csv')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "2500\n",
      "5000\n",
      "7500\n",
      "10000\n",
      "12500\n",
      "15000\n",
      "17500\n",
      "20000\n",
      "22500\n",
      "25000\n",
      "27500\n",
      "30000\n",
      "32500\n",
      "35000\n",
      "37500\n",
      "40000\n",
      "42500\n",
      "45000\n",
      "47500\n",
      "50000\n",
      "52500\n",
      "55000\n",
      "57500\n",
      "60000\n",
      "62500\n",
      "65000\n",
      "67500\n",
      "70000\n",
      "72500\n",
      "75000\n",
      "77500\n",
      "80000\n",
      "82500\n",
      "85000\n",
      "87500\n",
      "90000\n",
      "92500\n",
      "95000\n",
      "97500\n",
      "100000\n",
      "102500\n",
      "105000\n",
      "107500\n",
      "110000\n",
      "112500\n",
      "115000\n",
      "117500\n",
      "120000\n",
      "122500\n"
     ]
    }
   ],
   "source": [
    "git_links = []\n",
    "progress = [i for i in range(0, 125000, 2500)]\n",
    "\n",
    "for i, j in enumerate(unique_ids[275000:]):\n",
    "    if i in progress:\n",
    "        print(i)\n",
    "    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + j + '/unicode'\n",
    "    try:\n",
    "        r = requests.get(pth)\n",
    "        git_links.append(return_useful_data(r) + (j, i))\n",
    "    except:\n",
    "        missed.append([i,j])\n",
    "    if i==125000:\n",
    "        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>date</th>\n",
       "      <th>src</th>\n",
       "      <th>doi</th>\n",
       "      <th>git_hits</th>\n",
       "      <th>osf_hits</th>\n",
       "      <th>nda_hits</th>\n",
       "      <th>open_neuro</th>\n",
       "      <th>fmri</th>\n",
       "      <th>res</th>\n",
       "      <th>yr</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>idx</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>20181121</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1186/gm171</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>2011</td>\n",
       "      <td>PMC2945007</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20181121</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1186/gm172</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2010</td>\n",
       "      <td>PMC2945008</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20181120</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1186/gm173</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2011</td>\n",
       "      <td>PMC2945009</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20181121</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1186/gm174</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>2011</td>\n",
       "      <td>PMC2945010</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20181121</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1186/gm175</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2011</td>\n",
       "      <td>PMC2945011</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       date  src            doi  git_hits  osf_hits  nda_hits  open_neuro  \\\n",
       "0  20181121  PMC  10.1186/gm171         0         0         0           0   \n",
       "1  20181121  PMC  10.1186/gm172         0         0         0           0   \n",
       "2  20181120  PMC  10.1186/gm173         0         0         0           0   \n",
       "3  20181121  PMC  10.1186/gm174         0         0         0           0   \n",
       "4  20181121  PMC  10.1186/gm175         0         0         0           0   \n",
       "\n",
       "   fmri  res    yr       pmcid  idx  \n",
       "0     0    6  2011  PMC2945007    0  \n",
       "1     0    2  2010  PMC2945008    1  \n",
       "2     0    0  2011  PMC2945009    2  \n",
       "3     0    5  2011  PMC2945010    3  \n",
       "4     0    1  2011  PMC2945011    4  "
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.DataFrame(git_links, columns=['date', 'src', 'doi', 'git_hits', \n",
    "                                      'osf_hits', 'nda_hits', 'open_neuro', \n",
    "                                      'fmri', 'res', 'yr', 'pmcid', 'idx'])\n",
    "df.to_csv('/home/riddleta/ac_knowl/output/bionlp_09.csv')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "2500\n",
      "5000\n",
      "7500\n",
      "10000\n",
      "12500\n",
      "15000\n",
      "17500\n",
      "20000\n",
      "22500\n",
      "25000\n",
      "27500\n",
      "30000\n",
      "32500\n",
      "35000\n",
      "37500\n",
      "40000\n",
      "42500\n"
     ]
    }
   ],
   "source": [
    "git_links = []\n",
    "progress = [i for i in range(0, 45000, 2500)]\n",
    "\n",
    "for i, j in enumerate(unique_ids[400000:]):\n",
    "    if i in progress:\n",
    "        print(i)\n",
    "    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + j + '/unicode'\n",
    "    try:\n",
    "        r = requests.get(pth)\n",
    "        git_links.append(return_useful_data(r) + (j, i))\n",
    "    except:\n",
    "        missed.append([i,j])\n",
    "    if i==45000:\n",
    "        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>date</th>\n",
       "      <th>src</th>\n",
       "      <th>doi</th>\n",
       "      <th>git_hits</th>\n",
       "      <th>osf_hits</th>\n",
       "      <th>nda_hits</th>\n",
       "      <th>open_neuro</th>\n",
       "      <th>fmri</th>\n",
       "      <th>res</th>\n",
       "      <th>yr</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>idx</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>20181207</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1186/1471-2105-12-420</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>2011</td>\n",
       "      <td>PMC3225379</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20181110</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1371/journal.pone.0027740</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>2011</td>\n",
       "      <td>PMC3225380</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20181130</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1371/journal.pone.0028098</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>2011</td>\n",
       "      <td>PMC3225381</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20181110</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1371/journal.pone.0027946</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>2011</td>\n",
       "      <td>PMC3225382</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20181121</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1371/journal.pone.0028053</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>2011</td>\n",
       "      <td>PMC3225383</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       date  src                           doi  git_hits  osf_hits  nda_hits  \\\n",
       "0  20181207  PMC      10.1186/1471-2105-12-420         0         0         0   \n",
       "1  20181110  PMC  10.1371/journal.pone.0027740         0         0         0   \n",
       "2  20181130  PMC  10.1371/journal.pone.0028098         0         0         0   \n",
       "3  20181110  PMC  10.1371/journal.pone.0027946         0         0         0   \n",
       "4  20181121  PMC  10.1371/journal.pone.0028053         0         0         0   \n",
       "\n",
       "   open_neuro  fmri  res    yr       pmcid  idx  \n",
       "0           0     0    5  2011  PMC3225379    0  \n",
       "1           0     0   12  2011  PMC3225380    1  \n",
       "2           0     0   13  2011  PMC3225381    2  \n",
       "3           0     0    9  2011  PMC3225382    3  \n",
       "4           0     0    6  2011  PMC3225383    4  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.DataFrame(git_links, columns=['date', 'src', 'doi', 'git_hits', \n",
    "                                      'osf_hits', 'nda_hits', 'open_neuro', \n",
    "                                      'fmri', 'res', 'yr', 'pmcid', 'idx'])\n",
    "df.to_csv('/home/riddleta/ac_knowl/output/bionlp_10.csv')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "2500\n",
      "5000\n",
      "7500\n",
      "10000\n",
      "12500\n",
      "15000\n",
      "17500\n"
     ]
    }
   ],
   "source": [
    "git_links = []\n",
    "missed = []\n",
    "progress = [i for i in range(0, 20000, 2500)]\n",
    "\n",
    "for i, j in enumerate(unique_ids[445000:]):\n",
    "    if i in progress:\n",
    "        print(i)\n",
    "    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + j + '/unicode'\n",
    "    try:\n",
    "        r = requests.get(pth)\n",
    "        git_links.append(return_useful_data(r) + (j, i))\n",
    "    except:\n",
    "        missed.append([i,j])\n",
    "    if i==20000:\n",
    "        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>date</th>\n",
       "      <th>src</th>\n",
       "      <th>doi</th>\n",
       "      <th>git_hits</th>\n",
       "      <th>osf_hits</th>\n",
       "      <th>nda_hits</th>\n",
       "      <th>open_neuro</th>\n",
       "      <th>fmri</th>\n",
       "      <th>res</th>\n",
       "      <th>yr</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>idx</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>20181123</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1186/1471-2210-8-S1-A54</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2008</td>\n",
       "      <td>PMC3313246</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20181201</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1186/1471-2210-8-S1-A55</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2008</td>\n",
       "      <td>PMC3313247</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20181201</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1186/1471-2210-8-S1-A56</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2008</td>\n",
       "      <td>PMC3313248</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20181201</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1186/1471-2210-8-S1-A57</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2008</td>\n",
       "      <td>PMC3313249</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20181207</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1186/1471-2210-8-S1-A58</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2008</td>\n",
       "      <td>PMC3313250</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       date  src                         doi  git_hits  osf_hits  nda_hits  \\\n",
       "0  20181123  PMC  10.1186/1471-2210-8-S1-A54         0         0         0   \n",
       "1  20181201  PMC  10.1186/1471-2210-8-S1-A55         0         0         0   \n",
       "2  20181201  PMC  10.1186/1471-2210-8-S1-A56         0         0         0   \n",
       "3  20181201  PMC  10.1186/1471-2210-8-S1-A57         0         0         0   \n",
       "4  20181207  PMC  10.1186/1471-2210-8-S1-A58         0         0         0   \n",
       "\n",
       "   open_neuro  fmri  res    yr       pmcid  idx  \n",
       "0           0     0    1  2008  PMC3313246    0  \n",
       "1           0     0    1  2008  PMC3313247    1  \n",
       "2           0     0    2  2008  PMC3313248    2  \n",
       "3           0     0    1  2008  PMC3313249    3  \n",
       "4           0     0    1  2008  PMC3313250    4  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.DataFrame(git_links, columns=['date', 'src', 'doi', 'git_hits', \n",
    "                                      'osf_hits', 'nda_hits', 'open_neuro', \n",
    "                                      'fmri', 'res', 'yr', 'pmcid', 'idx'])\n",
    "df.to_csv('/home/riddleta/ac_knowl/output/bionlp_11.csv')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "2500\n",
      "5000\n",
      "7500\n"
     ]
    }
   ],
   "source": [
    "git_links = []\n",
    "missed = []\n",
    "progress = [i for i in range(0, 10000, 2500)]\n",
    "\n",
    "for i, j in enumerate(unique_ids[465000:]):\n",
    "    if i in progress:\n",
    "        print(i)\n",
    "    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + j + '/unicode'\n",
    "    try:\n",
    "        r = requests.get(pth)\n",
    "        git_links.append(return_useful_data(r) + (j, i))\n",
    "    except:\n",
    "        missed.append([i,j])\n",
    "    if i==10000:\n",
    "        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>date</th>\n",
       "      <th>src</th>\n",
       "      <th>doi</th>\n",
       "      <th>git_hits</th>\n",
       "      <th>osf_hits</th>\n",
       "      <th>nda_hits</th>\n",
       "      <th>open_neuro</th>\n",
       "      <th>fmri</th>\n",
       "      <th>res</th>\n",
       "      <th>yr</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>idx</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>20181121</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1371/journal.pgen.1002706</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>2012</td>\n",
       "      <td>PMC3349741</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20181121</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1371/journal.pgen.1002705</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>2012</td>\n",
       "      <td>PMC3349742</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20181121</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1371/journal.ppat.1002607</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2012</td>\n",
       "      <td>PMC3349743</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20190113</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1371/journal.ppat.1002679</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>2012</td>\n",
       "      <td>PMC3349744</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20181114</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1371/journal.ppat.1002675</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>2012</td>\n",
       "      <td>PMC3349745</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       date  src                           doi  git_hits  osf_hits  nda_hits  \\\n",
       "0  20181121  PMC  10.1371/journal.pgen.1002706         0         0         0   \n",
       "1  20181121  PMC  10.1371/journal.pgen.1002705         0         0         0   \n",
       "2  20181121  PMC  10.1371/journal.ppat.1002607         0         0         0   \n",
       "3  20190113  PMC  10.1371/journal.ppat.1002679         0         0         0   \n",
       "4  20181114  PMC  10.1371/journal.ppat.1002675         0         0         0   \n",
       "\n",
       "   open_neuro  fmri  res    yr       pmcid  idx  \n",
       "0           0     0    9  2012  PMC3349741    0  \n",
       "1           0     0    7  2012  PMC3349742    1  \n",
       "2           0     0    2  2012  PMC3349743    2  \n",
       "3           0     0    8  2012  PMC3349744    3  \n",
       "4           0     0   20  2012  PMC3349745    4  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.DataFrame(git_links, columns=['date', 'src', 'doi', 'git_hits', \n",
    "                                      'osf_hits', 'nda_hits', 'open_neuro', \n",
    "                                      'fmri', 'res', 'yr', 'pmcid', 'idx'])\n",
    "df.to_csv('/home/riddleta/ac_knowl/output/bionlp_12.csv')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "5000\n",
      "10000\n",
      "15000\n",
      "20000\n",
      "25000\n",
      "30000\n",
      "35000\n",
      "40000\n",
      "45000\n",
      "50000\n",
      "55000\n",
      "60000\n",
      "65000\n",
      "70000\n",
      "75000\n",
      "80000\n",
      "85000\n",
      "90000\n",
      "95000\n",
      "100000\n",
      "105000\n",
      "110000\n",
      "115000\n",
      "120000\n",
      "125000\n",
      "130000\n",
      "135000\n",
      "140000\n",
      "145000\n"
     ]
    }
   ],
   "source": [
    "git_links = []\n",
    "missed = []\n",
    "progress = [i for i in range(0, 150000, 5000)]\n",
    "\n",
    "for i, j in enumerate(unique_ids[475000:]):\n",
    "    if i in progress:\n",
    "        print(i)\n",
    "    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + j + '/unicode'\n",
    "    try:\n",
    "        r = requests.get(pth)\n",
    "        git_links.append(return_useful_data(r) + (j, i))\n",
    "    except:\n",
    "        missed.append([i,j])\n",
    "    if i==150000:\n",
    "        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>date</th>\n",
       "      <th>src</th>\n",
       "      <th>doi</th>\n",
       "      <th>git_hits</th>\n",
       "      <th>osf_hits</th>\n",
       "      <th>nda_hits</th>\n",
       "      <th>open_neuro</th>\n",
       "      <th>fmri</th>\n",
       "      <th>res</th>\n",
       "      <th>yr</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>idx</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>20181124</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.3201/eid1112.041293</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2005</td>\n",
       "      <td>PMC3367621</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20181207</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.3201/eid1112.050079</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2005</td>\n",
       "      <td>PMC3367622</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20181207</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.3201/eid1112.050158</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2005</td>\n",
       "      <td>PMC3367623</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20181124</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.3201/eid1112.050217</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2005</td>\n",
       "      <td>PMC3367624</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20181207</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.3201/eid1112.050220</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2005</td>\n",
       "      <td>PMC3367625</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       date  src                     doi  git_hits  osf_hits  nda_hits  \\\n",
       "0  20181124  PMC  10.3201/eid1112.041293         0         0         0   \n",
       "1  20181207  PMC  10.3201/eid1112.050079         0         0         0   \n",
       "2  20181207  PMC  10.3201/eid1112.050158         0         0         0   \n",
       "3  20181124  PMC  10.3201/eid1112.050217         0         0         0   \n",
       "4  20181207  PMC  10.3201/eid1112.050220         0         0         0   \n",
       "\n",
       "   open_neuro  fmri  res    yr       pmcid  idx  \n",
       "0           0     0    0  2005  PMC3367621    0  \n",
       "1           0     0    0  2005  PMC3367622    1  \n",
       "2           0     0    0  2005  PMC3367623    2  \n",
       "3           0     0    0  2005  PMC3367624    3  \n",
       "4           0     0    0  2005  PMC3367625    4  "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.DataFrame(git_links, columns=['date', 'src', 'doi', 'git_hits', \n",
    "                                      'osf_hits', 'nda_hits', 'open_neuro', \n",
    "                                      'fmri', 'res', 'yr', 'pmcid', 'idx'])\n",
    "df.to_csv('/home/riddleta/ac_knowl/output/bionlp_13.csv')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "2500\n",
      "5000\n",
      "7500\n",
      "10000\n",
      "12500\n"
     ]
    }
   ],
   "source": [
    "git_links = []\n",
    "missed = []\n",
    "progress = [i for i in range(0, 15000, 2500)]\n",
    "\n",
    "for i, j in enumerate(unique_ids[625000:]):\n",
    "    if i in progress:\n",
    "        print(i)\n",
    "    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + j + '/unicode'\n",
    "    try:\n",
    "        r = requests.get(pth)\n",
    "        git_links.append(return_useful_data(r) + (j, i))\n",
    "    except:\n",
    "        missed.append([i,j])\n",
    "    if i==15000:\n",
    "        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>date</th>\n",
       "      <th>src</th>\n",
       "      <th>doi</th>\n",
       "      <th>git_hits</th>\n",
       "      <th>osf_hits</th>\n",
       "      <th>nda_hits</th>\n",
       "      <th>open_neuro</th>\n",
       "      <th>fmri</th>\n",
       "      <th>res</th>\n",
       "      <th>yr</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>idx</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>20181203</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.3341/kjo.2013.27.3.211</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2013</td>\n",
       "      <td>PMC3663066</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20181202</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.3341/kjo.2013.27.3.215</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2013</td>\n",
       "      <td>PMC3663067</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20181206</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.3341/kjo.2013.27.3.219</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2013</td>\n",
       "      <td>PMC3663068</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20181203</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.3341/kjo.2013.27.3.224</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>2013</td>\n",
       "      <td>PMC3663069</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20181113</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1021/ja4020925</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>2013</td>\n",
       "      <td>PMC3663070</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       date  src                        doi  git_hits  osf_hits  nda_hits  \\\n",
       "0  20181203  PMC  10.3341/kjo.2013.27.3.211         0         0         0   \n",
       "1  20181202  PMC  10.3341/kjo.2013.27.3.215         0         0         0   \n",
       "2  20181206  PMC  10.3341/kjo.2013.27.3.219         0         0         0   \n",
       "3  20181203  PMC  10.3341/kjo.2013.27.3.224         0         0         0   \n",
       "4  20181113  PMC          10.1021/ja4020925         0         0         0   \n",
       "\n",
       "   open_neuro  fmri  res    yr       pmcid  idx  \n",
       "0           0     0    2  2013  PMC3663066    0  \n",
       "1           0     0    0  2013  PMC3663067    1  \n",
       "2           0     0    1  2013  PMC3663068    2  \n",
       "3           0     0    4  2013  PMC3663069    3  \n",
       "4           0     0    4  2013  PMC3663070    4  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.DataFrame(git_links, columns=['date', 'src', 'doi', 'git_hits', \n",
    "                                      'osf_hits', 'nda_hits', 'open_neuro', \n",
    "                                      'fmri', 'res', 'yr', 'pmcid', 'idx'])\n",
    "df.to_csv('/home/riddleta/ac_knowl/output/bionlp_14.csv')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "2500\n",
      "5000\n",
      "7500\n",
      "10000\n",
      "12500\n",
      "15000\n",
      "17500\n"
     ]
    }
   ],
   "source": [
    "git_links = []\n",
    "missed = []\n",
    "progress = [i for i in range(0, 20000, 2500)]\n",
    "\n",
    "for i, j in enumerate(unique_ids[640000:]):\n",
    "    if i in progress:\n",
    "        print(i)\n",
    "    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + j + '/unicode'\n",
    "    try:\n",
    "        r = requests.get(pth)\n",
    "        git_links.append(return_useful_data(r) + (j, i))\n",
    "    except:\n",
    "        missed.append([i,j])\n",
    "    if i==20000:\n",
    "        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>date</th>\n",
       "      <th>src</th>\n",
       "      <th>doi</th>\n",
       "      <th>git_hits</th>\n",
       "      <th>osf_hits</th>\n",
       "      <th>nda_hits</th>\n",
       "      <th>open_neuro</th>\n",
       "      <th>fmri</th>\n",
       "      <th>res</th>\n",
       "      <th>yr</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>idx</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>20181207</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.3389/fpsyg.2013.00354</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>2013</td>\n",
       "      <td>PMC3691502</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20181207</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.3389/fpsyg.2013.00370</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>2013</td>\n",
       "      <td>PMC3691503</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20181207</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.3389/fnhum.2013.00304</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>2013</td>\n",
       "      <td>PMC3691504</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20181210</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.3389/fmicb.2013.00161</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>2013</td>\n",
       "      <td>PMC3691505</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20181210</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.3389/fnsys.2013.00020</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2013</td>\n",
       "      <td>PMC3691506</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       date  src                       doi  git_hits  osf_hits  nda_hits  \\\n",
       "0  20181207  PMC  10.3389/fpsyg.2013.00354         0         0         0   \n",
       "1  20181207  PMC  10.3389/fpsyg.2013.00370         0         0         0   \n",
       "2  20181207  PMC  10.3389/fnhum.2013.00304         0         0         0   \n",
       "3  20181210  PMC  10.3389/fmicb.2013.00161         0         0         0   \n",
       "4  20181210  PMC  10.3389/fnsys.2013.00020         0         0         0   \n",
       "\n",
       "   open_neuro  fmri  res    yr       pmcid  idx  \n",
       "0           0     0    7  2013  PMC3691502    0  \n",
       "1           0     0    9  2013  PMC3691503    1  \n",
       "2           0     2    5  2013  PMC3691504    2  \n",
       "3           0     0   10  2013  PMC3691505    3  \n",
       "4           0     1    2  2013  PMC3691506    4  "
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.DataFrame(git_links, columns=['date', 'src', 'doi', 'git_hits', \n",
    "                                      'osf_hits', 'nda_hits', 'open_neuro', \n",
    "                                      'fmri', 'res', 'yr', 'pmcid', 'idx'])\n",
    "df.to_csv('/home/riddleta/ac_knowl/output/bionlp_15.csv')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "2500\n",
      "5000\n",
      "7500\n",
      "10000\n",
      "12500\n",
      "15000\n",
      "17500\n",
      "20000\n",
      "22500\n"
     ]
    }
   ],
   "source": [
    "git_links = []\n",
    "missed = []\n",
    "progress = [i for i in range(0, 25000, 2500)]\n",
    "\n",
    "for i, j in enumerate(unique_ids[660000:]):\n",
    "    if i in progress:\n",
    "        print(i)\n",
    "    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + j + '/unicode'\n",
    "    try:\n",
    "        r = requests.get(pth)\n",
    "        git_links.append(return_useful_data(r) + (j, i))\n",
    "    except:\n",
    "        missed.append([i,j])\n",
    "    if i==25000:\n",
    "        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>date</th>\n",
       "      <th>src</th>\n",
       "      <th>doi</th>\n",
       "      <th>git_hits</th>\n",
       "      <th>osf_hits</th>\n",
       "      <th>nda_hits</th>\n",
       "      <th>open_neuro</th>\n",
       "      <th>fmri</th>\n",
       "      <th>res</th>\n",
       "      <th>yr</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>idx</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>20181121</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1371/journal.pone.0069952</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>2013</td>\n",
       "      <td>PMC3732294</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20181121</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1371/journal.pone.0069785</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>2013</td>\n",
       "      <td>PMC3732295</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20181121</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1371/journal.pone.0069488</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>2013</td>\n",
       "      <td>PMC3732296</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20181206</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.4161/biom.23284</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>2013</td>\n",
       "      <td>PMC3732320</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20181206</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.4161/biom.23799</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>2013</td>\n",
       "      <td>PMC3732321</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       date  src                           doi  git_hits  osf_hits  nda_hits  \\\n",
       "0  20181121  PMC  10.1371/journal.pone.0069952         0         0         0   \n",
       "1  20181121  PMC  10.1371/journal.pone.0069785         0         0         0   \n",
       "2  20181121  PMC  10.1371/journal.pone.0069488         0         0         0   \n",
       "3  20181206  PMC            10.4161/biom.23284         0         0         0   \n",
       "4  20181206  PMC            10.4161/biom.23799         0         0         0   \n",
       "\n",
       "   open_neuro  fmri  res    yr       pmcid  idx  \n",
       "0           0     0    9  2013  PMC3732294    0  \n",
       "1           0     0    5  2013  PMC3732295    1  \n",
       "2           0     0    6  2013  PMC3732296    2  \n",
       "3           0     0    3  2013  PMC3732320    3  \n",
       "4           0     0    9  2013  PMC3732321    4  "
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.DataFrame(git_links, columns=['date', 'src', 'doi', 'git_hits', \n",
    "                                      'osf_hits', 'nda_hits', 'open_neuro', \n",
    "                                      'fmri', 'res', 'yr', 'pmcid', 'idx'])\n",
    "df.to_csv('/home/riddleta/ac_knowl/output/bionlp_16.csv')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "2500\n",
      "5000\n",
      "7500\n",
      "10000\n",
      "12500\n",
      "15000\n",
      "17500\n",
      "20000\n",
      "22500\n",
      "25000\n",
      "27500\n"
     ]
    }
   ],
   "source": [
    "git_links = []\n",
    "missed = []\n",
    "progress = [i for i in range(0, 30000, 2500)]\n",
    "\n",
    "for i, j in enumerate(unique_ids[685000:]):\n",
    "    if i in progress:\n",
    "        print(i)\n",
    "    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + j + '/unicode'\n",
    "    try:\n",
    "        r = requests.get(pth)\n",
    "        git_links.append(return_useful_data(r) + (j, i))\n",
    "    except:\n",
    "        missed.append([i,j])\n",
    "    if i==30000:\n",
    "        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>date</th>\n",
       "      <th>src</th>\n",
       "      <th>doi</th>\n",
       "      <th>git_hits</th>\n",
       "      <th>osf_hits</th>\n",
       "      <th>nda_hits</th>\n",
       "      <th>open_neuro</th>\n",
       "      <th>fmri</th>\n",
       "      <th>res</th>\n",
       "      <th>yr</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>idx</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>20181202</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1155/2013/347242</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2013</td>\n",
       "      <td>PMC3782758</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20181202</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1155/2013/196409</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>2013</td>\n",
       "      <td>PMC3782759</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20181112</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1155/2013/278978</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>2013</td>\n",
       "      <td>PMC3782760</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20181112</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1155/2013/383685</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2013</td>\n",
       "      <td>PMC3782761</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20181113</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1155/2013/132759</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>2013</td>\n",
       "      <td>PMC3782762</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       date  src                  doi  git_hits  osf_hits  nda_hits  \\\n",
       "0  20181202  PMC  10.1155/2013/347242         0         0         0   \n",
       "1  20181202  PMC  10.1155/2013/196409         0         0         0   \n",
       "2  20181112  PMC  10.1155/2013/278978         0         0         0   \n",
       "3  20181112  PMC  10.1155/2013/383685         0         0         0   \n",
       "4  20181113  PMC  10.1155/2013/132759         0         0         0   \n",
       "\n",
       "   open_neuro  fmri  res    yr       pmcid  idx  \n",
       "0           0     0    2  2013  PMC3782758    0  \n",
       "1           0     0    7  2013  PMC3782759    1  \n",
       "2           0     0    6  2013  PMC3782760    2  \n",
       "3           0     0    2  2013  PMC3782761    3  \n",
       "4           0     0   15  2013  PMC3782762    4  "
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.DataFrame(git_links, columns=['date', 'src', 'doi', 'git_hits', \n",
    "                                      'osf_hits', 'nda_hits', 'open_neuro', \n",
    "                                      'fmri', 'res', 'yr', 'pmcid', 'idx'])\n",
    "df.to_csv('/home/riddleta/ac_knowl/output/bionlp_17.csv')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "5000\n",
      "10000\n",
      "15000\n",
      "20000\n",
      "25000\n",
      "30000\n",
      "35000\n",
      "40000\n",
      "45000\n",
      "50000\n",
      "55000\n",
      "60000\n",
      "65000\n",
      "70000\n",
      "75000\n",
      "80000\n",
      "85000\n",
      "90000\n",
      "95000\n",
      "100000\n",
      "105000\n",
      "110000\n",
      "115000\n",
      "120000\n",
      "125000\n",
      "130000\n",
      "135000\n",
      "140000\n",
      "145000\n",
      "150000\n",
      "155000\n",
      "160000\n",
      "165000\n",
      "170000\n"
     ]
    }
   ],
   "source": [
    "git_links = []\n",
    "missed = []\n",
    "progress = [i for i in range(0, 175000, 5000)]\n",
    "\n",
    "for i, j in enumerate(unique_ids[715000:]):\n",
    "    if i in progress:\n",
    "        print(i)\n",
    "    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + j + '/unicode'\n",
    "    try:\n",
    "        r = requests.get(pth)\n",
    "        git_links.append(return_useful_data(r) + (j, i))\n",
    "    except:\n",
    "        missed.append([i,j])\n",
    "    if i==175000:\n",
    "        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>date</th>\n",
       "      <th>src</th>\n",
       "      <th>doi</th>\n",
       "      <th>git_hits</th>\n",
       "      <th>osf_hits</th>\n",
       "      <th>nda_hits</th>\n",
       "      <th>open_neuro</th>\n",
       "      <th>fmri</th>\n",
       "      <th>res</th>\n",
       "      <th>yr</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>idx</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>20181124</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1155/2013/436932</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>2013</td>\n",
       "      <td>PMC3842064</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20181124</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1155/2013/791918</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>2013</td>\n",
       "      <td>PMC3842065</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20181124</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1155/2013/924276</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>2013</td>\n",
       "      <td>PMC3842066</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20181124</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1155/2013/367283</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>2013</td>\n",
       "      <td>PMC3842067</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20181124</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1155/2013/731346</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>2013</td>\n",
       "      <td>PMC3842068</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       date  src                  doi  git_hits  osf_hits  nda_hits  \\\n",
       "0  20181124  PMC  10.1155/2013/436932         0         0         0   \n",
       "1  20181124  PMC  10.1155/2013/791918         0         0         0   \n",
       "2  20181124  PMC  10.1155/2013/924276         0         0         0   \n",
       "3  20181124  PMC  10.1155/2013/367283         0         0         0   \n",
       "4  20181124  PMC  10.1155/2013/731346         0         0         0   \n",
       "\n",
       "   open_neuro  fmri  res    yr       pmcid  idx  \n",
       "0           0     0    3  2013  PMC3842064    0  \n",
       "1           0     0    8  2013  PMC3842065    1  \n",
       "2           0     0    5  2013  PMC3842066    2  \n",
       "3           0     0   15  2013  PMC3842067    3  \n",
       "4           0     0    3  2013  PMC3842068    4  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.DataFrame(git_links, columns=['date', 'src', 'doi', 'git_hits', \n",
    "                                      'osf_hits', 'nda_hits', 'open_neuro', \n",
    "                                      'fmri', 'res', 'yr', 'pmcid', 'idx'])\n",
    "df.to_csv('/home/riddleta/ac_knowl/output/bionlp_18.csv')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "2500\n",
      "5000\n",
      "7500\n",
      "10000\n",
      "12500\n",
      "15000\n",
      "17500\n",
      "20000\n",
      "22500\n",
      "25000\n",
      "27500\n",
      "30000\n",
      "32500\n"
     ]
    }
   ],
   "source": [
    "git_links = []\n",
    "missed = []\n",
    "progress = [i for i in range(0, 35000, 2500)]\n",
    "\n",
    "for i, j in enumerate(unique_ids[890000:]):\n",
    "    if i in progress:\n",
    "        print(i)\n",
    "    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + j + '/unicode'\n",
    "    try:\n",
    "        r = requests.get(pth)\n",
    "        git_links.append(return_useful_data(r) + (j, i))\n",
    "    except:\n",
    "        missed.append([i,j])\n",
    "    if i==35000:\n",
    "        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>date</th>\n",
       "      <th>src</th>\n",
       "      <th>doi</th>\n",
       "      <th>git_hits</th>\n",
       "      <th>osf_hits</th>\n",
       "      <th>nda_hits</th>\n",
       "      <th>open_neuro</th>\n",
       "      <th>fmri</th>\n",
       "      <th>res</th>\n",
       "      <th>yr</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>idx</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>20181210</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.5090/kjtcs.2014.47.3.317</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2014</td>\n",
       "      <td>PMC4157490</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20181113</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.5090/kjtcs.2014.47.3.320</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2014</td>\n",
       "      <td>PMC4157491</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20181113</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.5090/kjtcs.2014.47.3.325</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2014</td>\n",
       "      <td>PMC4157492</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20181113</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1177/2048004014539666</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2014</td>\n",
       "      <td>PMC4157493</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20181114</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.5090/kjtcs.2014.47.4.327</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>2014</td>\n",
       "      <td>PMC4157494</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       date  src                          doi  git_hits  osf_hits  nda_hits  \\\n",
       "0  20181210  PMC  10.5090/kjtcs.2014.47.3.317         0         0         0   \n",
       "1  20181113  PMC  10.5090/kjtcs.2014.47.3.320         0         0         0   \n",
       "2  20181113  PMC  10.5090/kjtcs.2014.47.3.325         0         0         0   \n",
       "3  20181113  PMC     10.1177/2048004014539666         0         0         0   \n",
       "4  20181114  PMC  10.5090/kjtcs.2014.47.4.327         0         0         0   \n",
       "\n",
       "   open_neuro  fmri  res    yr       pmcid  idx  \n",
       "0           0     0    2  2014  PMC4157490    0  \n",
       "1           0     0    0  2014  PMC4157491    1  \n",
       "2           0     0    0  2014  PMC4157492    2  \n",
       "3           0     0    2  2014  PMC4157493    3  \n",
       "4           0     0    3  2014  PMC4157494    4  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.DataFrame(git_links, columns=['date', 'src', 'doi', 'git_hits', \n",
    "                                      'osf_hits', 'nda_hits', 'open_neuro', \n",
    "                                      'fmri', 'res', 'yr', 'pmcid', 'idx'])\n",
    "df.to_csv('/home/riddleta/ac_knowl/output/bionlp_19.csv')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "2500\n",
      "5000\n",
      "7500\n",
      "10000\n",
      "12500\n",
      "15000\n",
      "17500\n",
      "20000\n",
      "22500\n",
      "25000\n",
      "27500\n"
     ]
    }
   ],
   "source": [
    "git_links = []\n",
    "missed = []\n",
    "progress = [i for i in range(0, 30000, 2500)]\n",
    "\n",
    "for i, j in enumerate(unique_ids[925000:]):\n",
    "    if i in progress:\n",
    "        print(i)\n",
    "    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + j + '/unicode'\n",
    "    try:\n",
    "        r = requests.get(pth)\n",
    "        git_links.append(return_useful_data(r) + (j, i))\n",
    "    except:\n",
    "        missed.append([i,j])\n",
    "    if i==30000:\n",
    "        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>date</th>\n",
       "      <th>src</th>\n",
       "      <th>doi</th>\n",
       "      <th>git_hits</th>\n",
       "      <th>osf_hits</th>\n",
       "      <th>nda_hits</th>\n",
       "      <th>open_neuro</th>\n",
       "      <th>fmri</th>\n",
       "      <th>res</th>\n",
       "      <th>yr</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>idx</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>20181123</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1371/journal.pone.0113017</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2014</td>\n",
       "      <td>PMC4218860</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20181115</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1371/journal.pone.0109125</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>2014</td>\n",
       "      <td>PMC4218861</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20181115</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1371/journal.pone.0109700</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>18</td>\n",
       "      <td>2014</td>\n",
       "      <td>PMC4218862</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20181120</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1113/expphysiol.2014.081190</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>2014</td>\n",
       "      <td>PMC4218865</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20181115</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1021/mp5003043</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>2014</td>\n",
       "      <td>PMC4218868</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       date  src                             doi  git_hits  osf_hits  \\\n",
       "0  20181123  PMC    10.1371/journal.pone.0113017         0         0   \n",
       "1  20181115  PMC    10.1371/journal.pone.0109125         0         0   \n",
       "2  20181115  PMC    10.1371/journal.pone.0109700         0         0   \n",
       "3  20181120  PMC  10.1113/expphysiol.2014.081190         0         0   \n",
       "4  20181115  PMC               10.1021/mp5003043         0         0   \n",
       "\n",
       "   nda_hits  open_neuro  fmri  res    yr       pmcid  idx  \n",
       "0         0           0     0    0  2014  PMC4218860    0  \n",
       "1         0           0     0    8  2014  PMC4218861    1  \n",
       "2         0           0     1   18  2014  PMC4218862    2  \n",
       "3         0           0     0    5  2014  PMC4218865    3  \n",
       "4         0           0     0    9  2014  PMC4218868    4  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.DataFrame(git_links, columns=['date', 'src', 'doi', 'git_hits', \n",
    "                                      'osf_hits', 'nda_hits', 'open_neuro', \n",
    "                                      'fmri', 'res', 'yr', 'pmcid', 'idx'])\n",
    "df.to_csv('/home/riddleta/ac_knowl/output/bionlp_20.csv')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "10000\n",
      "20000\n",
      "30000\n"
     ]
    }
   ],
   "source": [
    "git_links = []\n",
    "missed = []\n",
    "progress = [i for i in range(0, 35000, 10000)]\n",
    "\n",
    "for i, j in enumerate(unique_ids[955000:]):\n",
    "    if i in progress:\n",
    "        print(i)\n",
    "    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + j + '/unicode'\n",
    "    try:\n",
    "        r = requests.get(pth)\n",
    "        git_links.append(return_useful_data(r) + (j, i))\n",
    "    except:\n",
    "        missed.append([i,j])\n",
    "    if i==35000:\n",
    "        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>date</th>\n",
       "      <th>src</th>\n",
       "      <th>doi</th>\n",
       "      <th>git_hits</th>\n",
       "      <th>osf_hits</th>\n",
       "      <th>nda_hits</th>\n",
       "      <th>open_neuro</th>\n",
       "      <th>fmri</th>\n",
       "      <th>res</th>\n",
       "      <th>yr</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>idx</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>20181124</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.5812/ircmj.15308</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>2014</td>\n",
       "      <td>PMC4270654</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20181124</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.5812/ircmj.21835</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>2014</td>\n",
       "      <td>PMC4270655</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20181124</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.5812/ircmj.15312</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>2014</td>\n",
       "      <td>PMC4270656</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20181123</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.5812/ircmj.14919</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2014</td>\n",
       "      <td>PMC4270657</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20181124</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.5812/ircmj.15164</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2014</td>\n",
       "      <td>PMC4270658</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       date  src                  doi  git_hits  osf_hits  nda_hits  \\\n",
       "0  20181124  PMC  10.5812/ircmj.15308         0         0         0   \n",
       "1  20181124  PMC  10.5812/ircmj.21835         0         0         0   \n",
       "2  20181124  PMC  10.5812/ircmj.15312         0         0         0   \n",
       "3  20181123  PMC  10.5812/ircmj.14919         0         0         0   \n",
       "4  20181124  PMC  10.5812/ircmj.15164         0         0         0   \n",
       "\n",
       "   open_neuro  fmri  res    yr       pmcid  idx  \n",
       "0           0     0   10  2014  PMC4270654    0  \n",
       "1           0     0    3  2014  PMC4270655    1  \n",
       "2           0     0    4  2014  PMC4270656    2  \n",
       "3           0     0    2  2014  PMC4270657    3  \n",
       "4           0     0    0  2014  PMC4270658    4  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.DataFrame(git_links, columns=['date', 'src', 'doi', 'git_hits', \n",
    "                                      'osf_hits', 'nda_hits', 'open_neuro', \n",
    "                                      'fmri', 'res', 'yr', 'pmcid', 'idx'])\n",
    "df.to_csv('/home/riddleta/ac_knowl/output/bionlp_21.csv')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "5000\n",
      "10000\n",
      "15000\n",
      "20000\n"
     ]
    }
   ],
   "source": [
    "git_links = []\n",
    "missed = []\n",
    "progress = [i for i in range(0, 25000, 5000)]\n",
    "\n",
    "for i, j in enumerate(unique_ids[990000:]):\n",
    "    if i in progress:\n",
    "        print(i)\n",
    "    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + j + '/unicode'\n",
    "    try:\n",
    "        r = requests.get(pth)\n",
    "        git_links.append(return_useful_data(r) + (j, i))\n",
    "    except:\n",
    "        missed.append([i,j])\n",
    "    if i==25000:\n",
    "        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>date</th>\n",
       "      <th>src</th>\n",
       "      <th>doi</th>\n",
       "      <th>git_hits</th>\n",
       "      <th>osf_hits</th>\n",
       "      <th>nda_hits</th>\n",
       "      <th>open_neuro</th>\n",
       "      <th>fmri</th>\n",
       "      <th>res</th>\n",
       "      <th>yr</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>idx</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>20181201</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.4103/0973-1075.150164</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2015</td>\n",
       "      <td>PMC4332122</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20181123</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.4103/0973-1075.150170</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>2015</td>\n",
       "      <td>PMC4332123</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20181201</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.4103/0973-1075.150148</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2015</td>\n",
       "      <td>PMC4332124</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20181201</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.4103/0973-1075.150172</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2015</td>\n",
       "      <td>PMC4332125</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20181201</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.4103/0973-1075.150175</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>2015</td>\n",
       "      <td>PMC4332126</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       date  src                       doi  git_hits  osf_hits  nda_hits  \\\n",
       "0  20181201  PMC  10.4103/0973-1075.150164         0         0         0   \n",
       "1  20181123  PMC  10.4103/0973-1075.150170         0         0         0   \n",
       "2  20181201  PMC  10.4103/0973-1075.150148         0         0         0   \n",
       "3  20181201  PMC  10.4103/0973-1075.150172         0         0         0   \n",
       "4  20181201  PMC  10.4103/0973-1075.150175         0         0         0   \n",
       "\n",
       "   open_neuro  fmri  res    yr       pmcid  idx  \n",
       "0           0     0    0  2015  PMC4332122    0  \n",
       "1           0     0    4  2015  PMC4332123    1  \n",
       "2           0     0    2  2015  PMC4332124    2  \n",
       "3           0     0    2  2015  PMC4332125    3  \n",
       "4           0     0    3  2015  PMC4332126    4  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.DataFrame(git_links, columns=['date', 'src', 'doi', 'git_hits', \n",
    "                                      'osf_hits', 'nda_hits', 'open_neuro', \n",
    "                                      'fmri', 'res', 'yr', 'pmcid', 'idx'])\n",
    "df.to_csv('/home/riddleta/ac_knowl/output/bionlp_22.csv')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "10000\n",
      "20000\n",
      "30000\n",
      "40000\n",
      "50000\n",
      "60000\n",
      "70000\n",
      "80000\n",
      "90000\n",
      "100000\n",
      "110000\n",
      "120000\n",
      "130000\n",
      "140000\n",
      "150000\n",
      "160000\n",
      "170000\n",
      "180000\n",
      "190000\n"
     ]
    }
   ],
   "source": [
    "git_links = []\n",
    "missed = []\n",
    "progress = [i for i in range(0, 200000, 10000)]\n",
    "\n",
    "for i, j in enumerate(unique_ids[1015000:]):\n",
    "    if i in progress:\n",
    "        print(i)\n",
    "    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + j + '/unicode'\n",
    "    try:\n",
    "        r = requests.get(pth)\n",
    "        git_links.append(return_useful_data(r) + (j, i))\n",
    "    except:\n",
    "        missed.append([i,j])\n",
    "    if i==200000:\n",
    "        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>date</th>\n",
       "      <th>src</th>\n",
       "      <th>doi</th>\n",
       "      <th>git_hits</th>\n",
       "      <th>osf_hits</th>\n",
       "      <th>nda_hits</th>\n",
       "      <th>open_neuro</th>\n",
       "      <th>fmri</th>\n",
       "      <th>res</th>\n",
       "      <th>yr</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>idx</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>20190105</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1080/14780887.2014.948697</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2015</td>\n",
       "      <td>PMC4376240</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20181121</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1080/14427591.2014.882250</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2014</td>\n",
       "      <td>PMC4376241</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20190105</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1080/15299732.2015.995021</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2015</td>\n",
       "      <td>PMC4376242</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20181121</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1002/sia.5429</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>2014</td>\n",
       "      <td>PMC4376248</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20181121</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1002/sia.5376</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>2014</td>\n",
       "      <td>PMC4376249</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       date  src                           doi  git_hits  osf_hits  nda_hits  \\\n",
       "0  20190105  PMC  10.1080/14780887.2014.948697         0         0         0   \n",
       "1  20181121  PMC  10.1080/14427591.2014.882250         0         0         0   \n",
       "2  20190105  PMC  10.1080/15299732.2015.995021         0         0         0   \n",
       "3  20181121  PMC              10.1002/sia.5429         0         0         0   \n",
       "4  20181121  PMC              10.1002/sia.5376         0         0         0   \n",
       "\n",
       "   open_neuro  fmri  res    yr       pmcid  idx  \n",
       "0           0     0    0  2015  PMC4376240    0  \n",
       "1           0     0    1  2014  PMC4376241    1  \n",
       "2           0     0    0  2015  PMC4376242    2  \n",
       "3           0     0    8  2014  PMC4376248    3  \n",
       "4           0     0    5  2014  PMC4376249    4  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.DataFrame(git_links, columns=['date', 'src', 'doi', 'git_hits', \n",
    "                                      'osf_hits', 'nda_hits', 'open_neuro', \n",
    "                                      'fmri', 'res', 'yr', 'pmcid', 'idx'])\n",
    "df.to_csv('/home/riddleta/ac_knowl/output/bionlp_23.csv')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "10000\n",
      "20000\n",
      "30000\n"
     ]
    }
   ],
   "source": [
    "git_links = []\n",
    "missed = []\n",
    "progress = [i for i in range(0, 40000, 10000)]\n",
    "\n",
    "for i, j in enumerate(unique_ids[1215000:]):\n",
    "    if i in progress:\n",
    "        print(i)\n",
    "    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + j + '/unicode'\n",
    "    try:\n",
    "        r = requests.get(pth)\n",
    "        git_links.append(return_useful_data(r) + (j, i))\n",
    "    except:\n",
    "        missed.append([i,j])\n",
    "    if i==40000:\n",
    "        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>date</th>\n",
       "      <th>src</th>\n",
       "      <th>doi</th>\n",
       "      <th>git_hits</th>\n",
       "      <th>osf_hits</th>\n",
       "      <th>nda_hits</th>\n",
       "      <th>open_neuro</th>\n",
       "      <th>fmri</th>\n",
       "      <th>res</th>\n",
       "      <th>yr</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>idx</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>20181130</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1038/nature14984</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>2016</td>\n",
       "      <td>PMC4713039</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20181112</td>\n",
       "      <td>PMC</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2016</td>\n",
       "      <td>PMC4713040</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20181112</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1038/srep19264</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>2016</td>\n",
       "      <td>PMC4713048</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20181112</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1038/srep19260</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>2016</td>\n",
       "      <td>PMC4713049</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20181207</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1038/srep19256</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>2016</td>\n",
       "      <td>PMC4713050</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       date  src                  doi  git_hits  osf_hits  nda_hits  \\\n",
       "0  20181130  PMC  10.1038/nature14984         0         0         0   \n",
       "1  20181112  PMC                 None         0         0         0   \n",
       "2  20181112  PMC    10.1038/srep19264         0         0         0   \n",
       "3  20181112  PMC    10.1038/srep19260         0         0         0   \n",
       "4  20181207  PMC    10.1038/srep19256         0         0         0   \n",
       "\n",
       "   open_neuro  fmri  res    yr       pmcid  idx  \n",
       "0           0     0    7  2016  PMC4713039    0  \n",
       "1           0     0    0  2016  PMC4713040    1  \n",
       "2           0     0    4  2016  PMC4713048    2  \n",
       "3           0     0    4  2016  PMC4713049    3  \n",
       "4           0     0   16  2016  PMC4713050    4  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.DataFrame(git_links, columns=['date', 'src', 'doi', 'git_hits', \n",
    "                                      'osf_hits', 'nda_hits', 'open_neuro', \n",
    "                                      'fmri', 'res', 'yr', 'pmcid', 'idx'])\n",
    "df.to_csv('/home/riddleta/ac_knowl/output/bionlp_24.csv')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "10000\n",
      "20000\n",
      "30000\n",
      "40000\n"
     ]
    }
   ],
   "source": [
    "git_links = []\n",
    "missed = []\n",
    "progress = [i for i in range(0, 45000, 10000)]\n",
    "\n",
    "for i, j in enumerate(unique_ids[1255000:]):\n",
    "    if i in progress:\n",
    "        print(i)\n",
    "    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + j + '/unicode'\n",
    "    try:\n",
    "        r = requests.get(pth)\n",
    "        git_links.append(return_useful_data(r) + (j, i))\n",
    "    except:\n",
    "        missed.append([i,j])\n",
    "    if i==45000:\n",
    "        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>date</th>\n",
       "      <th>src</th>\n",
       "      <th>doi</th>\n",
       "      <th>git_hits</th>\n",
       "      <th>osf_hits</th>\n",
       "      <th>nda_hits</th>\n",
       "      <th>open_neuro</th>\n",
       "      <th>fmri</th>\n",
       "      <th>res</th>\n",
       "      <th>yr</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>idx</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>20181121</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1186/s40064-016-1758-8</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2016</td>\n",
       "      <td>PMC4777971</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20190111</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1186/s40064-016-1933-y</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2016</td>\n",
       "      <td>PMC4777974</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20181121</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1186/s40064-016-1890-5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>2016</td>\n",
       "      <td>PMC4777976</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20181121</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1186/s40064-016-1774-8</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>2016</td>\n",
       "      <td>PMC4777977</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20181204</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1186/s13613-016-0121-0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>2016</td>\n",
       "      <td>PMC4777978</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       date  src                        doi  git_hits  osf_hits  nda_hits  \\\n",
       "0  20181121  PMC  10.1186/s40064-016-1758-8         0         0         0   \n",
       "1  20190111  PMC  10.1186/s40064-016-1933-y         0         0         0   \n",
       "2  20181121  PMC  10.1186/s40064-016-1890-5         0         0         0   \n",
       "3  20181121  PMC  10.1186/s40064-016-1774-8         0         0         0   \n",
       "4  20181204  PMC  10.1186/s13613-016-0121-0         0         0         0   \n",
       "\n",
       "   open_neuro  fmri  res    yr       pmcid  idx  \n",
       "0           0     0    1  2016  PMC4777971    0  \n",
       "1           0     0    1  2016  PMC4777974    1  \n",
       "2           0     0    3  2016  PMC4777976    2  \n",
       "3           0     0    9  2016  PMC4777977    3  \n",
       "4           0     0    3  2016  PMC4777978    4  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.DataFrame(git_links, columns=['date', 'src', 'doi', 'git_hits', \n",
    "                                      'osf_hits', 'nda_hits', 'open_neuro', \n",
    "                                      'fmri', 'res', 'yr', 'pmcid', 'idx'])\n",
    "df.to_csv('/home/riddleta/ac_knowl/output/bionlp_25.csv')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "10000\n",
      "20000\n",
      "30000\n",
      "40000\n",
      "50000\n",
      "60000\n",
      "70000\n",
      "80000\n",
      "90000\n",
      "100000\n",
      "110000\n",
      "120000\n",
      "130000\n",
      "140000\n",
      "150000\n",
      "160000\n",
      "170000\n",
      "180000\n",
      "190000\n",
      "200000\n",
      "210000\n",
      "220000\n"
     ]
    }
   ],
   "source": [
    "git_links = []\n",
    "missed = []\n",
    "progress = [i for i in range(0, 225000, 10000)]\n",
    "\n",
    "for i, j in enumerate(unique_ids[1300000:]):\n",
    "    if i in progress:\n",
    "        print(i)\n",
    "    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + j + '/unicode'\n",
    "    try:\n",
    "        r = requests.get(pth)\n",
    "        git_links.append(return_useful_data(r) + (j, i))\n",
    "    except:\n",
    "        missed.append([i,j])\n",
    "    if i==225000:\n",
    "        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>date</th>\n",
       "      <th>src</th>\n",
       "      <th>doi</th>\n",
       "      <th>git_hits</th>\n",
       "      <th>osf_hits</th>\n",
       "      <th>nda_hits</th>\n",
       "      <th>open_neuro</th>\n",
       "      <th>fmri</th>\n",
       "      <th>res</th>\n",
       "      <th>yr</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>idx</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>20190218</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1371/journal.pone.0149955</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>2016</td>\n",
       "      <td>PMC4854442</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20190218</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1371/journal.pone.0154629</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>2016</td>\n",
       "      <td>PMC4854444</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20190219</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1371/journal.pone.0154324</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>2016</td>\n",
       "      <td>PMC4854445</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20181124</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1016/j.jhealeco.2015.09.007</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>19</td>\n",
       "      <td>2016</td>\n",
       "      <td>PMC4854446</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20190218</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1371/journal.pone.0154763</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>2016</td>\n",
       "      <td>PMC4854447</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       date  src                             doi  git_hits  osf_hits  \\\n",
       "0  20190218  PMC    10.1371/journal.pone.0149955         0         0   \n",
       "1  20190218  PMC    10.1371/journal.pone.0154629         0         0   \n",
       "2  20190219  PMC    10.1371/journal.pone.0154324         3         0   \n",
       "3  20181124  PMC  10.1016/j.jhealeco.2015.09.007         0         0   \n",
       "4  20190218  PMC    10.1371/journal.pone.0154763         0         0   \n",
       "\n",
       "   nda_hits  open_neuro  fmri  res    yr       pmcid  idx  \n",
       "0         0           0     0    5  2016  PMC4854442    0  \n",
       "1         0           0     0    6  2016  PMC4854444    1  \n",
       "2         0           0     0    6  2016  PMC4854445    2  \n",
       "3         0           0     0   19  2016  PMC4854446    3  \n",
       "4         0           0     0    3  2016  PMC4854447    4  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.DataFrame(git_links, columns=['date', 'src', 'doi', 'git_hits', \n",
    "                                      'osf_hits', 'nda_hits', 'open_neuro', \n",
    "                                      'fmri', 'res', 'yr', 'pmcid', 'idx'])\n",
    "df.to_csv('/home/riddleta/ac_knowl/output/bionlp_25.csv')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "10000\n",
      "20000\n",
      "30000\n",
      "40000\n"
     ]
    }
   ],
   "source": [
    "git_links = []\n",
    "missed = []\n",
    "progress = [i for i in range(0, 50000, 10000)]\n",
    "\n",
    "for i, j in enumerate(unique_ids[1525000:]):\n",
    "    if i in progress:\n",
    "        print(i)\n",
    "    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + j + '/unicode'\n",
    "    try:\n",
    "        r = requests.get(pth)\n",
    "        git_links.append(return_useful_data(r) + (j, i))\n",
    "    except:\n",
    "        missed.append([i,j])\n",
    "    if i==50000:\n",
    "        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>date</th>\n",
       "      <th>src</th>\n",
       "      <th>doi</th>\n",
       "      <th>git_hits</th>\n",
       "      <th>osf_hits</th>\n",
       "      <th>nda_hits</th>\n",
       "      <th>open_neuro</th>\n",
       "      <th>fmri</th>\n",
       "      <th>res</th>\n",
       "      <th>yr</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>idx</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>20181207</td>\n",
       "      <td>PMC</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1927</td>\n",
       "      <td>PMC5197919</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20181111</td>\n",
       "      <td>PMC</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1927</td>\n",
       "      <td>PMC5197920</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20181123</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1017/S0033291716002166</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>2016</td>\n",
       "      <td>PMC5197921</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20181206</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1017/S0963180116000712</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2017</td>\n",
       "      <td>PMC5197922</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20181206</td>\n",
       "      <td>PMC</td>\n",
       "      <td>10.1017/S0007114516003688</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>2016</td>\n",
       "      <td>PMC5197923</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       date  src                        doi  git_hits  osf_hits  nda_hits  \\\n",
       "0  20181207  PMC                       None         0         0         0   \n",
       "1  20181111  PMC                       None         0         0         0   \n",
       "2  20181123  PMC  10.1017/S0033291716002166         0         0         0   \n",
       "3  20181206  PMC  10.1017/S0963180116000712         0         0         0   \n",
       "4  20181206  PMC  10.1017/S0007114516003688         0         0         0   \n",
       "\n",
       "   open_neuro  fmri  res    yr       pmcid  idx  \n",
       "0           0     0    0  1927  PMC5197919    0  \n",
       "1           0     0    0  1927  PMC5197920    1  \n",
       "2           0     0    6  2016  PMC5197921    2  \n",
       "3           0     0    0  2017  PMC5197922    3  \n",
       "4           0     0   15  2016  PMC5197923    4  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.DataFrame(git_links, columns=['date', 'src', 'doi', 'git_hits', \n",
    "                                      'osf_hits', 'nda_hits', 'open_neuro', \n",
    "                                      'fmri', 'res', 'yr', 'pmcid', 'idx'])\n",
    "df.to_csv('/home/riddleta/ac_knowl/output/bionlp_26.csv')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "5000\n",
      "10000\n",
      "15000\n",
      "20000\n"
     ]
    }
   ],
   "source": [
    "git_links = []\n",
    "missed = []\n",
    "progress = [i for i in range(0, 25000, 5000)]\n",
    "\n",
    "for i, j in enumerate(unique_ids[1575000:]):\n",
    "    if i in progress:\n",
    "        print(i)\n",
    "    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + j + '/unicode'\n",
    "    try:\n",
    "        r = requests.get(pth)\n",
    "        git_links.append(return_useful_data(r) + (j, i))\n",
    "    except:\n",
    "        missed.append([i,j])\n",
    "    if i==25000:\n",
    "        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>date</th>\n",
       "      <th>src</th>\n",
       "      <th>doi</th>\n",
       "      <th>git_hits</th>\n",
       "      <th>osf_hits</th>\n",
       "      <th>nda_hits</th>\n",
       "      <th>open_neuro</th>\n",
       "      <th>fmri</th>\n",
       "      <th>res</th>\n",
       "      <th>yr</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>idx</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>20181115</td>\n",
       "      <td>PMC</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1894</td>\n",
       "      <td>PMC5263745</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20181115</td>\n",
       "      <td>PMC</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1894</td>\n",
       "      <td>PMC5263747</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20181116</td>\n",
       "      <td>PMC</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1894</td>\n",
       "      <td>PMC5263748</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20181116</td>\n",
       "      <td>PMC</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1894</td>\n",
       "      <td>PMC5263749</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20181106</td>\n",
       "      <td>PMC</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1894</td>\n",
       "      <td>PMC5263750</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       date  src   doi  git_hits  osf_hits  nda_hits  open_neuro  fmri  res  \\\n",
       "0  20181115  PMC  None         0         0         0           0     0    0   \n",
       "1  20181115  PMC  None         0         0         0           0     0    0   \n",
       "2  20181116  PMC  None         0         0         0           0     0    0   \n",
       "3  20181116  PMC  None         0         0         0           0     0    0   \n",
       "4  20181106  PMC  None         0         0         0           0     0    0   \n",
       "\n",
       "     yr       pmcid  idx  \n",
       "0  1894  PMC5263745    0  \n",
       "1  1894  PMC5263747    1  \n",
       "2  1894  PMC5263748    2  \n",
       "3  1894  PMC5263749    3  \n",
       "4  1894  PMC5263750    4  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.DataFrame(git_links, columns=['date', 'src', 'doi', 'git_hits', \n",
    "                                      'osf_hits', 'nda_hits', 'open_neuro', \n",
    "                                      'fmri', 'res', 'yr', 'pmcid', 'idx'])\n",
    "df.to_csv('/home/riddleta/ac_knowl/output/bionlp_27.csv')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "10000\n",
      "20000\n",
      "30000\n",
      "40000\n",
      "50000\n",
      "60000\n",
      "70000\n",
      "80000\n",
      "90000\n",
      "100000\n",
      "110000\n",
      "120000\n",
      "130000\n",
      "140000\n",
      "150000\n",
      "160000\n",
      "170000\n",
      "180000\n",
      "190000\n",
      "200000\n",
      "210000\n",
      "220000\n",
      "230000\n",
      "240000\n",
      "0\n",
      "10000\n",
      "20000\n",
      "30000\n",
      "40000\n",
      "50000\n",
      "60000\n",
      "70000\n",
      "80000\n",
      "90000\n",
      "100000\n",
      "110000\n",
      "120000\n",
      "130000\n",
      "140000\n",
      "150000\n",
      "160000\n",
      "170000\n",
      "180000\n",
      "190000\n",
      "200000\n",
      "210000\n",
      "220000\n",
      "230000\n",
      "240000\n",
      "0\n",
      "10000\n",
      "20000\n",
      "30000\n",
      "40000\n",
      "50000\n",
      "60000\n",
      "70000\n",
      "80000\n",
      "90000\n",
      "100000\n",
      "110000\n",
      "120000\n",
      "130000\n",
      "140000\n",
      "150000\n",
      "160000\n",
      "170000\n",
      "180000\n",
      "190000\n",
      "200000\n",
      "210000\n",
      "220000\n",
      "230000\n",
      "240000\n"
     ]
    }
   ],
   "source": [
    "git_links = []\n",
    "missed = []\n",
    "progress = [i for i in range(0, 250000, 10000)]\n",
    "\n",
    "for i, j in enumerate(unique_ids[1600000:]):\n",
    "    if i in progress:\n",
    "        print(i)\n",
    "    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + j + '/unicode'\n",
    "    try:\n",
    "        r = requests.get(pth)\n",
    "        git_links.append(return_useful_data(r) + (j, i))\n",
    "    except:\n",
    "        missed.append([i,j])\n",
    "    if i==250000:\n",
    "        break\n",
    "\n",
    "df = pd.DataFrame(git_links, columns=['date', 'src', 'doi', 'git_hits', \n",
    "                                      'osf_hits', 'nda_hits', 'open_neuro', \n",
    "                                      'fmri', 'res', 'yr', 'pmcid', 'idx'])\n",
    "df.to_csv('/home/riddleta/ac_knowl/output/bionlp_28.csv')\n",
    "\n",
    "git_links = []\n",
    "missed = []\n",
    "progress = [i for i in range(0, 250000, 10000)]\n",
    "\n",
    "for i, j in enumerate(unique_ids[1850000:]):\n",
    "    if i in progress:\n",
    "        print(i)\n",
    "    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + j + '/unicode'\n",
    "    try:\n",
    "        r = requests.get(pth)\n",
    "        git_links.append(return_useful_data(r) + (j, i))\n",
    "    except:\n",
    "        missed.append([i,j])\n",
    "    if i==250000:\n",
    "        break\n",
    "\n",
    "df = pd.DataFrame(git_links, columns=['date', 'src', 'doi', 'git_hits', \n",
    "                                      'osf_hits', 'nda_hits', 'open_neuro', \n",
    "                                      'fmri', 'res', 'yr', 'pmcid', 'idx'])\n",
    "df.to_csv('/home/riddleta/ac_knowl/output/bionlp_29.csv')\n",
    "\n",
    "git_links = []\n",
    "missed = []\n",
    "progress = [i for i in range(0, 250000, 10000)]\n",
    "\n",
    "for i, j in enumerate(unique_ids[2125000:]):\n",
    "    if i in progress:\n",
    "        print(i)\n",
    "    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + j + '/unicode'\n",
    "    try:\n",
    "        r = requests.get(pth)\n",
    "        git_links.append(return_useful_data(r) + (j, i))\n",
    "    except:\n",
    "        missed.append([i,j])\n",
    "    if i==250000:\n",
    "        break\n",
    "\n",
    "df = pd.DataFrame(git_links, columns=['date', 'src', 'doi', 'git_hits', \n",
    "                                      'osf_hits', 'nda_hits', 'open_neuro', \n",
    "                                      'fmri', 'res', 'yr', 'pmcid', 'idx'])\n",
    "df.to_csv('/home/riddleta/ac_knowl/output/bionlp_30.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "5000\n",
      "10000\n",
      "15000\n",
      "20000\n",
      "25000\n",
      "30000\n",
      "35000\n",
      "40000\n",
      "45000\n",
      "50000\n"
     ]
    }
   ],
   "source": [
    "git_links = []\n",
    "missed = []\n",
    "progress = [i for i in range(0, 55000, 5000)]\n",
    "\n",
    "for i, j in enumerate(unique_ids[2375000:]):\n",
    "    if i in progress:\n",
    "        print(i)\n",
    "    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + j + '/unicode'\n",
    "    try:\n",
    "        r = requests.get(pth)\n",
    "        git_links.append(return_useful_data(r) + (j, i))\n",
    "    except:\n",
    "        missed.append([i,j])\n",
    "    if i==55000:\n",
    "        break\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.DataFrame(git_links, columns=['date', 'src', 'doi', 'git_hits', \n",
    "                                      'osf_hits', 'nda_hits', 'open_neuro', \n",
    "                                      'fmri', 'res', 'yr', 'pmcid', 'idx'])\n",
    "df.to_csv('/home/riddleta/ac_knowl/output/bionlp_31.csv')\n",
    "#df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "10000\n",
      "20000\n",
      "30000\n",
      "40000\n",
      "50000\n",
      "60000\n",
      "70000\n",
      "80000\n",
      "90000\n",
      "100000\n",
      "110000\n",
      "120000\n",
      "130000\n",
      "140000\n",
      "150000\n",
      "160000\n",
      "170000\n",
      "180000\n",
      "190000\n",
      "200000\n",
      "210000\n",
      "220000\n",
      "230000\n",
      "240000\n",
      "250000\n",
      "260000\n",
      "270000\n"
     ]
    }
   ],
   "source": [
    "git_links = []\n",
    "missed = []\n",
    "progress = [i for i in range(0, 275000, 10000)]\n",
    "\n",
    "for i, j in enumerate(unique_ids[2430000:]):\n",
    "    if i in progress:\n",
    "        print(i)\n",
    "    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + j + '/unicode'\n",
    "    try:\n",
    "        r = requests.get(pth)\n",
    "        git_links.append(return_useful_data(r) + (j, i))\n",
    "    except:\n",
    "        missed.append([i,j])\n",
    "    if i==275000:\n",
    "        break\n",
    "\n",
    "df = pd.DataFrame(git_links, columns=['date', 'src', 'doi', 'git_hits', \n",
    "                                      'osf_hits', 'nda_hits', 'open_neuro', \n",
    "                                      'fmri', 'res', 'yr', 'pmcid', 'idx'])\n",
    "df.to_csv('/home/riddleta/ac_knowl/output/bionlp_32.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "10000\n",
      "20000\n",
      "30000\n",
      "40000\n",
      "50000\n",
      "60000\n",
      "70000\n",
      "80000\n",
      "90000\n",
      "100000\n",
      "110000\n",
      "120000\n",
      "130000\n",
      "140000\n",
      "150000\n",
      "160000\n",
      "170000\n",
      "180000\n",
      "190000\n",
      "200000\n",
      "210000\n",
      "220000\n",
      "230000\n",
      "240000\n",
      "250000\n",
      "260000\n",
      "270000\n"
     ]
    }
   ],
   "source": [
    "git_links = []\n",
    "missed = []\n",
    "progress = [i for i in range(0, 275000, 10000)]\n",
    "\n",
    "for i, j in enumerate(unique_ids[2704000:]):\n",
    "    if i in progress:\n",
    "        print(i)\n",
    "    pth = 'https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/' + j + '/unicode'\n",
    "    try:\n",
    "        r = requests.get(pth)\n",
    "        git_links.append(return_useful_data(r) + (j, i))\n",
    "    except:\n",
    "        missed.append([i,j])\n",
    "    if i==275000:\n",
    "        break\n",
    "\n",
    "df = pd.DataFrame(git_links, columns=['date', 'src', 'doi', 'git_hits', \n",
    "                                      'osf_hits', 'nda_hits', 'open_neuro', \n",
    "                                      'fmri', 'res', 'yr', 'pmcid', 'idx'])\n",
    "df.to_csv('/home/riddleta/ac_knowl/output/bionlp_33.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2975022"
      ]
     },
     "execution_count": 133,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(unique_ids)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2011"
      ]
     },
     "execution_count": 150,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmpyr = '2011'\n",
    "int(tmpyr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "github_ex = re.compile('github')\n",
    "re.search(github_ex, r_out['documents'][0]['passages'][0]['text'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "('20181121', 'PMC', 0, 0, 0, 0, 0, 'PMC126257', 350)"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "return_useful_data(r) + (j, i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'20181121'"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "r_out['date']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'resolution of airway inflammation and hyperreactivity after in vivo transfer of cd4+cd25+ regulatory t cells is interleukin 10 dependent'"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "r_out['documents'][0]['passages'][0]['text'].lower()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "temp_txt = 'resolution of airway inflammation and hyperreactivity after in vivo transfer of cd4+cd25+ regulatory t cells is interleukin 10 dependent. we also posted our data on github.com'\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<re.Match object; span=(165, 171), match='github'>"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "re.search(re.compile('github'), temp_txt.lower())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "r = requests.get('https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/PMC1350743/unicode')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "r_out = r.json()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['date', 'source', 'infons', 'documents', 'key'])"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "r_out.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "r = requests.get('https://www.ncbi.nlm.nih.gov/research/bionlp/RESTful/pmcoa.cgi/BioC_json/PMC1249490/unicode')\n",
    "r_out2 = r.json()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "out_dat = [r_out, r_out2]\n",
    "len(out_dat)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['passages', 'infons', 'id', 'relations'])"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "r_out['documents'][0].keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'text': 'Resolution of airway inflammation and hyperreactivity after in vivo transfer of CD4+CD25+ regulatory T cells is interleukin 10 dependent',\n",
       " 'offset': 0,\n",
       " 'relations': [],\n",
       " 'infons': {'name_2': 'surname:Robinson;given-names:Douglas S.',\n",
       "  'name_3': 'surname:Lloyd;given-names:Clare M.',\n",
       "  'name_0': 'surname:Kearley;given-names:Jennifer',\n",
       "  'name_1': 'surname:Barker;given-names:Jane E.',\n",
       "  'license': 'This article is distributed under the terms of an Attributionâ€“Noncommercialâ€“Share Alikeâ€“No Mirror Sites license for the first six months after the publication date (see http://www.rupress.org/terms). After six months it is available under a Creative Commons License (Attributionâ€“Noncommercialâ€“Share Alike 4.0 Unported license, as described at http://creativecommons.org/licenses/by-nc-sa/4.0/).',\n",
       "  'article-id_publisher-id': '20051166',\n",
       "  'issue': '11',\n",
       "  'article-id_doi': '10.1084/jem.20051166',\n",
       "  'section_type': 'TITLE',\n",
       "  'article-id_pmid': '16314435',\n",
       "  'lpage': '1547',\n",
       "  'article-id_pmc': '1350743',\n",
       "  'fpage': '1539',\n",
       "  'year': '2005',\n",
       "  'type': 'front',\n",
       "  'fn': 'Abbreviations used: AHR, airway hyperreactivity; BAL, bronchoalveolar lavage; Cdyn, dynamic compliance; RL, lung resistance.',\n",
       "  'volume': '202'},\n",
       " 'sentences': [],\n",
       " 'annotations': []}"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "r_out['documents'][0]['passages'][0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TITLE\n",
      "ABSTRACT\n",
      "INTRO\n",
      "INTRO\n",
      "INTRO\n",
      "RESULTS\n",
      "RESULTS\n",
      "RESULTS\n",
      "RESULTS\n",
      "RESULTS\n",
      "RESULTS\n",
      "FIG\n",
      "RESULTS\n",
      "RESULTS\n",
      "RESULTS\n",
      "FIG\n",
      "RESULTS\n",
      "RESULTS\n",
      "RESULTS\n",
      "RESULTS\n",
      "RESULTS\n",
      "RESULTS\n",
      "FIG\n",
      "RESULTS\n",
      "RESULTS\n",
      "FIG\n",
      "RESULTS\n",
      "RESULTS\n",
      "FIG\n",
      "DISCUSS\n",
      "DISCUSS\n",
      "DISCUSS\n",
      "DISCUSS\n",
      "DISCUSS\n",
      "DISCUSS\n",
      "DISCUSS\n",
      "DISCUSS\n",
      "METHODS\n",
      "METHODS\n",
      "METHODS\n",
      "METHODS\n",
      "METHODS\n",
      "METHODS\n",
      "METHODS\n",
      "METHODS\n",
      "METHODS\n",
      "METHODS\n",
      "METHODS\n",
      "METHODS\n",
      "METHODS\n",
      "METHODS\n",
      "METHODS\n",
      "METHODS\n",
      "METHODS\n",
      "METHODS\n",
      "METHODS\n",
      "METHODS\n",
      "METHODS\n",
      "METHODS\n",
      "METHODS\n",
      "METHODS\n",
      "METHODS\n",
      "METHODS\n",
      "METHODS\n",
      "METHODS\n",
      "METHODS\n",
      "METHODS\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n",
      "REF\n"
     ]
    }
   ],
   "source": [
    "for passage in r_out['documents'][0]['passages']:\n",
    "    print(passage['infons']['section_type'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# affiliations https://www.ncbi.nlm.nih.gov/pmc/tools/get-metadata/"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "ac_knowl",
   "language": "python",
   "name": "ac_knowl"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
